---
title: "vcp_barbar_deseq2_overlap"
author: "Oliver Ziff"
date: "11/11/2020"
output: html_document
---

```{r setup, include=FALSE}
.libPaths("/camp/lab/luscomben/home/users/ziffo/.conda/envs/r4.0.3/lib/R/library")
source("/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/scripts/astrocyte_a1_vcp_comparison_functions.R")
load("/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/deseq2.RData")
mass_spec.res <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-a1-als-meta/expression/mass-spec/mass_spec.res_vcp_vs_ctrl.rds")
```

## DESeq2 objects

Create sym links so that all Salmon results accessible in /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

```{bash}
cd /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/astrocyte-fractionation-bulk-rnaseq/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-c9orf72-ipsc-birger-2019/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-sod1-ipsc-tyzack-2017/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/purified-cortex-human-zhang-2016/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/regional-astrocytes-ipsc-bradley-2019/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/cytokine-astrocyte-ipsc-perriot-2018/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/inflammation-astrocyte-ipsc-santos-2017/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/il1b-human-astrocyte-telese-2021/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-membralin-mouse-jiang-2019/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon
ln -s /camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-tdp43-mouse-peng-2020/nfcore/star_salmon/* /camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon

find . -xtype l -delete # remove only broken soft links
```


```{r}
dir <- "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/alignment/salmon"

# Patani alone
patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv") %>% filter(fraction == "whole", cellline != "ctrl2") %>% distinct(fraction_cellline, .keep_all = TRUE) %>% # 
  mutate(group = vcp, replicate = c(1,2,1,2), sample = cellline, file_salmon = file.path(dir, paste("ac_who_", vcp, "_R", replicate, sep = ""), "quant.sf")) %>%
  dplyr::select(group, replicate, sample, file_salmon) %>% mutate(group = factor(group, levels = c("ctrl", "vcp")))
patani.files <- patani$file_salmon
names(patani.files) <- patani$sample
rownames(patani) <- patani$sample
patani.txi <- tximport(patani.files, type="salmon", tx2gene=tx2gene)
patani.dds <- DESeqDataSetFromTximport(patani.txi, colData = patani, design = ~ group)
patani.dds <- DESeq(patani.dds) # run DESeq2
patani.vsd <- vst(patani.dds, blind=FALSE) # VST transformation)
patani.res <- DESeq2::results(patani.dds, name = "group_vcp_vs_ctrl") # Astrocyte VCP vs CTRL
gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(patani.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
patani.res <- patani.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Barbar alone
barbar <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/sample-details/samplesheet.csv") %>% 
  mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), reactivity = c("A0","A1","A0","A1","A0","A1"), group = factor(reactivity, levels = c("A0", "A1"))) %>% dplyr::select(group, replicate, sample, file_salmon)
barbar.files <- barbar$file_salmon
names(barbar.files) <- barbar$sample
rownames(barbar) <- barbar$sample
barbar.txi <- tximport(barbar.files, type="salmon", tx2gene=tx2gene)
barbar.dds <- DESeqDataSetFromTximport(barbar.txi, colData = barbar, design = ~ group)
barbar.dds <- DESeq(barbar.dds) # run DESeq2
barbar.vsd <- vst(barbar.dds, blind=FALSE) # VST transformation)
resultsNames(barbar.dds)
barbar.res <- DESeq2::results(barbar.dds, name = "group_A1_vs_A0") # Astrocyte VCP vs CTRL
gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(barbar.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
barbar.res <- barbar.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Patani & Zhang & Barbar & Bradley together
zhang <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/purified-cortex-human-zhang-2016/sample-details/samplesheet.csv") %>% 
  filter(!group %in% c("astrocyte_mouse_cortex", "astrocyte_adult_tumour", "astrocyte_adult_tumour", "astrocyte_adult_hippocampus", "endothelial_adult_temporal_lobe", "whole_adult_temporal_lobe")) %>% 
  mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), celltype_source = paste(celltype,source, sep = "_"), region = gsub("temporal_lobe", "cortex", region)) %>% 
  dplyr::select(group, replicate, region, celltype, celltype_source, sample, file_salmon)
patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv") %>% 
  filter(fraction == "whole", cellline != "ctrl2") %>% #vcp == "ctrl", 
  distinct(fraction_cellline, .keep_all = TRUE) %>% 
  mutate(celltype= "astrocyte", group = "astrocyte_ipsc", replicate = c(1,2,1,2), rpt = c(1,2,3,4), region = "ipsc", celltype_source = "astrocyte_ipsc", sample = paste(group, rpt, sep = "_"), file_salmon = file.path(dir, paste("ac_who_", vcp, "_R", replicate, sep = ""), "quant.sf")) %>% dplyr::select(group, replicate, region, celltype, celltype_source, sample, file_salmon)
barbar <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/sample-details/samplesheet.csv") %>% 
  mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), reactivity = c("A0","A1","A0","A1","A0","A1"), region = "ipsc", celltype_source = "barbar_ipsc", celltype= "astrocyte", group = "astrocyte_ipsc", sample = paste("CD49", group, replicate, sep = "_")) %>% filter(reactivity != "A1") %>% dplyr::select(group, replicate, region, celltype, celltype_source, sample, file_salmon)
bradley <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/regional-astrocytes-ipsc-bradley-2019/sample-details/samplesheet.csv") %>%
  mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), sample = paste(group, "_R", replicate, sep = ""), region = "ipsc", celltype_source = "bradley_ipsc", celltype= "astrocyte", group = "astrocyte_ipsc") %>%  dplyr::select(group, replicate, region, celltype, celltype_source, sample, file_salmon) %>% distinct(sample, .keep_all = TRUE)
zhang_barbar_bradley_patani.sampleTable <- bind_rows(zhang, patani, barbar, bradley) %>% mutate(celltype_source = factor(celltype_source, levels = c("astrocyte_ipsc", "barbar_ipsc","bradley_ipsc", "astrocyte_fetal", "astrocyte_adult", "oligodendrocyte_adult", "neuron_adult", "myeloid_adult", "endothelial_adult", "whole_adult"))) %>% as.data.frame()
zhang_barbar_bradley_patani.files <- zhang_barbar_bradley_patani.sampleTable$file_salmon
names(zhang_barbar_bradley_patani.files) <- zhang_barbar_bradley_patani.sampleTable$sample
rownames(zhang_barbar_bradley_patani.sampleTable) <- zhang_barbar_bradley_patani.sampleTable$sample
zhang_barbar_bradley_patani.txi <- tximport(zhang_barbar_bradley_patani.files, type="salmon", tx2gene=tx2gene)
zhang_barbar_bradley_patani.dds <- DESeqDataSetFromTximport(zhang_barbar_bradley_patani.txi, colData = zhang_barbar_bradley_patani.sampleTable, design = ~ group)
zhang_barbar_bradley_patani.vsd <- vst(zhang_barbar_bradley_patani.dds, blind=FALSE) # VST transformation)

# Tyzack SOD1
tyzack <- read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-sod1-ipsc-tyzack-2017/sample-details/metadata.xlsx") %>%
  mutate(replicate = c(1,2,3,4,1,2,3), sample = paste("ac_tyz_", condition, "_R", replicate, sep = ""), file_salmon = file.path(dir, paste("ac_tyz_", condition, "_R", replicate, sep = ""), "quant.sf"), dataset = "tyzack",
         group = factor(gsub("sod1", "als", condition), levels = c("ctrl", "als"))) %>%  dplyr::select(group, sample, dataset, file_salmon) 
tyzack.files <- tyzack$file_salmon
names(tyzack.files) <- tyzack$sample
rownames(tyzack) <- tyzack$sample
tyzack.txi <- tximport(tyzack.files, type="salmon", tx2gene=tx2gene)
tyzack.dds <- DESeqDataSetFromTximport(tyzack.txi, colData = tyzack, design = ~ group)
tyzack.dds <- DESeq(tyzack.dds) # run DESeq2
tyzack.vsd <- vst(tyzack.dds, blind=FALSE) # VST transformation)
tyzack.res <- DESeq2::results(tyzack.dds, name = "group_als_vs_ctrl")
gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(tyzack.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
tyzack.res <- tyzack.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Birger C9orf72
birger <-read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-c9orf72-ipsc-birger-2019/sample-details/metadata.xlsx") %>%
  mutate(condition = gsub("C9orf72", "c9orf72", condition), condition = gsub("control", "ctrl", condition), group = factor(gsub("c9orf72", "als", condition), levels = c("ctrl", "als")), replicate = c(1,2,1,2), 
         file_salmon = file.path(dir, paste("ac_bir_", condition, "_R", replicate, sep = ""), "quant.sf"), dataset = "birger") %>%  dplyr::select(group, sample, dataset, file_salmon) 
birger.files <- birger$file_salmon
names(birger.files) <- birger$sample
rownames(birger) <- birger$sample
birger.txi <- tximport(birger.files, type="salmon", tx2gene=tx2gene)
birger.dds <- DESeqDataSetFromTximport(birger.txi, colData = birger, design = ~ group)
birger.dds <- DESeq(birger.dds) # run DESeq2
birger.vsd <- vst(birger.dds, blind=FALSE) # VST transformation)
birger.res <- DESeq2::results(birger.dds, name = "group_als_vs_ctrl")
gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(birger.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
birger.res <- birger.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Cleveland bacTRAP alone
cleveland <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-sod1-mouse-bacTRAP-cleveland-2015/sample-details/samplesheet.csv") %>% 
  mutate(sample = paste(group, "_R", replicate, sep = ""), file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), condition = gsub("ac_sun_", "", group), group = factor(condition, levels = c("ctrl", "sod1"))) %>% dplyr::select(group, replicate, sample, file_salmon)
cleveland.files <- cleveland$file_salmon
names(cleveland.files) <- cleveland$sample
rownames(cleveland) <- cleveland$sample
cleveland.txi <- tximport(cleveland.files, type="salmon", tx2gene = mus.musculus.tx2gene)
cleveland.dds <- DESeqDataSetFromTximport(cleveland.txi, colData = cleveland, design = ~ group)
cleveland.dds <- DESeq(cleveland.dds) # run DESeq2
cleveland.vsd <- vst(cleveland.dds, blind=FALSE) # VST transformation)
resultsNames(cleveland.dds)
cleveland.res <- DESeq2::results(cleveland.dds, name = "group_sod1_vs_ctrl") # Astrocyte VCP vs CTRL
gene2ens_filtered <-  filter(mus.musculus.gene2ens, gene_id %in% rownames(cleveland.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
cleveland.res <- cleveland.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% mutate(gene_name= toupper(gene_name)) %>% arrange(padj) # join gene gene_names to res tibble, upper case enables overlap with hsapiens & order by padj

# Peng TDP43 alone
peng <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-tdp43-mouse-peng-2020/sample-details/samplesheet.csv") %>% filter(group != "mou_tdp43_het") %>%
  mutate(sample = paste(group, "_R", replicate, sep = ""), file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), condition = gsub("mou_tdp43_", "", group), group = factor(condition, levels = c("ctrl", "ko"))) %>% dplyr::select(group, replicate, sample, file_salmon)
peng.files <- peng$file_salmon
names(peng.files) <- peng$sample
rownames(peng) <- peng$sample
peng.txi <- tximport(peng.files, type="salmon", tx2gene = mus.musculus.tx2gene)
peng.dds <- DESeqDataSetFromTximport(peng.txi, colData = peng, design = ~ group)
peng.dds <- DESeq(peng.dds) # run DESeq2
peng.vsd <- vst(peng.dds, blind=FALSE) # VST transformation)
resultsNames(peng.dds)
peng.res <- DESeq2::results(peng.dds, name = "group_ko_vs_ctrl") # Astrocyte VCP vs CTRL
gene2ens_filtered <-  filter(mus.musculus.gene2ens, gene_id %in% rownames(peng.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
peng.res <- peng.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% mutate(gene_name= toupper(gene_name)) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Jiang Membralin alone
jiang <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-membralin-mouse-jiang-2019/sample-details/samplesheet.csv") %>% 
  mutate(sample = paste(group, "_R", replicate, sep = ""), file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), condition = gsub("mou_mem_", "", group), group = factor(condition, levels = c("ctrl", "ko"))) %>% dplyr::select(group, replicate, sample, file_salmon)
jiang.files <- jiang$file_salmon
names(jiang.files) <- jiang$sample
rownames(jiang) <- jiang$sample
jiang.txi <- tximport(jiang.files, type="salmon", tx2gene = mus.musculus.tx2gene)
jiang.dds <- DESeqDataSetFromTximport(jiang.txi, colData = jiang, design = ~ group)
jiang.dds <- DESeq(jiang.dds) # run DESeq2
jiang.vsd <- vst(jiang.dds, blind=FALSE) # VST transformation)
resultsNames(jiang.dds)
jiang.res <- DESeq2::results(jiang.dds, name = "group_ko_vs_ctrl") # Astrocyte VCP vs CTRL
jiang.res <- jiang.res %>% as_tibble(rownames = "ensID") %>% left_join(mus.musculus.gene2ens, by=c("ensID"="gene_id")) %>% mutate(gene_name= toupper(gene_name)) %>% arrange(padj) 

# jiang.pcaData <- plotPCA(jiang.vsd, intgroup=c("group","replicate"), returnData=TRUE) #  
# jiang.percentVar <- round(100 * attr(jiang.pcaData, "percentVar"))
# jiang.pca <- ggplot(jiang.pcaData, aes(PC1, PC2, color=group.1)) + geom_text_repel(aes(label=colnames(jiang.vsd)), size = 2, max.overlaps = Inf) +  
#   scale_colour_manual( values = get_palette("npg", 2)) +  theme_bw() +  
#   theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
#   xlab(paste0("PC1: ",jiang.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",jiang.percentVar[2],"% variance")) + coord_fixed() + ggforce::geom_mark_ellipse()
# jiang.pca


# Patani & Zhang together
# zhang_patani.sampleTable <- bind_rows(zhang, patani) %>% mutate(celltype_source = factor(celltype_source, levels = c("astrocyte_ipsc", "astrocyte_fetal", "astrocyte_adult", "oligodendrocyte_adult", "neuron_adult", "myeloid_adult", "endothelial_adult", "whole_adult"))) %>% as.data.frame()
# zhang_patani.files <- zhang_patani.sampleTable$file_salmon
# names(zhang_patani.files) <- zhang_patani.sampleTable$sample
# rownames(zhang_patani.sampleTable) <- zhang_patani.sampleTable$sample
# zhang_patani.txi <- tximport(zhang_patani.files, type="salmon", tx2gene=tx2gene)
# zhang_patani.dds <- DESeqDataSetFromTximport(zhang_patani.txi, colData = zhang_patani.sampleTable, design = ~ group)
# zhang_patani.vsd <- vst(zhang_patani.dds, blind=FALSE) # VST transformation)

# # Patani & Bradley together
# bradley_patani.sampleTable <- bind_rows(bradley, patani) %>% as.data.frame()
# bradley_patani.files <- bradley_patani.sampleTable$file_salmon
# names(bradley_patani.files) <- bradley_patani.sampleTable$sample
# rownames(bradley_patani.sampleTable) <- bradley_patani.sampleTable$sample
# bradley_patani.txi <- tximport(bradley_patani.files, type="salmon", tx2gene=tx2gene)
# bradley_patani.dds <- DESeqDataSetFromTximport(bradley_patani.txi, colData = bradley_patani.sampleTable, design = ~ group)
# bradley_patani.vsd <- vst(bradley_patani.dds, blind=FALSE) # VST transformation)

# # ALS datasets ~ dataset + condition
# patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv") %>% filter(fraction == "whole", cellline != "ctrl2") %>% distinct(fraction_cellline, .keep_all = TRUE) %>% 
#   mutate(group = vcp, replicate = c(1,2,1,2), sample = cellline, file_salmon = file.path(dir, paste("ac_who_", vcp, "_R", replicate, sep = ""), "quant.sf"), dataset = "patani") %>%  dplyr::select(group, sample, dataset, file_salmon) 
# patani$group <- gsub("vcp", "als", patani$group)
# 
# birger <-read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-c9orf72-ipsc-birger-2019/sample-details/metadata.xlsx")
# birger$group <- gsub("C9orf72", "als", birger$condition) %>% gsub("control", "ctrl",.)
# birger$condition <- gsub("C9orf72", "c9orf72", birger$condition) %>% gsub("control", "ctrl",.)
# birger <- birger %>%  mutate(group, replicate = c(1,2,1,2), sample, file_salmon = file.path(dir, paste("ac_bir_", condition, "_R", replicate, sep = ""), "quant.sf"), dataset = "birger") %>%  dplyr::select(group, sample, dataset, file_salmon) 
# 
# tyzack <- read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/astrocyte-sod1-ipsc-tyzack-2017/sample-details/metadata.xlsx") %>%
#   mutate(group = condition, replicate = c(1,2,3,4,1,2,3), sample = paste("ac_tyz_", condition, "_R", replicate, sep = ""), file_salmon = file.path(dir, paste("ac_tyz_", condition, "_R", replicate, sep = ""), "quant.sf"), dataset = "tyzack") %>%  dplyr::select(group, sample, dataset, file_salmon) 
# tyzack$group <- gsub("sod1", "als", tyzack$group)
#   
# als_datasets <- bind_rows(patani, birger, tyzack) %>% mutate(group = factor(group, levels = c("ctrl", "als"))) #bind rows
# 
# als_datasets.files <- als_datasets$file_salmon
# names(als_datasets.files) <- als_datasets$sample
# rownames(als_datasets) <- als_datasets$sample
# 
# als_datasets.txi <- tximport(als_datasets.files, type="salmon", tx2gene=tx2gene)
# als_datasets.dds <- DESeqDataSetFromTximport(als_datasets.txi, colData = als_datasets, design = ~ dataset + group)
# als_datasets.dds <- DESeq(als_datasets.dds) # run DESeq2
# als_datasets.vsd <- vst(als_datasets.dds, blind=FALSE) # VST transformation)
# 
# resultsNames(als_datasets.dds)
# als_datasets.res <- DESeq2::results(als_datasets.dds, name = "group_als_vs_ctrl") # Astrocyte VCP vs CTRL
# gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(als_datasets.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
# als_datasets.res <- als_datasets.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj
# 
# # A1 datasets ~ dataset + condition
# barbar <- read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/sample-details/sample_table.xls")
# barbar$sample_group <- gsub("m", "", barbar$sample) %>% gsub("f", "",.) %>% gsub("u","",.)
# barbar <- barbar %>% mutate(replicate = c(1,1,2,2,3,3), file_salmon = file.path(dir, paste(sample_group, "_R", replicate, sep = ""), "quant.sf"), group = reactivity, dataset = "barbar") %>% 
#   dplyr::select(group, sample, dataset, file_salmon)
# 
# perriot <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/cytokine-astrocyte-ipsc-perriot-2018/sample-details/samplesheet.csv") %>% 
#   mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), sample = paste(group, "_R", replicate, sep = ""), treatment = str_split_fixed(group, "_", 2)[,2], celltype = str_split_fixed(group, "_", 2)[,1], group = case_when(treatment == "Unstim" ~ "A0", TRUE ~ "A1"), dataset = "perriot") %>%  filter(treatment != "FBS") %>% #, celltype != "fetal"
#   dplyr::select(group, sample, dataset, file_salmon)
# 
# santos <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/inflammation-astrocyte-ipsc-santos-2017/sample-details/samplesheet.csv") %>% 
#   mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), sample = paste(group, "_R", replicate, sep = ""), treatment = str_split_fixed(group, "_", 2)[,2], group = case_when(treatment == "CTRL" ~ "A0", TRUE ~ "A1"), dataset = "santos") %>% dplyr::select(group, sample, dataset, file_salmon)
# 
# telese <- read.csv("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/il1b-human-astrocyte-telese-2021/sample-details/samplesheet.csv") %>% 
#   mutate(file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf"), sample = paste(group, "_R", replicate, sep = ""), treatment = str_split_fixed(group, "_", 3)[,3], group = case_when(treatment == "ctrl" ~ "A0", TRUE ~ "A1"), dataset = "telese") %>% dplyr::select(group, sample, dataset, file_salmon)
# 
# a1_datasets <- bind_rows(barbar) %>% #, perriot, santos, telese) %>% 
#   mutate(group = factor(group, levels = c("A0", "A1"))) #bind rows
# 
# a1_datasets.files <- a1_datasets$file_salmon
# names(a1_datasets.files) <- a1_datasets$sample
# rownames(a1_datasets) <- a1_datasets$sample
# a1_datasets.txi <- tximport(a1_datasets.files, type="salmon", tx2gene=tx2gene)
# a1_datasets.dds <- DESeqDataSetFromTximport(a1_datasets.txi, colData = a1_datasets, design = ~ group)
# a1_datasets.dds <- DESeq(a1_datasets.dds) # run DESeq2
# a1_datasets.vsd <- vst(a1_datasets.dds, blind=FALSE) # VST transformation)
# resultsNames(a1_datasets.dds)
# a1_datasets.res <- DESeq2::results(a1_datasets.dds, name = "group_A1_vs_A0") # Astrocyte VCP vs CTRL
# gene2ens_filtered <-  filter(gene2ens, gene_id %in% rownames(a1_datasets.res)) %>% unique # filter gene to ensID to only ENS names in the results tibble
# a1_datasets.res <- a1_datasets.res %>% as_tibble(rownames = "ensID") %>% left_join(gene2ens_filtered, by=c("ensID"="gene_id")) %>% arrange(padj) # join gene gene_nameS to res tibble & order by padj

# Patani & Barbar together
# patani <- read.csv("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/sample-details/sample_table.csv") %>% filter(fraction == "whole") %>% distinct(fraction_cellline, .keep_all = TRUE) %>% #cellline != "ctrl2", ?remove ctrl2
#   mutate(group = vcp, replicate = c(1,2,3,1,2), sample = cellline, file_salmon = file.path(dir, paste("ac_who_", vcp, "_R", replicate, sep = ""), "quant.sf")) %>% 
#   dplyr::select(group, replicate, sample, file_salmon)
# barbar <- read_excel("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/sample-details/sample_table.xls") %>% 
  # mutate(group = factor(reactivity, levels = c("A0", "A1")), replicate = c(1,1,2,2,3,3), file_salmon = file.path(dir, paste(group, "_R", replicate, sep = ""), "quant.sf")) %>%
  # dplyr::select(group, replicate, sample, file_salmon)
# sampleTable <- bind_rows(barbar, patani) %>% mutate(group = factor(group, levels = c("ctrl", "vcp", "A0", "A1"))) %>% as.data.frame()
# 
# files <- sampleTable$file_salmon
# names(files) <- sampleTable$sample
# rownames(sampleTable) <- sampleTable$sample
# 
# txi <- tximport(files, type="salmon", tx2gene=tx2gene)
# dds <- DESeqDataSetFromTximport(txi, colData = sampleTable, design = ~ group)
# vsd <- vst(dds, blind=FALSE) # VST transformation)


# overlap all datasets in single table: VCP, SOD1, C9orf72, A1, SOD1 TRAP, TDP43 KO & membralin KO
patani.dge <- patani.res %>% dplyr::select(gene_name, hsap.ensembl = ensID, log2FC.VCP = log2FoldChange, FDR.VCP = padj) %>% mutate(direction.VCP = case_when(FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "up", FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "down", TRUE ~ "non-significant"))
birger.dge <- birger.res %>% dplyr::select(gene_name, hsap.ensembl = ensID, log2FC.C9orf72 = log2FoldChange, FDR.C9orf72 = padj) %>% mutate(direction.C9orf72 = case_when(FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0 ~ "up", FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0 ~ "down", TRUE ~ "non-significant"))
tyzack.dge <- tyzack.res %>% dplyr::select(gene_name, hsap.ensembl = ensID, log2FC.SOD1 = log2FoldChange, FDR.SOD1 = padj) %>% mutate(direction.SOD1 = case_when(FDR.SOD1 < 0.05 & log2FC.SOD1 > 0 ~ "up", FDR.SOD1 < 0.05 & log2FC.SOD1 < 0 ~ "down", TRUE ~ "non-significant"))
barbar.dge <- barbar.res %>% dplyr::select(gene_name, hsap.ensembl = ensID, log2FC.A1 = log2FoldChange, FDR.A1 = padj) %>% mutate(direction.A1 = case_when(FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "up", FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "down", TRUE ~ "non-significant"))
cleveland.dge <- cleveland.res %>% dplyr::select(gene_name, mmus.ensembl = ensID, log2FC.TRAP = log2FoldChange, FDR.TRAP = padj) %>% mutate(direction.TRAP = case_when(FDR.TRAP < 0.05 & log2FC.TRAP > 0 ~ "up", FDR.TRAP < 0.05 & log2FC.TRAP < 0 ~ "down", TRUE ~ "non-significant"))
peng.dge <- peng.res %>% dplyr::select(gene_name, mmus.ensembl = ensID, log2FC.TDP43 = log2FoldChange, FDR.TDP43 = padj) %>% mutate(direction.TDP43 = case_when(FDR.TDP43 < 0.05 & log2FC.TDP43 > 0 ~ "up", FDR.TDP43 < 0.05 & log2FC.TDP43 < 0 ~ "down", TRUE ~ "non-significant"))
jiang.dge <- jiang.res %>% dplyr::select(gene_name, mmus.ensembl = ensID, log2FC.MEM = log2FoldChange, FDR.MEM = padj) %>% mutate(direction.MEM = case_when(FDR.MEM < 0.05 & log2FC.MEM > 0 ~ "up", FDR.MEM < 0.05 & log2FC.MEM < 0 ~ "down", TRUE ~ "non-significant"))
mass_spec.summary <- mass_spec.res %>% dplyr::select(gene_name, logFC.VCP.MS = lfc, FDR.VCP.MS = padj)
patani_birger_tyzack_barbar.dge <- full_join(birger.dge, patani.dge, by = c("gene_name", "hsap.ensembl")) %>% full_join(tyzack.dge, by = c("gene_name", "hsap.ensembl")) %>% full_join(barbar.dge, by = c("gene_name", "hsap.ensembl"))
cleveland_peng_jiang_mass_spec.dge <- full_join(cleveland.dge, peng.dge, by = c("gene_name", "mmus.ensembl")) %>% full_join(jiang.dge, by = c("gene_name", "mmus.ensembl")) %>% full_join(mass_spec.summary, by = c("gene_name"))
all_datasets.dge <- full_join(patani_birger_tyzack_barbar.dge, cleveland_peng_jiang_mass_spec.dge, by = "gene_name") %>% select(gene_name, hsap.ensembl, mmus.ensembl, everything())
write.csv(all_datasets.dge, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Table.overlap_all_datasets.dge.csv", row.names = FALSE)


patani_birger_tyzack_barbar.dge <- patani_birger_tyzack_barbar.dge %>% 
  mutate(
    # overlap = case_when(FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & FDR.A1 < 0.05 ~ "overlapping", TRUE ~ "None"),
         overlap_direction = case_when(FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & FDR.A1 < 0.05 & log2FC.VCP > 0 & log2FC.C9orf72 > 0 & log2FC.SOD1 > 0 & log2FC.A1 > 0 ~ "all up", 
                                       FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & FDR.A1 < 0.05 & log2FC.VCP < 0 & log2FC.C9orf72 < 0 & log2FC.SOD1 < 0 & log2FC.A1 < 0 ~ "all down", 
                                       FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & log2FC.VCP > 0 & log2FC.C9orf72 > 0 & log2FC.SOD1 > 0 ~ "VCP C9orf72 SOD1 up", 
                                       FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & log2FC.VCP < 0 & log2FC.C9orf72 < 0 & log2FC.SOD1 < 0 ~ "VCP C9orf72 SOD1 down", 
                                       FDR.A1 < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 > 0 & log2FC.C9orf72 > 0 & log2FC.SOD1 > 0 ~ "A1 C9orf72 SOD1 up", 
                                       FDR.A1 < 0.05 & FDR.C9orf72 < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 < 0 & log2FC.C9orf72 < 0 & log2FC.SOD1 < 0 ~ "A1 C9orf72 SOD1 down", 
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 > 0 & log2FC.VCP > 0 & log2FC.SOD1 > 0 ~ "A1 VCP SOD1 up", 
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 < 0 & log2FC.VCP < 0 & log2FC.SOD1 < 0 ~ "A1 VCP SOD1 down", 
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & log2FC.A1 > 0 & log2FC.VCP > 0 & log2FC.C9orf72 > 0 ~ "A1 VCP C9orf72 up", 
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & log2FC.A1 < 0 & log2FC.VCP < 0 & log2FC.C9orf72 < 0 ~ "A1 VCP C9orf72 down",
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & log2FC.A1 > 0 & log2FC.VCP > 0 ~ "A1 VCP up", 
                                       FDR.A1 < 0.05 & FDR.VCP < 0.05 & log2FC.A1 < 0 & log2FC.VCP < 0 ~ "A1 VCP down",
                                       FDR.A1 < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 > 0 & log2FC.SOD1 > 0 ~ "A1 SOD1 up", 
                                       FDR.A1 < 0.05 & FDR.SOD1 < 0.05 & log2FC.A1 < 0 & log2FC.SOD1 < 0 ~ "A1 SOD1 down",
                                       FDR.A1 < 0.05 & FDR.C9orf72 < 0.05 & log2FC.A1 > 0 & log2FC.C9orf72 > 0 ~ "A1 SOD1 up", 
                                       FDR.A1 < 0.05 & FDR.C9orf72 < 0.05 & log2FC.A1 < 0 & log2FC.C9orf72 < 0 ~ "A1 SOD1 down",
                                       FDR.VCP < 0.05 & FDR.SOD1 < 0.05 & log2FC.VCP > 0 & log2FC.SOD1 > 0 ~ "VCP SOD1 up", 
                                       FDR.VCP < 0.05 & FDR.SOD1 < 0.05 & log2FC.VCP < 0 & log2FC.SOD1 < 0 ~ "VCP SOD1 down",
                                       FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & log2FC.VCP > 0 & log2FC.C9orf72 > 0 ~ "VCP C9orf72 up", 
                                       FDR.VCP < 0.05 & FDR.C9orf72 < 0.05 & log2FC.VCP < 0 & log2FC.C9orf72 < 0 ~ "VCP C9orf72 down",
                                       FDR.SOD1 < 0.05 & FDR.C9orf72 < 0.05 & log2FC.SOD1 > 0 & log2FC.C9orf72 > 0 ~ "SOD1 C9orf72 up", 
                                       FDR.SOD1 < 0.05 & FDR.C9orf72 < 0.05 & log2FC.SOD1 < 0 & log2FC.C9orf72 < 0 ~ "SOD1 C9orf72 down",
                                       FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 up", FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 down",
                                       FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP up", FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP down",
                                       FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0 ~ "C9orf72 up", FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0 ~ "C9orf72 down",
                                       FDR.SOD1 < 0.05 & log2FC.SOD1 > 0 ~ "SOD1 up", FDR.SOD1 < 0.05 & log2FC.SOD1 < 0 ~ "SOD1 down",
                                       TRUE ~ "None")) %>%  arrange(overlap_direction)
table(patani_birger_tyzack_barbar.dge$overlap)
table(patani_birger_tyzack_barbar.dge$overlap_direction)
patani_birger_tyzack_barbar.dge
write.csv(patani_birger_tyzack_barbar.dge, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Table.overlapping.dge_VCP.SOD1.C9orf72.A1.csv", row.names = FALSE)

write.csv(mass_spec.res, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/mass-spec/Table.mass_spec.csv", row.names = FALSE)

save.image("/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/deseq2.RData")
```


## Fig S1 Patani, Barbar, Bradkey, Zhang clustering

```{r}
zhang_barbar_bradley_patani.pcaData <- plotPCA(zhang_barbar_bradley_patani.vsd, intgroup=c("celltype","group", "region", "replicate", "celltype_source"), returnData=TRUE) #  
zhang_barbar_bradley_patani.percentVar <- round(100 * attr(zhang_barbar_bradley_patani.pcaData, "percentVar"))
zhang_barbar_bradley_patani.pca <- ggplot(zhang_barbar_bradley_patani.pcaData, aes(PC1, PC2, color=celltype_source, shape = region)) + geom_text_repel(aes(label=colnames(zhang_barbar_bradley_patani.vsd)), size = 2, max.overlaps = Inf) +  
  scale_colour_manual( values = get_palette("npg", 8)) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
  xlab(paste0("PC1: ",zhang_barbar_bradley_patani.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",zhang_barbar_bradley_patani.percentVar[2],"% variance")) + coord_fixed() + ggforce::geom_mark_ellipse() +
  annotate(geom="text", x=-35, y = 9, label = "iPSC astrocytes", size = 3) +  annotate(geom="text", x=4, y = 8, label = "Fetal astrocytes", size = 3) +  annotate(geom="text", x=26, y = -22, label = "Adult astrocytes", size = 3) +
  annotate(geom="text", x= 44, y = 6, label = "Adult Neuron", size = 3) +  annotate(geom="text", x=20, y = 30, label = "Adult Oligodendrocytes", size = 3) +  annotate(geom="text", x=16, y = 62, label = "Adult Myeloid", size = 3)
zhang_barbar_bradley_patani.pca
ggsave(zhang_barbar_bradley_patani.pca, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/zhang_barbar_bradley_patani.pca.png", height = 9, width = 9)

zhang_barbar_bradley_patani.sampleDists <- dist(t(assay(zhang_barbar_bradley_patani.vsd)))
zhang_barbar_bradley_patani.hc <- hclust(zhang_barbar_bradley_patani.sampleDists, method = "ward.D2")
zhang_barbar_bradley_patani.hc$labels <- gsub("_cortex", "", zhang_barbar_bradley_patani.hc$labels) %>% gsub("_temporal_lobe", "", .) %>% gsub("CD49", "barbar", .) %>% gsub("dorsal_forebrain", "bradley_ipsc_dfb", .) %>% gsub("ventral_forebrain", "bradley_ipsc_vfb", .) %>% gsub("ventral_spinalcord", "bradley_ipsc_vsc", .) %>% gsub("dorsal_spinalcord", "bradley_ipsc_dsc", .) %>% gsub("barbar_astrocyte", "barbar", .)
library("scales", "ggsci")
show_col(pal_npg("nrc")(8))
get_palette("npg", 8)
# astrocyte_ipsc "#E64B35FF"
# barbar_ipsc "#4DBBD5FF"
# bradley_ipsc "#00A087FF"
# astrocyte_fetal "#3C5488FF"
# astrocyte_adult "#F39B7FFF"
# oligodendrocyte_adult  "#8491B4FF"
# neuron_adult "#91D1C2FF"
# myeloid_adult"#DC0000FF"
zhang_barbar_bradley_patani.a <- c("#00A087FF", "#00A087FF","#00A087FF", "#00A087FF", "#00A087FF","#00A087FF", "#00A087FF", "#00A087FF","#00A087FF", "#00A087FF", "#00A087FF","#00A087FF",
                                   "#4DBBD5FF", "#4DBBD5FF","#4DBBD5FF",
                                   "#E64B35FF", "#E64B35FF", "#E64B35FF","#E64B35FF",
                                   "#3C5488FF","#3C5488FF","#3C5488FF","#3C5488FF","#3C5488FF","#3C5488FF",
                                   "#F39B7FFF","#F39B7FFF", "#F39B7FFF", "#F39B7FFF", "#F39B7FFF","#F39B7FFF","#F39B7FFF","#F39B7FFF", "#F39B7FFF", "#F39B7FFF", "#F39B7FFF","#F39B7FFF",
                                   "#DC0000FF","#DC0000FF","#DC0000FF",
                                   "#91D1C2FF",
                                   "#8491B4FF","#8491B4FF","#8491B4FF","#8491B4FF","#8491B4FF")
zhang_barbar_bradley_patani.dendrogram <- ggdendrogram(zhang_barbar_bradley_patani.hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = zhang_barbar_bradley_patani.a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
zhang_barbar_bradley_patani.dendrogram
ggsave(zhang_barbar_bradley_patani.dendrogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/zhang_barbar_bradley_patani.dendrogram_celltypes.png", height = 7, width = 4)

zhang_barbar_bradley_patani.clustering <- ggarrange(zhang_barbar_bradley_patani.pca, zhang_barbar_bradley_patani.dendrogram, nrow = 2, labels = c("A", "B"))
zhang_barbar_bradley_patani.clustering
ggsave(zhang_barbar_bradley_patani.clustering, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/FigS1.zhang_barbar_bradley_patani.clustering.png", height = 11, width = 8)

# PCA
# zhang_patani.pcaData <- plotPCA(zhang_patani.vsd, intgroup=c("celltype","group", "region", "replicate", "celltype_source"), returnData=TRUE) #  
# zhang_patani.percentVar <- round(100 * attr(zhang_patani.pcaData, "percentVar"))
# zhang_patani.pca <- ggplot(zhang_patani.pcaData, aes(PC1, PC2, color=celltype_source, shape = region, group = celltype_source)) + #geom_text_repel(aes(label=celltype_source), size = 3) +  
#   scale_colour_manual( values = get_palette("npg", 10)) +  theme_bw() +  
#   theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
#   xlab(paste0("PC1: ",zhang_patani.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",zhang_patani.percentVar[2],"% variance")) + coord_fixed() +# stat_ellipse() +
#   ggforce::geom_mark_ellipse()
# zhang_patani.pca
# ggsave(zhang_patani.pca, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/zhang_patani.pca_celltypes.png", height = 9, width = 9)

# 
# zhang_patani.sampleDists <- dist(t(assay(zhang_patani.vsd)))
# zhang_patani.hc <- hclust(zhang_patani.sampleDists, method = "ward.D2")
# zhang_patani.a <- c("firebrick2", "firebrick2","firebrick2","firebrick2", "firebrick2", "firebrick2","firebrick2", "firebrick2", "firebrick2", "firebrick2","firebrick2", "firebrick2", "firebrick2", "firebrick2","firebrick2", "firebrick2", "firebrick2", "firebrick2","firebrick2", "firebrick2","firebrick2", "firebrick2", "firebrick2", "firebrick2","firebrick2","dodgerblue3","dodgerblue3","dodgerblue3","#00A087FF","#3C5488FF","#3C5488FF","#3C5488FF","#3C5488FF","#3C5488FF")
# zhang_patani.dendrogram <- ggdendrogram(zhang_patani.hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = zhang_patani.a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
# zhang_patani.dendrogram
# ggsave(zhang_patani.dendrogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/zhang_patani.dendrogram_celltypes.png", height = 7, width = 4)
# 
# 
# zhang_patani.clustering <- ggarrange(zhang_patani.pca, zhang_patani.dendrogram, nrow = 2, labels = c("A", "B"))
# zhang_patani.clustering
# ggsave(zhang_patani.clustering, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/FigS1.zhang_patani.clustering.png", height = 11, width = 8)
```

Only Patani vs Bradley

```{r}
# PCA
bradley_patani.pcaData <- plotPCA(bradley_patani.vsd, intgroup=c("group", "replicate"), returnData=TRUE) #  
bradley_patani.percentVar <- round(100 * attr(bradley_patani.pcaData, "percentVar"))
bradley_patani.pca <- ggplot(bradley_patani.pcaData, aes(PC1, PC2, color=group.1)) + geom_text_repel(aes(label=colnames(bradley_patani.vsd)), size = 3, max.overlaps = Inf) +  
  scale_colour_manual( values = get_palette("npg", 14)) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
  xlab(paste0("PC1: ",bradley_patani.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",bradley_patani.percentVar[2],"% variance")) + coord_fixed() + ggforce::geom_mark_ellipse()
bradley_patani.pca
ggsave(bradley_patani.pca, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/bradley_patani.pca.png", height = 9, width = 9)


bradley_patani.sampleDists <- dist(t(assay(bradley_patani.vsd)))
bradley_patani.hc <- hclust(bradley_patani.sampleDists, method = "ward.D2")
bradley_patani.a <- c("#E64B35FF", "#E64B35FF", "#E64B35FF","#E64B35FF", "#4DBBD5FF", "#4DBBD5FF", "#4DBBD5FF","#00A087FF", "#00A087FF", "#00A087FF", "#3C5488FF","#3C5488FF", "#3C5488FF","#F39B7FFF","#F39B7FFF","#F39B7FFF")
bradley_patani.dendrogram <- ggdendrogram(bradley_patani.hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = bradley_patani.a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
bradley_patani.dendrogram
ggsave(bradley_patani.dendrogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/bradley_patani.dendrogram_celltypes.png", height = 7, width = 4)

bradley_patani.clustering <- ggarrange(bradley_patani.pca, bradley_patani.dendrogram, nrow = 2, labels = c("A", "B"))
bradley_patani.clustering
ggsave(bradley_patani.clustering, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Bradley_patani.clustering.png", height = 11, width = 8)
```


## Fig 1B Astrocyte marker heatmap {.tabset .tabset-fade .tabset-pills}
***

We confirmed the astrocytic identity of our cultures through RNA sequencing (Fig. 1B, D).

```{r}
# plot heatmap of astrocyte & MN markers
astrocyte.genes <- c("GFAP","AQP4", "VIM", "S100B","NFIX","EDNRB","MLC1","CHRDL1","F3","FZD2","EZR","RFX4","BMPR1B","GLI3","TMEM47","SLC9A3R1","FGFR3","CCDC80","CYBRD1","TNC","S1PR1","CBS","PBXIP1","SLC7A2","GJA1","EGFR","SOX9","ID4","EMP2","MGST1", "ACSBG1")

# "PLA2G7","NTSR2","THRSP","LONRF3","GJB6","SLC14A1","ALDOC","CLDN10","SLC7A10","KCNE5","TTPA","ENTPD2","ALDH1L1","FMO1","GRIN2C","PPP1R3C","SLC39A12","TLCD1","FAM20A","HAPLN1","PRODH","PDK4","TLR3","PLCD4","ACOT11","CTH","PPP1R3G","ATP1A2","SLC4A4","VCAM1","PAPSS2","EVA1A","AGT","SLC25A18","GLDC","DIO2","ADHFE1","SLC1A2","SLC15A2","SLC1A3","MERTK","HTRA1"

vsd_counts.marker <- assay(zhang_barbar_bradley_patani.vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id")  %>% filter(gene_name %in% astrocyte.genes) %>% rowwise() %>%
  mutate(astrocyte_ipsc = mean(c_across(starts_with("astrocyte_ipsc")), na.rm = TRUE), astrocyte_ipsc_barbar = mean(c_across(starts_with("CD49_astrocyte_ipsc")), na.rm = TRUE), 
         astrocyte_ipsc_spinalcord = mean(c_across(starts_with(c("ventral_spinalcord","dorsal_spinalcord"))), na.rm = TRUE), #dsc_astrocyte_ipsc = mean(c_across(starts_with("dorsal_spinalcord")), na.rm = TRUE),
         astrocyte_ipsc_forebrain = mean(c_across(starts_with(c("ventral_forebrain","dorsal_forebrain"))), na.rm = TRUE), #dfb_astrocyte_ipsc = mean(c_across(starts_with("dorsal_forebrain")), na.rm = TRUE), 
         astrocyte_fetal = mean(c_across(starts_with("astrocyte_fetal")), na.rm = TRUE), astrocyte_adult = mean(c_across(starts_with("astrocyte_adult_temporal")), na.rm = TRUE), 
         oligodendrocyte = mean(c_across(starts_with("oligodendrocyte")), na.rm = TRUE), 
         myeloid = mean(c_across(starts_with("myeloid")), na.rm = TRUE), neuron = mean(c_across(starts_with("neuron")), na.rm = TRUE)) %>% 
  dplyr::select(gene_name, astrocyte_ipsc, astrocyte_ipsc_barbar, astrocyte_ipsc_forebrain, astrocyte_ipsc_spinalcord, astrocyte_fetal, astrocyte_adult, oligodendrocyte, neuron, myeloid, -gene_id) %>%
  arrange(-(astrocyte_ipsc + astrocyte_ipsc_barbar + astrocyte_fetal + astrocyte_adult + astrocyte_ipsc_spinalcord - oligodendrocyte - myeloid - neuron))  #   
vsd_counts.marker.mat <- make_matrix(dplyr::select(vsd_counts.marker,-gene_name), pull(vsd_counts.marker,gene_name))
colnames(vsd_counts.marker.mat)[1] <- "**astrocyte_ipsc**" # change first gene name
celltype_marker_heatmap <- pheatmap(vsd_counts.marker.mat, cluster_rows=F, show_rownames=T, show_colnames = T, labels_column = colnames(vsd_counts.marker.mat), cluster_cols=T, fontsize = 8, scale = "row")
ggsave(celltype_marker_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Fig1B.celltype_marker_heatmap.png", height = 6, width = 4)

vsd_counts.marker <- assay(zhang_barbar_bradley_patani.vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id")  %>% filter(gene_name %in% astrocyte.genes) %>% rowwise() %>%
  mutate(astrocyte_ipsc = mean(c_across(starts_with("astrocyte_ipsc")), na.rm = TRUE), astrocyte_ipsc_barbar = mean(c_across(starts_with("CD49_astrocyte_ipsc")), na.rm = TRUE), 
         astrocyte_ipsc_ventral_cord = mean(c_across(starts_with("ventral_spinalcord")), na.rm = TRUE), #dsc_astrocyte_ipsc = mean(c_across(starts_with("dorsal_spinalcord")), na.rm = TRUE),
         # astrocyte_ipsc_forebrain = mean(c_across(starts_with(c("ventral_forebrain","dorsal_forebrain"))), na.rm = TRUE), #dfb_astrocyte_ipsc = mean(c_across(starts_with("dorsal_forebrain")), na.rm = TRUE), 
         astrocyte_fetal = mean(c_across(starts_with("astrocyte_fetal")), na.rm = TRUE), astrocyte_adult = mean(c_across(starts_with("astrocyte_adult_temporal")), na.rm = TRUE), 
         oligodendrocyte = mean(c_across(starts_with("oligodendrocyte")), na.rm = TRUE), 
         myeloid = mean(c_across(starts_with("myeloid")), na.rm = TRUE), neuron = mean(c_across(starts_with("neuron")), na.rm = TRUE)) %>% 
  dplyr::select(gene_name, astrocyte_ipsc, astrocyte_ipsc_barbar, astrocyte_ipsc_ventral_cord, astrocyte_fetal, astrocyte_adult, oligodendrocyte, neuron, myeloid, -gene_id) %>%
  arrange(-(astrocyte_ipsc + astrocyte_ipsc_barbar + astrocyte_fetal + astrocyte_adult + astrocyte_ipsc_ventral_cord - oligodendrocyte - myeloid - neuron))  #   
vsd_counts.marker.mat <- make_matrix(dplyr::select(vsd_counts.marker,-gene_name), pull(vsd_counts.marker,gene_name))

colnames(vsd_counts.marker.mat)[1] <- "**astrocyte_ipsc**" # change first gene name
celltype_marker_heatmap <- pheatmap(vsd_counts.marker.mat, cluster_rows=F, show_rownames=T, show_colnames = T, labels_column = colnames(vsd_counts.marker.mat), cluster_cols=T, fontsize = 8, scale = "row")
ggsave(celltype_marker_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Fig1B.celltype_marker_heatmap.png", height = 6, width = 4)
```


## Fig 1C Phagocytosis and glutamate update 

hiPSC astrocytes expressed several genes known to be critical in phagocytosis including AXL, ABCA1, MERTK, MEGF10, GAS6 (...) and glutamate uptake including EAAT1, EAAT2, GRIN2b, GRIA1, GRIK1 (...) (Fig. 1D)

```{r}
all.gene.phag_glut <- c(go.pathways.msigdb$GO_PHAGOCYTOSIS, glutamate.uptake.go) %>% unique
all.gene.phag_glut.df = data.frame("gene_name" = all.gene.phag_glut) %>% 
  mutate(group = case_when(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS ~ "phagocytosis",gene_name %in% glutamate.uptake.go ~ "glutamate uptake"))
annot <- data.frame(row.names = all.gene.phag_glut.df$gene_name, group = factor(all.gene.phag_glut.df$group))
annot_cols <- c("#E64B35FF", "#4DBBD5FF") # , "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF"
names(annot_cols) <- unique(all.gene.phag_glut.df$group)
annot_cols <- list(group = annot_cols)

vsd_counts.phag_glut <-  assay(zhang_barbar_bradley_patani.vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id")  %>% filter(gene_name %in% all.gene.phag_glut) %>% rowwise() %>%
 mutate(astrocyte_ipsc = mean(c_across(starts_with("astrocyte_ipsc")), na.rm = TRUE), astrocyte_ipsc_barbar = mean(c_across(starts_with("CD49_astrocyte_ipsc")), na.rm = TRUE), 
         astrocyte_ipsc_ventral_cord = mean(c_across(starts_with("ventral_spinalcord")), na.rm = TRUE), #dsc_astrocyte_ipsc = mean(c_across(starts_with("dorsal_spinalcord")), na.rm = TRUE),
         # astrocyte_ipsc_forebrain = mean(c_across(starts_with(c("ventral_forebrain","dorsal_forebrain"))), na.rm = TRUE), #dfb_astrocyte_ipsc = mean(c_across(starts_with("dorsal_forebrain")), na.rm = TRUE), 
         astrocyte_fetal = mean(c_across(starts_with("astrocyte_fetal")), na.rm = TRUE), astrocyte_adult = mean(c_across(starts_with("astrocyte_adult_temporal")), na.rm = TRUE), 
         oligodendrocyte = mean(c_across(starts_with("oligodendrocyte")), na.rm = TRUE), 
         myeloid = mean(c_across(starts_with("myeloid")), na.rm = TRUE), neuron = mean(c_across(starts_with("neuron")), na.rm = TRUE)) %>% 
  dplyr::select(gene_name, astrocyte_ipsc, astrocyte_ipsc_barbar, astrocyte_ipsc_ventral_cord, astrocyte_fetal, astrocyte_adult, oligodendrocyte, neuron, myeloid, -gene_id) %>%
  arrange(factor(gene_name, levels = all.gene.phag_glut)) %>% filter( (astrocyte_ipsc + astrocyte_ipsc_barbar + astrocyte_ipsc_ventral_cord + astrocyte_fetal + astrocyte_adult + oligodendrocyte + neuron + myeloid) > 32.5)

vsd_counts.phag_glut.mat <- make_matrix(dplyr::select(vsd_counts.phag_glut,-gene_name), pull(vsd_counts.phag_glut,gene_name))
colnames(vsd_counts.phag_glut.mat)[1] <- "**astrocyte_ipsc**" # change first gene name
phag_glut_heatmap <- pheatmap(vsd_counts.phag_glut.mat, cluster_rows=F, show_rownames=F, show_colnames = T, cluster_cols=F, fontsize = 8, scale = "row", gaps_row = 259,
                                    annotation_row = annot, annotation_names_col = F, annotation_legend = FALSE, annotation_colors = annot_cols, legend = TRUE)
ggsave(phag_glut_heatmap, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Fig1C.phag_glut_heatmap.png", height = 6, width = 4)

# most highly expressed in ipsc_astrocytes
glutamate.receptor.go <- unique(c(go.pathways.msigdb$GO_GLUTAMATE_BINDING, go.pathways.msigdb$GO_IONOTROPIC_GLUTAMATE_RECEPTOR_ACTIVITY, go.pathways.msigdb$GO_IONOTROPIC_GLUTAMATE_RECEPTOR_SIGNALING_PATHWAY, go.pathways.msigdb$GO_IONOTROPIC_GLUTAMATE_RECEPTOR_SIGNALING_PATHWAY, go.pathways.msigdb$GO_IONOTROPIC_GLUTAMATE_RECEPTOR_BINDING, go.pathways.msigdb$GO_AMPA_GLUTAMATE_RECEPTOR_CLUSTERING))

vsd_counts.phag_glut %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% arrange(-astrocyte_ipsc) %>% pull(gene_name)
vsd_counts.phag_glut %>% filter(gene_name %in% glutamate.uptake.go) %>% arrange(-astrocyte_ipsc) %>% pull(gene_name)
vsd_counts.phag_glut %>% filter(gene_name %in% glutamate.receptor.go) %>% arrange(-astrocyte_ipsc) %>% select(gene_name,astrocyte_ipsc)


vsd_counts.phag_glut %>% filter(gene_name == "SLC1A2")
patani.res  %>% filter(gene_name == "SLC1A2")
barbar.res  %>% filter(gene_name == "SLC1A2")

patani.res  %>% filter(gene_name == "GRIA2")
barbar.res  %>% filter(gene_name == "GRIA2")
assay(zhang_patani.vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id")   %>% filter(gene_name == "GRIA2") %>%
 mutate(astrocyte_ipsc = mean(c_across(starts_with("astrocyte_ipsc")), na.rm = TRUE), CD49_astrocyte_ipsc = mean(c_across(starts_with("CD49_astrocyte_ipsc")), na.rm = TRUE), 
         astrocyte_fetal = mean(c_across(starts_with("astrocyte_fetal")), na.rm = TRUE), astrocyte_adult = mean(c_across(starts_with("astrocyte_adult_temporal")), na.rm = TRUE), 
         oligodendrocyte = mean(c_across(starts_with("oligodendrocyte")), na.rm = TRUE), 
         myeloid = mean(c_across(starts_with("myeloid")), na.rm = TRUE), neuron = mean(c_across(starts_with("neuron")), na.rm = TRUE)) %>% 
  dplyr::select(gene_name, astrocyte_ipsc, CD49_astrocyte_ipsc, astrocyte_fetal, astrocyte_adult, oligodendrocyte, neuron, myeloid, -gene_id)
```


## Table S1. Phagocytosis & Glutamate uptake genes

```{r}
vsd_counts.phag_glut %>% mutate(category = case_when(gene_name %in% glutamate.uptake.go ~ "Glutamate Uptake", gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS ~ "Phagocytosis")) %>% arrange(desc(category)) %>% dplyr::select(gene_name, category, everything()) %>% write.csv(., "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/Table.S1.phagocytosis_glutamate.uptake_gene_expression.csv")


phagocytosis_glutamate.uptake <- cbindX(arrange(as_tibble_col(go.pathways.msigdb$GO_PHAGOCYTOSIS, column_name = "Phagocytosis"), Phagocytosis), 
        arrange(as_tibble_col(glutamate.uptake.go, column_name = "Glutamate.uptake"), Glutamate.uptake))
phagocytosis_glutamate.uptake[is.na(phagocytosis_glutamate.uptake)] <- ""
write.csv(phagocytosis_glutamate.uptake, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/phagocytosis_glutamate.uptake_genes.csv", row.names=FALSE)
```

## Fig 2 VCP vs CTRL

```{r}
# pcaData <- plotPCA(vsd, intgroup=c("group"), returnData=TRUE) #  
# percentVar <- round(100 * attr(pcaData, "percentVar"))
# pca <- ggplot(pcaData, aes(PC1, PC2, color=group)) + geom_text_repel(aes(label=colnames(vsd)), size = 3) +  
#   scale_colour_manual( values = get_palette("npg", 10)) +  theme_bw() +  
#   theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
#   xlab(paste0("PC1: ",percentVar[1],"% variance")) +  ylab(paste0("PC2: ",percentVar[2],"% variance")) + coord_fixed()
# pca

# Volcano
interesting.gene.labels <- patani.res %>% filter(padj < 0.05, gene_name %in% c(go.pathways.msigdb$GO_PHAGOCYTOSIS, glutamate.uptake.go, all.reactivity.go, "ASPN", "OGN", "ADAMTS8", "GJA5", "CALB2", "GABRA2", "GABRB2", "FAM107A", "SLC3A2", "GAD2")) %>% pull(gene_name)
immune.up.genes <- patani.res %>% filter(padj < 0.05, log2FoldChange > 1, gene_name %in% all.reactivity.go, !gene_name %in% c("ANGPT1", "PID1", "TXNIP")) %>%  pull(gene_name)
phagocytosis.down.genes <- patani.res %>% filter(padj < 0.05, log2FoldChange < 1, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  pull(gene_name)
glutamate.uptake.down.genes <- patani.res %>% filter(padj < 0.05, log2FoldChange < 1, gene_name %in% glutamate.uptake.go) %>%  pull(gene_name)
interesting.gene.labels.filt <- c(phagocytosis.down.genes, glutamate.uptake.down.genes, immune.up.genes, "ASPN", "GJA5", "CALB2", "GABRA2", "GABRB2", "SLC3A2") 

patani.volcano <- volcano_plot_deseq(data = patani.res, labels_list = interesting.gene.labels.filt) + xlab(expression(Log[2]~fold~change~VCP:CTRL)) + 
  xlim(-10,10) + ylim(0,8) +
  geom_segment(aes(x = 3, xend = 9, y= 7.5, yend= 7.5), arrow=arrow(length=unit(0.3,"cm")), color = "firebrick2" ) +
  geom_segment(aes(x = -3, xend = -9, y= 7.5, yend= 7.5),arrow=arrow(length=unit(0.3,"cm")), color = "dodgerblue3") +
  annotate("text", x = 6, y = 8, label = "Up in VCP", color = "firebrick2") + annotate("text", x = -6, y = 8, label = "Down in VCP", color = "dodgerblue3")
patani.volcano

# Differential gene expression VCP vs CTRL
patani.dge_genes <- patani.res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
patani.dge_genes <- gost(patani.dge_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
patani.dge_gprofiles_filt <- patani.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ " Down in VCP ", query == "query_2" ~ " Up in VCP "), term_name = as.factor(term_name)) 
patani.dge_gprofiles_down <- patani.dge_gprofiles_filt %>% filter(query == " Down in VCP ")
patani.dge_gprofiles_up <- patani.dge_gprofiles_filt %>% filter(query == " Up in VCP ")

paste('"',unique(patani.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"Type I diabetes mellitus",
"cell periphery",
"brain development",
"GABA receptor Signaling",
# "integral component of plasma membrane",
# "plasma membrane",
"intrinsic component of membrane",
"plasma membrane region")
patani.dge_gprofiles_filt_down <- patani.dge_gprofiles_filt %>% filter(query == " Down in VCP ")  %>% filter(term_name %in% down_list)

paste('"',unique(patani.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"gap junction",
"enzyme linked receptor protein signaling pathway",
"extracellular matrix structural constituent conferring compression resistance")
# "blood circulation",
# "circulatory system process",
# "animal organ development"
patani.dge_gprofiles_filt_up <- patani.dge_gprofiles_filt %>% filter(query == " Up in VCP ")  %>% filter(term_name %in% up_list)
patani.dge_gprofiles_filt_combined <- bind_rows(patani.dge_gprofiles_filt_down, patani.dge_gprofiles_filt_up) %>% 
  mutate(query = factor(query, levels = c(" Up in VCP ", " Down in VCP ")))
patani.dge_gprofiles_filt_combined$term_name <- gsub("extracellular matrix structural constituent conferring compression resistance", "extracellular matrix", patani.dge_gprofiles_filt_combined$term_name)
patani.dge_gprofiles_filt_combined$term_name <- gsub("enzyme linked receptor protein signaling pathway", "enzyme linked receptor signaling", patani.dge_gprofiles_filt_combined$term_name)
patani.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", patani.dge_gprofiles_filt_combined$term_name)
patani.go_curated <- curated_profiles(gprofiles = patani.dge_gprofiles_filt_combined) + theme(legend.position = "none")
patani.go_curated

# which genes are driving the terms
patani.res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
patani.res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
patani.res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
patani.res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)

# GSEA violin plots: Phagocytosis, Glutamate uptke
patani.res_phagocytosis <- filter(patani.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(gene_set = "phagocytosis")
patani.res_glutamate_uptake <- filter(patani.res, gene_name %in% glutamate.uptake.go) %>% mutate(gene_set = "glutamate uptake")
# patani.res_ecm <- filter(patani.res, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX) %>% mutate(gene_set = "extracellular matrix")
# patani.res_synapse <- filter(patani.res, gene_name %in% go.pathways.msigdb$GO_SYNAPTIC_SIGNALING) %>% mutate(gene_set = "synapse signaling")
# patani.res_antixidant <- filter(patani.res, gene_name %in% go.pathways.msigdb$GO_AUTOPHAGOSOME) %>% mutate(gene_set = "antioxidant")
# patani.res_transmitter <- filter(patani.res, gene_name %in% go.pathways.msigdb$GO_NEUROTRANSMITTER_REUPTAKE) %>% mutate(gene_set = "transmitter uptake")
patani.res_gene_sets <- bind_rows(patani.res_phagocytosis, patani.res_glutamate_uptake) #, patani.res_immune, patani.res_nkimmunity, patani.res_adhesion, patani.res_apmhc1)

# without gene labels
gene_sets_sina_plot <- sinaplot.one.sample(data =  patani.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log[2]~Fold~Change~VCP:CTRL )) + xlab("")  + 
  theme(axis.text=element_text(size=9)) + #, axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "VCP Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "VCP Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2)) + labs(subtitle = "GSEA RNA seq")
  # geom_text_repel(data = filter(patani.res_gene_sets, padj < 0.05), aes(x = gene_set, y = log2FoldChange, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0, size=2.7, max.overlaps = Inf)
gene_sets_sina_plot

#   group            .y.   group1 group2         n statistic           p p.signif y.position
# * <chr>            <chr> <chr>  <chr>      <int>     <dbl>       <dbl> <chr>         <dbl>
# 1 glutamate uptake lfc   1      null model   404     29069 0.000000938 ****              2
# 2 phagocytosis     lfc   1      null model   364     15971 0.161       ns                2


go_immune_processes <- c(
gprofiler_database$`antigen processing and presentation of endogenous peptide antigen`,
# gprofiler_database$`regulation of protein sumoylation`,
# gprofiler_database$`negative regulation of protein acetylation`,
# gprofiler_database$`positive regulation of substrate adhesion-dependent cell spreading`,
gprofiler_database$`natural killer cell mediated cytotoxicity`,
# gprofiler_database$`positive regulation of ubiquitin-protein transferase activity`,
gprofiler_database$`natural killer cell mediated immunity`)
# gprofiler_database$`astrocyte activation involved in immune response`,
# gprofiler_database$`astrocyte activation`)
# gprofiler_database$`astrocyte projection`,
# gprofiler_database$`astrocyte differentiation`)

go.pathways.msigdb$GO_SYNAPTIC_SIGNALING

#GSEA of all gene sets
geneList <- res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(geneList) <- res %>% drop_na(log2FoldChange) %>% pull(ensID) %>% as.character()
geneList = sort(geneList, decreasing = TRUE)
ego3 <- gseGO(geneList     = geneList,
              OrgDb        = org.Hs.eg.db,
              ont          = "ALL",
              keyType      = "ENSEMBL",
              minGSSize    = 10,
              maxGSSize    = 5000,
              pvalueCutoff = 0.2)
ego3
ego3@result %>% arrange(-NES) %>% dplyr::select(-core_enrichment) %>% head(20)
ridgeplot(ego3)

# GSEA phagocytosis
geneList <- patani.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(geneList) <- patani.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
geneList = sort(geneList, decreasing = TRUE)
phagocytosis_genes.list <- list("phagocytosis" = patani.res$gene_name[patani.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = geneList)
phagocytosis_fgseaRes
# pathway      pval      padj    log2err        ES     NES size                         leadingEdge
# 1: phagocytosis 0.6338028 0.6338028 0.09754492 0.2385496 0.94727  240 SIRPB1,LBP,VAV1,NCF4,FPR2,PPARG,...

# GSEA glutamate uptake
glutamate.uptake_genes.list <- list("glutamate.uptake" = patani.res$gene_name[patani.res$gene_name %in% glutamate.uptake.go])
glutamate.uptake_fgseaRes <- fgsea(pathways = glutamate.uptake_genes.list, stats = geneList)
glutamate.uptake_fgseaRes
#  pathway       pval       padj   log2err         ES       NES size                              leadingEdge
# 1: glutamate.uptake 0.02476068 0.02476068 0.3524879 -0.3276907 -1.257345  401 KCNA1,PTPRT,GAD2,ADRA1A,KCND2,CACNG5,...

# IR in glutamate uptake or phagocytosis genes
# import IR results

ac_who_vcp_vs_ctrl.IRFinder <- cleanIRFinder(IRFinder = read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/ac_who_vcp-vs-ctrl.txt", sep = '\t',header = TRUE))
ac_who_ctrl.IRFinder <- cleanIRFinderQuant(read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_who_ctrl/IRFinder-IR-dir.txt", sep = '\t',header = TRUE))
ac_who_vcp.IRFinder <- cleanIRFinderQuant(read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/splicing/IRFinder/pooled_ac_who_vcp/IRFinder-IR-dir.txt", sep = '\t',header = TRUE))
ac_who_vcp_vs_ctrl.absoluteIR <- cleanIRFinderQuantCompared(mutant = ac_who_vcp.IRFinder, control = ac_who_ctrl.IRFinder, differentialIR = ac_who_vcp_vs_ctrl.IRFinder)


ac_who_vcp_vs_ctrl.IRFinder %>% filter(gene_name %in% glutamate.uptake.go, p0.05_reliable == "significant") %>% select(gene_name, Intron.GeneName.GeneID, IRratio.diff, p0.05_reliable_IRdirection)
ac_who_vcp_vs_ctrl.IRFinder %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS, p0.05_reliable == "significant") %>% select(gene_name, Intron.GeneName.GeneID, IRratio.diff, p0.05_reliable_IRdirection)
```

### Proteomics

```{r}
# nuc_mass_spec.res<- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/nuclear_mass_spec_results.RDS")
# cyt_mass_spec.res <- readRDS("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/cytoplasmic_mass_spec_results.RDS")
# mass_spec.res <- select(nuc_mass_spec.res, gene_name = name, ID, nuc.pval = vcp_vs_ctrl_p.val, nuc.padj = vcp_vs_ctrl_p.adj, nuc.lfc = vcp_vs_ctrl_ratio, nuc.sig = vcp_vs_ctrl_significant) %>% 
#   full_join(select(cyt_mass_spec.res, gene_name = name, ID, cyt.pval = vcp_vs_ctrl_p.val, cyt.padj = vcp_vs_ctrl_p.adj, cyt.lfc = vcp_vs_ctrl_ratio, cyt.sig = vcp_vs_ctrl_significant), by = c("gene_name", "ID"))
library(tidyverse)
library(DEP)
mass_spec_ac <- read.table("/camp/home/ziffo/home/projects/astrocyte-meta-analysis/expression/mass-spectrometry/ac_vcp_fractions_massspec_protein_groups.txt", sep = '\t',header = TRUE)
mass_spec_ac$Gene.names %>% duplicated() %>% any()
mass_spec_ac %>% group_by(Gene.names) %>% dplyr::summarize(frequency = dplyr::n()) %>%  arrange(desc(frequency)) %>% filter(frequency > 1) # for genes without an annotated genename use the Uniprot ID
mass_spec_ac_all <- make_unique(mass_spec_ac, "Gene.names", "Protein.IDs", delim = ";")
mass_spec_ac_all$name %>% duplicated() %>% any()

# Generate a SummarizedExperiment object using an experimental design
LFQ_columns <- grep("LFQ.", colnames(mass_spec_ac_all)) # get LFQ column numbers
experimental_design <- data.frame("label" = c("C1_01","C1_02","C1_03","C5_01","C5_02","C5_03","CCbie_01","CCbie_02","CCbie_03","CGlia_01","CGlia_02","CGlia_03",
                                   "N1_01","N1_02","N1_03","N5_01","N5_02","N5_03","NCbie_01","NCbie_02","NCbie_03", "NGlia_01","NGlia_02","NGlia_03"),
                       "condition" = c("ctrl","ctrl","ctrl","ctrl","ctrl","ctrl","vcp","vcp","vcp","vcp","vcp","vcp",
                                       "ctrl","ctrl","ctrl","ctrl","ctrl","ctrl","vcp","vcp","vcp", "vcp","vcp","vcp"),
                       "replicate"= c("1","2","3","4","5","6","1","2","3","4","5","6",
                                      "7","8","9","10","11","12","7","8","9", "10","11","12"))
data_se <- make_se(mass_spec_ac_all, LFQ_columns, experimental_design) # create summarised experiment - this log2 transforms assay data
plot_frequency(data_se) # number of proteins quantified in each replicate is very different - missing values need imputing

# Filter out proteins with missing values across samples
# data_filt <- filter_missval(data_se, thr = 0) # Filter proteins identified in all replicates of at least one condition - too stringent - left with 56 / 1160 proteins
# data_filt <- filter_missval(data_se, thr = 1) # # Less stringent filtering - Filter proteins identified in 1/6 replicates of at least one condition 907 / 1160
# data_filt <- filter_missval(data_se, thr = 11) # # Less stringent filtering - Filter proteins identified in 1/6 replicates of at least one condition 1019 / 1160
data_filt <- filter_missval(data_se, thr = 12) # # Less stringent filtering - Filter proteins identified in 1/6 replicates of at least one condition 1160 / 1160 - ie no filtering
plot_numbers(data_filt) #  barplot of the number of identified proteins per samples
plot_coverage(data_se) # barplot of the protein identification overlap between samples

# Normalisation
data_norm <- normalize_vsn(data_filt) # normalise with VST
plot_normalization(data_filt, data_norm) # # Visualize normalization by boxplots for all samples before and after normalization
plot_missval(data_filt) # explore proteins with missing values with heatmap - bias of missing proteins in nuclear samples.
plot_detect(data_filt) # missing proteins have lower intense proteins and slightly higher densities

# Impute missing values
data_imp <- impute(data_norm, fun = "MinProb", q = 0.01) # Impute missing data using random draws from a Gaussian distribution centered around a minimal value (for MNAR)
# data_imp_man <- impute(data_norm, fun = "man", shift = 1.8, scale = 0.3)# Impute missing data using random draws from a manually defined left-shifted Gaussian distribution (for MNAR)
# data_imp_knn <- impute(data_norm, fun = "knn", rowmax = 0.9)# Impute missing data using the k-nearest neighbour approach (for MAR)
plot_imputation(data_norm, data_imp) # check effect of imputation on distributions

# Differential expression analysis
data_diff <- test_diff(data_imp, type = "control", control = "ctrl") # Differential enrichment analysis - linear models with empirical Bayes, utilises limma. specify the control sample. 
dep <- add_rejections(data_diff, alpha = 0.05)

# PCA
plot_pca(dep, x = 1, y = 2, n = 100, point_size = 4) 
# correlation matrix
plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Reds") 
# heatmap of all significant proteins
plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 6, col_limit = 4, show_row_names = FALSE,indicate = c("condition", "replicate"))
# Volcano
plot_volcano(dep, contrast = "cyt_R155C_vs_cyt_ctrl1", label_size = 2, add_names = TRUE, adjusted = FALSE)

# Generate a results table
data_results <- get_results(dep)
mass_spec.res <- select(data_results, gene_name = name, ID, pval = vcp_vs_ctrl_p.val, padj = vcp_vs_ctrl_p.adj, lfc = vcp_vs_ctrl_ratio)
saveRDS(mass_spec.res, "/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-a1-als-meta/expression/mass-spec/mass_spec.res_vcp_vs_ctrl.rds")
mass_spec.res <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-a1-als-meta/expression/mass-spec/mass_spec.res_vcp_vs_ctrl.rds")

# Volcano
mass_spec.volcano <- ggplot(data = mass_spec.res, aes(x = lfc, y =  -log10(pval))) +  geom_point(size = 0.5) + #xlim(1,5) + ylim(-3,3) + scale_colour_manual( values = c(unchanged = "darkgray", downregulated = "firebrick2", upregulated = "dodgerblue3") ) +  
        theme_bw() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "none") + ylab(expression( -log[10]~P~value )) +  xlab(expression( mass~spec~log~Fold~Change~VCP:CTRL )) +
      geom_text_repel(data = filter(mass_spec.res, pval < 0.05), aes(x = lfc, y = -log10(pval), label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size=2.6, max.overlaps = Inf) +
      geom_hline(yintercept = -log10(0.05), linetype = 2) + geom_vline(xintercept =0, linetype = 2)
mass_spec.volcano

# GSEA
mass_spec.res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS, lfc < 0) # 26 / 31
mass_spec.res %>% filter(gene_name %in% glutamate.uptake.go, lfc < 0) # 26 / 44

mass_spec.res_phagocytosis <-filter(mass_spec.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  mutate(gene_set = "phagocytosis")
mass_spec.res_glutamate_uptake <- filter(mass_spec.res, gene_name %in% glutamate.uptake.go) %>% mutate(gene_set = "glutamate uptake")
mass_spec.res_gene_sets <- bind_rows(mass_spec.res_phagocytosis, mass_spec.res_glutamate_uptake) 

mass_spec.gene_sets_sina_plot <- sinaplot.one.sample(data =  mass_spec.res_gene_sets, group = "gene_set", continuous = "lfc", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log~Fold~Change~VCP:CTRL )) + xlab("")  + theme(axis.text=element_text(size=9)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "VCP Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "VCP Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2))  + 
      # geom_text_repel(data = filter(mass_spec.res_gene_sets, gene_name %in% c(go.pathways.msigdb$GO_PHAGOCYTOSIS, glutamate.uptake.go)), aes(x = gene_set, y = lfc, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size=2.6, max.overlaps = Inf) 
  labs(subtitle = "GSEA mass spec")
mass_spec.gene_sets_sina_plot
#  group            .y.   group1 group2         n statistic         p p.signif y.position
# * <chr>            <chr> <chr>  <chr>      <int>     <dbl>     <dbl> <chr>         <dbl>
# 1 glutamate uptake lfc   1      null model    44       402 0.28      ns                2
# 2 phagocytosis     lfc   1      null model    31        59 0.0000778 ****              2

# Volcano
interesting.gene.labels <-  filter(mass_spec.res, gene_name %in% c(go.pathways.msigdb$GO_PHAGOCYTOSIS, glutamate.uptake.go)) %>% pull(gene_name)
# immune.up.genes <- mass_spec.res %>% filter(padj < 0.05, log2FoldChange > 1, gene_name %in% all.reactivity.go, !gene_name %in% c("ANGPT1", "PID1", "TXNIP")) %>%  pull(gene_name)
# phagocytosis.down.genes <- mass_spec.res %>% filter(padj < 0.05, log2FoldChange < 1, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  pull(gene_name)
# glutamate.uptake.down.genes <- mass_spec.res %>% filter(padj < 0.05, log2FoldChange < 1, gene_name %in% glutamate.uptake.go) %>%  pull(gene_name)
# interesting.gene.labels.filt <- c(phagocytosis.down.genes, glutamate.uptake.down.genes, immune.up.genes, "ASPN", "GJA5", "CALB2", "GABRA2", "GABRB2", "SLC3A2") 
   
mass_spec.volcano <- volcano_plot_deseq(data = mass_spec.res, labels_list = interesting.gene.labels) + xlab(expression(Log[2]~fold~change~VCP:CTRL)) + 
  xlim(-10,10) + ylim(0,8) +
  geom_segment(aes(x = 3, xend = 9, y= 7.5, yend= 7.5), arrow=arrow(length=unit(0.3,"cm")), color = "dodgerblue3") +
  geom_segment(aes(x = -3, xend = -9, y= 7.5, yend= 7.5),arrow=arrow(length=unit(0.3,"cm")), color = "firebrick2") +
  annotate("text", x = 6, y = 8, label = "Up in VCP", color = "dodgerblue3") + annotate("text", x = -6, y = 8, label = "Down in VCP", color = "firebrick2")
mass_spec.volcano

```

### Merge

```{r}
fig2.merged <- ggarrange(
  ggarrange(patani.volcano, VCP_vs_ctrl.go_curated, ncol = 2, labels = c("A", "B"), widths = c(1,0.8)),
  ggarrange(gene_sets_sina_plot, mass_spec.gene_sets_sina_plot, ncol = 2, labels = c("C", "D"), widths = c(1,1)),
  nrow = 2, heights = c(1,0.8))
fig2.merged
ggsave(fig2.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/fig2.merged.png", height = 7, width = 7)
```

## Fig 3 A1 vs VCP

### Venn overlap DEGs

```{r}
barbar.res.dge.up <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
barbar.res.dge.down <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
patani.res.dge.up <- patani.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
patani.res.dge.down <- patani.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

up_genes <- list("A1" = barbar.res.dge.up, "VCP" = patani.res.dge.up)
down_genes <- list("A1" = barbar.res.dge.down, "VCP" = patani.res.dge.down)

patani_barbar_venn_up <- ggvenn(up_genes, fill_color = c("firebrick2","dodgerblue3"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5))
patani_barbar_venn_up
patani_barbar_venn_down <- ggvenn(down_genes, fill_color = c("firebrick2","dodgerblue3"), show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4)+ labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5))
patani_barbar_venn_down
ggsave(patani_barbar_venn_up, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/patani_barbar_venn_up.png", height = 3, width = 3)
ggsave(patani_barbar_venn_down, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/patani_barbar_venn_down.png", height = 3, width = 3)

# # 4 way venn 
# library(VennDiagram)
# display_venn <- function(x, ...){
#   grid.newpage()
#   venn_object <- venn.diagram(x, filename = NULL, ...)
#   grid.draw(venn_object)
# }
# x <- list(
#   "A1 up" = barbar.res.dge.up, 
#   "A1 down" = barbar.res.dge.down, 
#   "VCP up" = patani.res.dge.up,
#   "VCP down" = patani.res.dge.down)
# ggvenn(
#   x, 
#   fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
#   stroke_size = 0.5, set_name_size = 4
#   )
# barbar_patani.venn_overlap <- display_venn(
#         x,
#         lwd = 2, lty = 'blank',
#         fill = c("dodgerblue1", "dodgerblue3", "firebrick1", "firebrick3"),
#         # cex = .9,
#         # fontface = "italic",
#         # Set names
#         cat.cex = 1,
#         cat.fontface = "bold",
#         cat.default.pos = "outer",
#         cat.dist = c(0.21, 0.21, 0.11, 0.11)
# )
# venn.diagram(
#         x,
#         lwd = 2, lty = 'blank',
#         fill = c("dodgerblue1", "dodgerblue3", "firebrick1", "firebrick3"),
#         # cex = .9,
#         # fontface = "italic",
#         # Set names
#         cat.cex = 1,
#         cat.fontface = "bold",
#         cat.default.pos = "outer",
#         cat.dist = c(0.21, 0.21, 0.11, 0.11),
#         filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar_patani.venn_overlap.png"
# )


# # PCA
# pcaData <- plotPCA(vsd, intgroup=c("group"), returnData=TRUE) #  
# percentVar <- round(100 * attr(pcaData, "percentVar"))
# pca <- ggplot(pcaData, aes(PC1, PC2, color=group)) + geom_text_repel(aes(label=colnames(vsd)), size = 3) +  
#   scale_colour_manual( values = get_palette("npg", 11)) +  theme_bw() +  
#   theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "right", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + geom_point(size=2) + 
#   xlab(paste0("PC1: ",percentVar[1],"% variance")) +  ylab(paste0("PC2: ",percentVar[2],"% variance")) + coord_fixed()
# pca
# ggsave(pca, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/pca_vcp_a1.png", height = 9, width = 9)
# 
# 
# sampleDists <- dist(t(assay(vsd)))
# hc <- hclust(sampleDists, method = "ward.D2")
# a <- c("black", "black", "black","firebrick2", "firebrick2", "black", "black","black", "firebrick2", "firebrick2", "firebrick2")
# dendrogram <- ggdendrogram(hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
# dendrogram
# ggsave(dendrogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/dendrogram_vcp_a1.png", height = 7, width = 4)
```

### Scatter correlation 

```{r}
patani.dge <- patani.res %>% dplyr::select(gene_name, ensID, log2FC.VCP = log2FoldChange, FDR.VCP = padj)
barbar.dge <- barbar.res %>% dplyr::select(gene_name, ensID, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
patani_barbar.dge <- full_join(patani.dge, barbar.dge, by = c("gene_name", "ensID"))

patani_barbar.dge <- patani_barbar.dge %>% mutate(overlap = case_when(FDR.VCP < 0.05 & FDR.A1 < 0.05 ~ "overlapping", # down in VCP
                                                        FDR.VCP < 0.05 ~ "VCP specific",
                                                        FDR.A1 < 0.05 ~ "A1 specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.VCP < 0.05 & log2FC.VCP > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in VCP
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up",
                                                                  FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "VCP specific up", "VCP specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "VCP specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(patani_barbar.dge$overlap)
table(patani_barbar.dge$overlap_direction)
patani_barbar.dge

cor.test(x = patani_barbar.dge$log2FC.VCP, y = patani_barbar.dge$log2FC.A1) # 0.1088784 
t.test( x = patani_barbar.dge$log2FC.VCP, y = patani_barbar.dge$log2FC.A1, paired = TRUE )

A1_VCP_scatter <- ggscatter(patani_barbar.dge, x = "log2FC.VCP", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~VCP:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = patani_barbar.dge, aes(x = log2FC.VCP, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(patani_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go), aes(x = log2FC.VCP, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +  
  annotate(geom="text", x=6, y=10, label="Up in both VCP & A1", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-6, y=-10, label="Down in both VCP & A1", color="dodgerblue3", size = 3.2) + labs(subtitle = "Differential gene expression")
A1_VCP_scatter
ggsave(A1_VCP_scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_VCP_scatter.png", height = 8, width = 8)


### Table S1. ALS & A1 DESeq2 results
write.csv(patani_barbar.dge, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/TableS1.A1.VCP.overlap.differentially.expressed.genes.csv")


# patani_barbar.dge_overlap.up <- patani_barbar.dge %>% filter(overlap_direction == "overlapping up") %>% dplyr::select(gene_name, Log2FC.A1 = log2FoldChange_a1, FDR.A1 = padj_a1, log2FC.VCP = log2FoldChange_vcp, FDR.VCP = padj_vcp, overlap_direction)
# patani_barbar.dge_overlap.down <- patani_barbar.dge %>% filter(overlap_direction == "overlapping down") %>% dplyr::select(gene_name, Log2FC.A1 = log2FoldChange_a1, FDR.A1 = padj_a1, log2FC.VCP = log2FoldChange_vcp, FDR.VCP = padj_vcp)
# patani_barbar.dge_overlap <- bind_rows(patani_barbar.dge_overlap.up, patani_barbar.dge_overlap.down)
# 
# patani_barbar.dge_overlap <- patani_barbar.dge %>% filter(overlap == "overlapping") %>% dplyr::select(gene_name, Log2FC.A1 = log2FoldChange_a1, FDR.A1 = padj_a1, log2FC.VCP = log2FoldChange_vcp, FDR.VCP = padj_vcp, overlap_direction) %>% arrange(-overlap_direction)
# patani_barbar.dge_overlap
# readr::write_csv(select(dge_res_ac_who_vcp_vs_ctrl.FDR0.05, gene_name, everything()), path=paste("/camp/home/ziffo/home/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/ac_who_vcp_vs_ctrl_deseq2_results.FDR0.05.csv", sep = ""))


# Overlap tested using Fisher's exact test (alternative=greater)
VCP.DGE.genes <- patani.res %>% filter(padj < 0.05) %>% pull(gene_name)
A1.DGE.genes <- barbar.res %>% filter(padj < 0.05) %>% pull(gene_name)
genome.size <- patani.res %>% distinct(gene_name) %>% nrow
go.obj <- newGeneOverlap(VCP.DGE.genes,
                         A1.DGE.genes,
                         genome.size=genome.size)
go.obj <- testGeneOverlap(go.obj)
go.obj # overlapping p-value indicates whether the overlap is significant
print(go.obj)
# GeneOverlap object:
# listA size=199
# listB size=9147
# Intersection size=69
# Overlapping p-value=5.8e-12
# Jaccard Index=0.0

VCP.DGE.up.genes <- patani.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
A1.DGE.up.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
genome.size <- patani.res %>% distinct(gene_name) %>% nrow
go.obj <- newGeneOverlap(VCP.DGE.up.genes,
                         A1.DGE.up.genes,
                         genome.size=genome.size)
go.obj <- testGeneOverlap(go.obj)
go.obj # overlapping p-value indicates whether the overlap is significant
print(go.obj)
# GeneOverlap object:
# listA size=91
# listB size=4686
# Intersection size=22
# Overlapping p-value=1.2e-06
# Jaccard Index=0.0


VCP.DGE.down.genes <- patani.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
A1.DGE.down.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
genome.size <- patani.res %>% distinct(gene_name) %>% nrow
go.obj <- newGeneOverlap(VCP.DGE.down.genes,
                         A1.DGE.down.genes,
                         genome.size=genome.size)
go.obj <- testGeneOverlap(go.obj)
go.obj # overlapping p-value indicates whether the overlap is significant
print(go.obj)
# GeneOverlap object:
# listA size=108
# listB size=4461
# Intersection size=29
# Overlapping p-value=6.5e-10
# Jaccard Index=0.0


# All DEG irrespective of direction
venn_a1_VCP <- as.ggplot(
  ~draw.pairwise.venn(
    area1 = nrow( filter(patani_barbar.dge, overlap %in% c("overlapping", "VCP specific") ) ),
    area2 = nrow( filter(patani_barbar.dge, overlap %in% c("overlapping", "a1 specific") ) ),
    cross.area = nrow(filter(patani_barbar.dge, overlap == "overlapping")),
    category = c("VCP", "a1"),  fill = c("red","blue"),
    alpha = 0.5, cat.fontfamily = "sans",ext.pos = 50)) +
    # cat.cex = c(1,1),# cat.pos = 360, cat.dist = c(0.1,0.1))) +
  labs(title = "A1 stimulated & VCP: FDR < 0.05") + theme(plot.title = element_text(hjust = 0.5))
venn_a1_VCP
ggsave(venn_a1_VCP, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/venn_a1_VCP_allFDR0.05.png", height = 4, width = 4)

# Upregulated DEG
venn_a1_VCP.up <- as.ggplot(
  ~draw.pairwise.venn(
    area1 = nrow( filter(patani_barbar.dge, overlap_direction %in% c("overlapping up", "VCP specific up" ) ) ),
    area2 = nrow( filter(patani_barbar.dge, overlap_direction %in% c("overlapping up", "a1 specific up" ) ) ),
    cross.area = nrow(filter(patani_barbar.dge, overlap_direction == "overlapping up")),
    category = c("VCP", "a1"),  fill = c("red","blue"),
    alpha = 0.5, cat.fontfamily = "sans",ext.pos = 50)) +
    # cat.cex = c(1,1),# cat.pos = 360, cat.dist = c(0.1,0.1))) +
  labs(title = "Upregulated A1 & VCP: FDR < 0.05") + theme(plot.title = element_text(hjust = 0.5))
venn_a1_VCP.up
ggsave(venn_a1_VCP.up, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/venn_a1_VCP_upregulatedFDR0.05.png", height = 4, width = 4)

# Downregulated DEG
venn_a1_VCP.down <- as.ggplot(
  ~draw.pairwise.venn(
    area1 = nrow( filter(patani_barbar.dge, overlap_direction %in% c("overlapping down", "VCP specific down" ) ) ),
    area2 = nrow( filter(patani_barbar.dge, overlap_direction %in% c("overlapping down", "a1 specific down" ) ) ),
    cross.area = nrow(filter(patani_barbar.dge, overlap_direction == "overlapping down")),
    category = c("VCP", "a1"),  fill = c("red","blue"),
    alpha = 0.5, cat.fontfamily = "sans",ext.pos = 50)) +
    # cat.cex = c(1,1),# cat.pos = 360, cat.dist = c(0.1,0.1))) +
  labs(title = "Downregulated A1 & VCP: FDR < 0.05") + theme(plot.title = element_text(hjust = 0.5))
venn_a1_VCP.down
ggsave(venn_a1_VCP.down, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/venn_a1_VCP_downregulatedFDR0.05.png", height = 4, width = 4)


# Overlaping genes involved with reactivity
patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down") %>% filter(gene_name %in% all.reactivity.go) 
patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down") %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) 
patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down") %>% filter(gene_name %in% glutamate.uptake.go) 
patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down") %>% filter(gene_name %in% all.reactivity.go) 


go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(gene_set = "phagocytosis", dataset = "SOD1")
tyzack.res_glutamate_uptake <- filter(tyzack.res, gene_name %in% glutamate.uptake.go)


go.pathways.msigdb[go.pathways.msigdb %in% "SYTL4"]
                                      
                                      
```




### GO of overlap

```{r}
patani.barbar.dge_overlap_genes <- patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
patani.barbar.dge_overlap_genes <- gost(patani.barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
patani.barbar.dge_overlap_genes$result
patani.barbar.dge_overlap_genes$result$term_name <- str_trunc(patani.barbar.dge_overlap_genes$result$term_name, 30, "right")
patani.barbar.dge_overlap_gprofiles_filt <- patani.barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name), query = "") 
patani.barbar.dge_overlap_gprofiles_filt

patani.barbar.dge_overlap_gprofiles_filt.go_curated <- curated_profiles(gprofiles = patani.barbar.dge_overlap_gprofiles_filt, colours = "dodgerblue3") + labs(subtitle = "Down in both VCP & A1")
patani.barbar.dge_overlap_gprofiles_filt.go_curated
ggsave(patani.barbar.dge_overlap_gprofiles_filt.go_curated, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/patani.barbar.dge_overlap_gprofiles_filt.go_curated.png", height = 5, width = 6)

# # FDR 0.1
# dge_overlap_genes.FDR0.1 <- als_a1.dge %>% filter(overlap_direction.FDR0.1 == "overlapping up" | overlap_direction.FDR0.1 == "overlapping down" ) %>% split(.$overlap_direction.FDR0.1) %>%  map("ensID") # create list of IR up/down & Expression up/down genes
# dge_overlap_genes.FDR0.1 <- gost(dge_overlap_genes.FDR0.1) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
# dge_overlap_genes.FDR0.1$result
# dge_overlap_gprofiles_filt.FDR0.1 <- dge_overlap_genes.FDR0.1$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
# dge_overlap_gprofiles_filt.FDR0.1
# 
# dge_overlap_gprofiles_filt.go_curated.FDR0.1 <- curated_profiles_colours(gprofiles = dge_overlap_gprofiles_filt.FDR0.1) 
# dge_overlap_gprofiles_filt.go_curated.FDR0.1
# ggsave(dge_overlap_gprofiles_filt.go_curated.FDR0.1, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/dge_overlap_gprofiles_filt.go_curated.FDR0.1.png", height = 5, width = 6)
```

### A1 GSEA phag/glut

```{r}
# GSEA violin plots: Phagocytosis, Glutamate uptke
barbar.res_phagocytosis <-filter(barbar.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  mutate(gene_set = "phagocytosis")
barbar.res_glutamate_uptake <- filter(barbar.res, gene_name %in% go.pathways.msigdb$GO_GLUTAMATERGIC_SYNAPSE) %>% mutate(gene_set = "glutamate uptake")
barbar.res_gene_sets <- bind_rows(barbar.res_phagocytosis, barbar.res_glutamate_uptake) 

barbar.gene_sets_sina_plot <- sinaplot.one.sample(data =  barbar.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log[2]~Fold~Change~A1:A0 )) + xlab("")  + theme(axis.text=element_text(size=9)) + 
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "A1 Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "A1 Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2))  + labs(subtitle = "GSEA A1")
barbar.gene_sets_sina_plot
# group         .y.   group1 group2        n statistic         p p.signif y.position
# * <chr>         <chr> <chr>  <chr>     <int>     <dbl>     <dbl> <chr>         <dbl>
# 1 glutamate up? lfc   1      null mod?   356     21427   2.16e-7 ****              2
# 2 phagocytosis  lfc   1      null mod?   364     21644   2.16e-5 ****              2

# GSEA phagocytosis
barbar.geneList <- barbar.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(barbar.geneList) <- barbar.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
barbar.geneList = sort(barbar.geneList, decreasing = TRUE)
barbar.phagocytosis_genes.list <- list("phagocytosis" = barbar.res$gene_name[barbar.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
barbar.phagocytosis_fgseaRes <- fgsea(pathways = barbar.phagocytosis_genes.list, stats = barbar.geneList)
barbar.phagocytosis_fgseaRes
#   pathway         pval         padj   log2err        ES
# 1: phagocytosis 1.164648e-06 1.164648e-06 0.6435518 0.4248308

# GSEA glutamate uptake
barbar.glutamate.uptake_genes.list <- list("glutamate.uptake" = barbar.res$gene_name[barbar.res$gene_name %in% glutamate.uptake.go])
barbar.glutamate.uptake_fgseaRes <- fgsea(pathways = barbar.glutamate.uptake_genes.list, stats = barbar.geneList)
barbar.glutamate.uptake_fgseaRes
#   pathway         pval         padj   log2err         ES
# 1: glutamate.uptake 9.090766e-05 9.090766e-05 0.5384341 -0.3118546
```


### Transcription factor footprint

#### PROGENy

https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md
https://saezlab.github.io/progeny/

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity 

```{r}
##################### VCP Patani ##################### 
# differential expression results as matrix
patani.res.matrix <- patani.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
patani.VCP.PathwayActivity_zscore <- progeny(patani.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(patani.VCP.PathwayActivity_zscore) <- "NES"
patani.VCP.PathwayActivity_zscore_df <- as.data.frame(patani.VCP.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))

### GENERATE P-VALUES FROM PROGENY Z-SCORES
progeny(patani.res.matrix, organism="Human", perm = 10000, z_scores = TRUE) %>%  t()
runProgenyFast(df = patani.res.matrix, weight_matrix = progeny_matrix_mouse_v1, k = 10000, z_score = T)

progeny_pvals <- as.data.frame(apply(patani.VCP.PathwayActivity_zscore_df, 2, pnorm))
progeny_pvals[] <- lapply(progeny_pvals, function(x){ifelse(x > 0.5, 1 - x, x)})
progeny_pvals <- progeny_pvals[c(6,2,9,7,11,5,10,4,8,1,13,14,3,12),]

barbar.res.matrix <- barbar.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
barbar.A1.PathwayActivity_zscore <- progeny(barbar.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t() # enrichment analysis of each pathway activity
colnames(barbar.A1.PathwayActivity_zscore) <- "NES"
barbar.A1.PathwayActivity_zscore_df <- as.data.frame(barbar.A1.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
barbar.patani.PathwayActivity_zscore_df <- mutate(barbar.A1.PathwayActivity_zscore_df, dataset = "A1") %>% bind_rows(mutate(patani.VCP.PathwayActivity_zscore_df, dataset = "VCP"))
barbar.patani.PathwayActivity_zscore_df.meanNES <- barbar.patani.PathwayActivity_zscore_df %>% group_by(Pathway) %>% summarise(mean_NES = mean(NES)) %>% arrange(mean_NES)
barbar.patani.PathwayActivity_zscore_df$Pathway <- factor(barbar.patani.PathwayActivity_zscore_df$Pathway, levels = barbar.patani.PathwayActivity_zscore_df.meanNES$Pathway)
barbar.patani.progeny.pathwayActivity <- ggplot(barbar.patani.PathwayActivity_zscore_df, aes(x=Pathway, y = NES, fill = dataset)) + theme_classic() +
  geom_bar(stat="identity", position=position_dodge()) + theme(legend.position = "top", legend.title = element_blank(), axis.title = element_text(size = 12),
        axis.text.x = element_text(angle = 45, hjust = 1, size = 8), axis.text.y = element_text(size =8)) +
  geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") + scale_fill_manual( values = c("firebrick3", "dodgerblue3")) + xlab(expression(Signalling~Pathway))+ ylab(expression(Normalised~enrichment~score)) + labs(subtitle = "Signalling Pathway activity")
barbar.patani.progeny.pathwayActivity
ggsave(barbar.patani.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar.patani.progeny.pathwayActivity.png", height = 5, width = 7)


# patani.VCP.progeny.pathwayActivity <- ggplot(patani.VCP.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),
#         axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"), axis.text.y = element_text(size =8, face= "bold")) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Pathways") + ylab( expression(Normalised~Enrichment) )  + labs(subtitle = "VCP vs CTRL")
# patani.VCP.progeny.pathwayActivity
# ggsave(patani.VCP.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/patani.VCP.progeny.pathwayActivity.png", height = 6, width = 6)
# 
# barbar.res.matrix <- barbar.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
# barbar.A1.PathwayActivity_zscore <- progeny(barbar.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t() # enrichment analysis of each pathway activity
# colnames(barbar.A1.PathwayActivity_zscore) <- "NES"
# barbar.A1.PathwayActivity_zscore_df <- as.data.frame(barbar.A1.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
# barbar.A1.progeny.pathwayActivity <- ggplot(barbar.A1.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),
#         axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"), axis.text.y = element_text(size =8, face= "bold")) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +   xlab("Pathways") + ylab( expression(Normalised~Enrichment) ) + labs(subtitle = "A1 vs A0")
# barbar.A1.progeny.pathwayActivity
# ggsave(barbar.A1.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar.A1.progeny.pathwayActivity.png", height = 6, width = 6)
# 
# progeny.pathwayActivity <- ggarrange(patani.VCP.progeny.pathwayActivity, barbar.A1.progeny.pathwayActivity, nrow = 1, ncol = 2)
# progeny.pathwayActivity
# ggsave(progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_VCP.progeny.pathwayActivity.png", height = 5, width = 7)


# A1.VCP.progeny <- rename(patani.VCP.PathwayActivity_zscore_df, log2FC.VCP = NES)  %>% left_join(rename(barbar.A1.PathwayActivity_zscore_df, log2FC.A1 = NES), by = "Pathway")
# cor.test(x = A1.VCP.progeny$log2FC.VCP, y = A1.VCP.progeny$log2FC.A1) # 0.1088784 
# t.test( x = A1.VCP.progeny$log2FC.VCP, y = A1.VCP.progeny$log2FC.A1, paired = TRUE )
# progeny_vcp_scatter <- ggscatter(A1.VCP.progeny, x = "log2FC.VCP", y = "log2FC.A1", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
#           cor.coef = TRUE, cor.method = "pearson",
#           xlab = expression( Log[2]~Fold~Change~VCP:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5) +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
#     geom_smooth(data = A1.VCP.progeny, aes(x = log2FC.VCP, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
#     geom_text_repel(data = A1.VCP.progeny, aes(x = log2FC.VCP, y = log2FC.A1, label = Pathway), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) #+
#   # annotate(geom="text", x=2, y=12, label="Up in both VCP & A1", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="Down in both VCP & A1", color="dodgerblue3", size = 3.2)
# progeny_vcp_scatter

# patani.VCP.PathwayActivity_zscore <- as_tibble(patani.VCP.PathwayActivity_zscore, rownames = "Pathway") %>% mutate(condition = "VCP")
# barbar.A1.PathwayActivity_zscore <- as_tibble(barbar.A1.PathwayActivity_zscore, rownames = "Pathway") %>% mutate(condition = "A1")
# PathwayActivity_zscore <- bind_rows(patani.VCP.PathwayActivity_zscore, barbar.A1.PathwayActivity_zscore)
# 
# progeny.pathwayActivity <- ggplot(PathwayActivity_zscore,aes(x = Pathway, y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +   theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12), axis.text.x = element_text(angle = 45, hjust = 1, size =10, face= "bold"), axis.text.y = element_text(size =10, face= "bold")) +
#   facet_wrap(. ~ condition, scales = "free") +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Pathways") + ylab( expression(Normalised~enrichment~score) )
# progeny.pathwayActivity


# normalised counts as matrix
# colnames(patani.vsd) <- colnames(patani.vsd) %>% gsub("cytoplasmic", "ac_cyt", .) %>% gsub("nuclear", "ac_nuc", .) #%>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .) %>% gsub("ctrl5", "ctrl2", .)
# rownames(patani.vsd) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(patani.vsd)]
# patani.vsd.mean <- assay(patani.vsd) %>% as_tibble(rownames = "gene_name") %>% mutate(CTRL = (ctrl1 + ctrl5) / 2, VCP = (cb1e + glia) / 2) %>% dplyr::select(CTRL, VCP) %>% as.matrix()
# rownames(patani.vsd.mean) <- assay(patani.vsd) %>% as_tibble(rownames = "gene_name") %>% mutate(CTRL = (ctrl1 + ctrl5) / 2, VCP = (cb1e + glia) / 2) %>% dplyr::pull(gene_name)
# 
# # compute Progeny scores for every sample
# patani.VCP.PathwayActivity_counts <- progeny(patani.vsd.mean, scale=TRUE, organism="Human", top = 100) # use 100 most responsive genes per pathway
# patani.VCP.Activity_counts <- as.vector(patani.VCP.PathwayActivity_counts)
# 
# # Progeny scores heatmap
# paletteLength <- 100
# myColor <- colorRampPalette(c("darkblue", "whitesmoke","indianred"))(paletteLength)
# patani.VCP.progenyBreaks <- c(seq(min(patani.VCP.Activity_counts), 0, 
#     length.out=ceiling(paletteLength/2) + 1),
#     seq(max(patani.VCP.Activity_counts)/paletteLength, 
#     max(patani.VCP.Activity_counts), 
#     length.out=floor(paletteLength/2)))
# patani.VCP.progeny_hmap <- pheatmap(t(patani.VCP.PathwayActivity_counts),fontsize=14, 
#     fontsize_row = 10, fontsize_col = 10, 
#     color=myColor, #breaks = patani.VCP.progenyBreaks, 
#     angle_col = 45, treeheight_col = 0,  border_color = NA)
# patani.VCP.progeny_hmap
# ggsave(patani.VCP.progeny_hmap, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/patani.VCP.progeny_hmap.png", height = 6, width = 4)
```


#### DoRothEA

https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

First compute TF activity enrihcment from differential expression analysis. 

```{r}
data(dorothea_hs, package = "dorothea")
regulons <- dorothea_hs %>%  dplyr::filter(confidence %in% c("A", "B","C"))

patani.VCP.tf_activities_stat <- dorothea::run_viper(patani.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
patani.VCP.tf_activities_stat_top25 <- patani.VCP.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))

barbar.A1.tf_activities_stat <- dorothea::run_viper(barbar.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
barbar.A1.tf_activities_stat_top25 <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))

patani.VCP.tf_activities_stat.df <- patani.VCP.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.VCP = "log2FoldChange")
barbar.A1.tf_activities_stat.df <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(log2FC.A1 = "log2FoldChange")

A1.VCP.tf_activities <- patani.VCP.tf_activities_stat.df %>% left_join(barbar.A1.tf_activities_stat.df, by = "gene_name")

cor.test(x = A1.VCP.tf_activities$log2FC.VCP, y = A1.VCP.tf_activities$log2FC.A1) # 0.1088784 
t.test( x = A1.VCP.tf_activities$log2FC.VCP, y = A1.VCP.tf_activities$log2FC.A1, paired = TRUE )

A1_vcp_TFactivities.scatter <- ggscatter(A1.VCP.tf_activities, x = "log2FC.VCP", y = "log2FC.A1", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~VCP:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = A1.VCP.tf_activities, aes(x = log2FC.VCP, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(A1.VCP.tf_activities, abs(log2FC.VCP) > 1, abs(log2FC.A1) > 2), aes(x = log2FC.VCP, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=2, y=12, label="Up in both VCP & A1", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="Down in both VCP & A1", color="dodgerblue3", size = 3.2) + labs(subtitle = "Transcription Factor activity")
A1_vcp_TFactivities.scatter
ggsave(A1_vcp_TFactivities.scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_vcp_TFactivities.scatter.png", height = 8, width = 8)

# patani.VCP.dorotheaActivity <- ggplot(patani.VCP.tf_activities_stat_top25,aes(x = reorder(gene_name, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred",  mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"),axis.text.y = element_text(size =8, face= "bold")) +
#     geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Transcription Factors") + ylab( expression(Normalised~enrichment) )  + labs(subtitle = "VCP vs CTRL")
# patani.VCP.dorotheaActivity
# ggsave(patani.VCP.dorotheaActivity, filename = "/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/patani.VCP.dorotheaActivity.png", height = 6, width = 6)
# 
# barbar.A1.dorotheaActivity <- ggplot(barbar.A1.tf_activities_stat_top25,aes(x = reorder(gene_name, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred",  mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"),axis.text.y = element_text(size =8, face= "bold")) +
#     geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Transcription Factors") + ylab( expression(Normalised~enrichment) )  + labs(subtitle = "A1 vs A0")
# barbar.A1.dorotheaActivity
# ggsave(barbar.A1.dorotheaActivity, filename = "/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/barbar.A1.dorotheaActivity.png", height = 6, width = 6)
# 
# 
# dorothea.TFactivity <- ggarrange(patani.VCP.dorotheaActivity, barbar.A1.dorotheaActivity, nrow = 1, ncol = 2)
# dorothea.TFactivity
# ggsave(dorothea.TFactivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_VCP.dorothea.TFactivity.png", height = 5, width = 9)
```

### Mass spec

```{r}
mass_spec.res <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-a1-als-meta/expression/mass-spec/mass_spec.res_vcp_vs_ctrl.rds")

mass_spec.dge <- mass_spec.res %>% dplyr::select(gene_name, log2FC.VCP = lfc, FDR.VCP = pval)
barbar.dge <- barbar.res %>% dplyr::select(gene_name, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
mass_spec_barbar.dge <- left_join(mass_spec.dge, barbar.dge, by = "gene_name")

mass_spec_barbar.dge <- mass_spec_barbar.dge %>% mutate(overlap = case_when(FDR.VCP < 0.05 & FDR.A1 < 0.05 ~ "overlapping", # down in VCP
                                                        FDR.VCP < 0.05 ~ "VCP specific",
                                                        FDR.A1 < 0.05 ~ "A1 specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.VCP < 0.05 & log2FC.VCP > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in VCP
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up",
                                                                  FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "VCP specific up", "VCP specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "VCP specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(mass_spec_barbar.dge$overlap)
table(mass_spec_barbar.dge$overlap_direction)
mass_spec_barbar.dge

cor.test(x = mass_spec_barbar.dge$log2FC.VCP, y = mass_spec_barbar.dge$log2FC.A1) # 0.1088784 
t.test( x = mass_spec_barbar.dge$log2FC.VCP, y = mass_spec_barbar.dge$log2FC.A1, paired = TRUE )

barbar.mass_spec.scatter <- ggscatter(mass_spec_barbar.dge, x = "log2FC.VCP", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( mass~spec~log~Fold~Change~VCP:CTRL), ylab = expression(log[2]~Fold~Change~A1:A0), size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  xlim(-3,3) + ylim(-5,5) +
    geom_smooth(data = mass_spec_barbar.dge, aes(x = log2FC.VCP, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(mass_spec_barbar.dge, gene_name %in% all.reactivity.go, log2FC.VCP > 0, log2FC.A1 > 0), aes(x = log2FC.VCP, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7, max.overlaps = Inf) +  
  # geom_text_repel(data = filter(mass_spec_barbar.dge, gene_name %in% all.reactivity.go, log2FC.VCP < 0, log2FC.A1 < 0), aes(x = log2FC.VCP, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +  
  annotate(geom="text", x=1.5, y=5, label="VCP & A1 Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-1.5, y=-5, label="VCP & A1 Down", color="dodgerblue3", size = 3.2) + labs(subtitle = "Differential protein expression")
barbar.mass_spec.scatter

mass_spec_barbar.dge %>% drop_na(log2FC.VCP, log2FC.A1) %>% filter(log2FC.VCP > 0) %>% nrow # 334
mass_spec_barbar.dge %>% drop_na(log2FC.VCP, log2FC.A1) %>% filter(log2FC.VCP > 0, log2FC.A1 > 0) %>% nrow # 178
mass_spec_barbar.dge %>% drop_na(log2FC.VCP, log2FC.A1) %>% filter(log2FC.VCP < 0) %>% nrow # 741
mass_spec_barbar.dge %>% drop_na(log2FC.VCP, log2FC.A1) %>% filter(log2FC.VCP < 0, log2FC.A1 < 0) %>% nrow # 357
```

### Merge

```{r}
fig3.merged <- ggarrange(
  ggarrange(patani_barbar_venn_up, patani_barbar_venn_down, ncol = 2, labels = c("A", "")), 
  ggarrange(A1_VCP_scatter, dge_overlap_gprofiles_filt.go_curated, ncol = 2, labels = c("B", "C"), widths = c(1,0.8)),
  ggarrange(barbar.patani.progeny.pathwayActivity, A1_vcp_TFactivities.scatter, ncol = 2, labels = c("D", "E")),
  nrow = 3, heights = c(0.7,1.5,1.5))
fig3.merged
ggsave(fig3.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/fig3.merged.png", height = 11.75, width = 8.75)
```

## Fig 4 C9orf72 and SOD1

### Venn overlap


```{r}
barbar.res.dge.up <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
barbar.res.dge.down <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
tyzack.res.dge.up <- tyzack.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
tyzack.res.dge.down <- tyzack.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
birger.res.dge.up <- birger.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
birger.res.dge.down <- birger.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

barbar.tyzack.birger.up_genes <- list("A1" = barbar.res.dge.up, "SOD1" = tyzack.res.dge.up, "C9orf72" = birger.res.dge.up)
barbar.tyzack.birger.down_genes <- list("A1" = barbar.res.dge.down, "SOD1" = tyzack.res.dge.down, "C9orf72" = birger.res.dge.down)
birger_tyzack_barbar_venn_up <- ggvenn(barbar.tyzack.birger.up_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen"),show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5))
birger_tyzack_barbar_venn_up
birger_tyzack_barbar_venn_down <- ggvenn(barbar.tyzack.birger.down_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5))
birger_tyzack_barbar_venn_down
ggsave(birger_tyzack_barbar_venn_up, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger_tyzack_barbar_venn_up.png", height = 5, width = 5)
ggsave(birger_tyzack_barbar_venn_down, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger_tyzack_barbar_venn_down.png", height = 5, width = 5)

# barbar.tyzack.up_genes <- list("A1 up" = barbar.res.dge.up, "SOD1 up" = tyzack.res.dge.up)
# barbar.tyzack.down_genes <- list("A1 down" = barbar.res.dge.down, "SOD1 down" = tyzack.res.dge.down)
# barbar.birger.up_genes <- list("A1 up" = barbar.res.dge.up, "C9orf72 up" = birger.res.dge.up)
# barbar.birger.down_genes <- list("A1 down" = barbar.res.dge.down, "C9orf72 down" = birger.res.dge.down)

# tyzack_barbar_venn_up <- ggvenn(barbar.tyzack.up_genes, fill_color = c("dodgerblue3", "firebrick1"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4)
# tyzack_barbar_venn_up
# tyzack_barbar_venn_down <- ggvenn(barbar.tyzack.down_genes, fill_color = c("dodgerblue3", "firebrick1"), show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4)
# tyzack_barbar_venn_down
# ggsave(tyzack_barbar_venn_up, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack_barbar_venn_up.png", height = 3, width = 3)
# ggsave(tyzack_barbar_venn_down, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack_barbar_venn_down.png", height = 3, width = 3)
# 
# birger_barbar_venn_up <- ggvenn(barbar.birger.up_genes, fill_color = c("dodgerblue3", "firebrick1"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4)
# birger_barbar_venn_up
# birger_barbar_venn_down <- ggvenn(barbar.birger.down_genes, fill_color = c("dodgerblue3", "firebrick1"), show_percentage = FALSE, stroke_size = 0.5, set_name_size = 4)
# birger_barbar_venn_down
# ggsave(birger_barbar_venn_up, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger_barbar_venn_up.png", height = 3, width = 3)
# ggsave(birger_barbar_venn_down, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger_barbar_venn_down.png", height = 3, width = 3)
```

### Scatter plots

```{r}
# SOD1 & A1
tyzack.dge <- tyzack.res %>% dplyr::select(gene_name, ensID, log2FC.SOD1 = log2FoldChange, FDR.SOD1 = padj)
barbar.dge <- barbar.res %>% dplyr::select(gene_name, ensID, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
tyzack_barbar.dge <- full_join(tyzack.dge, barbar.dge, by = c("gene_name", "ensID"))
tyzack_barbar.dge <- tyzack_barbar.dge %>% mutate(overlap = case_when(FDR.SOD1 < 0.05 & FDR.A1 < 0.05 ~ "overlapping", FDR.SOD1 < 0.05 ~ "SOD1 specific", FDR.A1 < 0.05 ~ "A1 specific", TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.SOD1 < 0.05 & log2FC.SOD1 > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in SOD1
                                                                  FDR.SOD1 < 0.05 & log2FC.SOD1 < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.SOD1 < 0.05 & log2FC.SOD1 > 0 ~ "SOD1 specific up", FDR.SOD1 < 0.05 & log2FC.SOD1 < 0 ~ "SOD1 specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up", FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down", TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "SOD1 specific up", "SOD1 specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "SOD1 specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(tyzack_barbar.dge$overlap)
table(tyzack_barbar.dge$overlap_direction)
tyzack_barbar.dge
write.csv(tyzack_barbar.dge, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar_tyzack.overlap.differentially.expressed.genes.csv")

cor.test(x = tyzack_barbar.dge$log2FC.SOD1, y = tyzack_barbar.dge$log2FC.A1) # 0.053
t.test( x = tyzack_barbar.dge$log2FC.SOD1, y = tyzack_barbar.dge$log2FC.A1, paired = TRUE )

a1_SOD1_scatter <- ggscatter(tyzack_barbar.dge, x = "log2FC.SOD1", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~SOD1:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = tyzack_barbar.dge, aes(x = log2FC.SOD1, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(tyzack_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go), aes(x = log2FC.SOD1, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5) +#, max.overlaps = Inf) +  
  annotate(geom="text", x=5, y=11, label="Up in SOD1 & A1", color="firebrick2", size = 2.8)  + annotate(geom="text", x=-5, y=-10, label="Down in SOD1 & A1", color="dodgerblue3", size = 2.8)
a1_SOD1_scatter
ggsave(a1_SOD1_scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/a1_SOD1_scatter.png", height = 8, width = 8)

# test overlap
SOD1.DGE.up.genes <- tyzack.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
A1.DGE.up.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
genome.size <- tyzack.res %>% distinct(gene_name) %>% nrow
tyzack.go.obj <- newGeneOverlap(SOD1.DGE.up.genes,
                         A1.DGE.up.genes,
                         genome.size=genome.size)
tyzack.go.obj <- testGeneOverlap(tyzack.go.obj)
tyzack.go.obj # overlapping p-value indicates whether the overlap is significant
print(tyzack.go.obj)
# GeneOverlap object:
# listA size=1305
# listB size=4686
# Intersection size=361
# Overlapping p-value=6.6e-106
# Jaccard Index=0.1


SOD1.DGE.down.genes <- tyzack.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
A1.DGE.down.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
genome.size <- tyzack.res %>% distinct(gene_name) %>% nrow
tyzack.go.obj <- newGeneOverlap(SOD1.DGE.down.genes,
                         A1.DGE.down.genes,
                         genome.size=genome.size)
tyzack.go.obj <- testGeneOverlap(tyzack.go.obj)
tyzack.go.obj # overlapping p-value indicates whether the overlap is significant
print(tyzack.go.obj)
# GeneOverlap object:
# listA size=1923
# listB size=4461
# Intersection size=614
# Overlapping p-value=4.1e-231
# Jaccard Index=0.1

# C9orf72 & A1
birger.dge <- birger.res %>% dplyr::select(gene_name, ensID, log2FC.C9orf72 = log2FoldChange, FDR.C9orf72 = padj)
barbar.dge <- barbar.res %>% dplyr::select(gene_name, ensID, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
birger_barbar.dge <- full_join(birger.dge, barbar.dge, by = c("gene_name", "ensID"))
birger_barbar.dge <- birger_barbar.dge %>% mutate(overlap = case_when(FDR.C9orf72 < 0.05 & FDR.A1 < 0.05 ~ "overlapping", FDR.C9orf72 < 0.05 ~ "C9orf72 specific", FDR.A1 < 0.05 ~ "A1 specific", TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in C9orf72
                                                                  FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0 ~ "C9orf72 specific up", FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0 ~ "C9orf72 specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up", FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down", TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "C9orf72 specific up", "C9orf72 specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "C9orf72 specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(birger_barbar.dge$overlap)
table(birger_barbar.dge$overlap_direction)
birger_barbar.dge
write.csv(birger_barbar.dge, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar_birger.overlap.differentially.expressed.genes.csv")

cor.test(x = birger_barbar.dge$log2FC.C9orf72, y = birger_barbar.dge$log2FC.A1) # 0.101
t.test( x = birger_barbar.dge$log2FC.C9orf72, y = birger_barbar.dge$log2FC.A1, paired = TRUE )

a1_C9orf72_scatter <- ggscatter(birger_barbar.dge, x = "log2FC.C9orf72", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~C9orf72:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), cor.coef.size = 2.8, size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = birger_barbar.dge, aes(x = log2FC.C9orf72, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(birger_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go), aes(x = log2FC.C9orf72, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5) +#, max.overlaps = Inf) +  
  annotate(geom="text", x=5, y=10, label="Up in C9orf72 & A1", color="firebrick2", size = 2.8)  + annotate(geom="text", x=-5, y=-10, label="Down in C9orf72 & A1", color="dodgerblue3", size = 2.8)
a1_C9orf72_scatter
ggsave(a1_C9orf72_scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/a1_C9orf72_scatter.png", height = 8, width = 8)

# test overlap
C9orf72.DGE.up.genes <- birger.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
A1.DGE.up.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
genome.size <- birger.res %>% distinct(gene_name) %>% nrow
birger.go.obj <- newGeneOverlap(C9orf72.DGE.up.genes,
                         A1.DGE.up.genes,
                         genome.size=genome.size)
birger.go.obj <- testGeneOverlap(birger.go.obj)
birger.go.obj # overlapping p-value indicates whether the overlap is significant
print(birger.go.obj)
# GeneOverlap object:
# listA size=847
# listB size=4686
# Intersection size=222
# Overlapping p-value=1.9e-60
# Jaccard Index=0.0


C9orf72.DGE.down.genes <- birger.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
A1.DGE.down.genes <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
genome.size <- birger.res %>% distinct(gene_name) %>% nrow
birger.go.obj <- newGeneOverlap(C9orf72.DGE.down.genes,
                         A1.DGE.down.genes,
                         genome.size=genome.size)
birger.go.obj <- testGeneOverlap(birger.go.obj)
birger.go.obj # overlapping p-value indicates whether the overlap is significant
print(birger.go.obj)
# GeneOverlap object:
# listA size=989
# listB size=4461
# Intersection size=293
# Overlapping p-value=5.6e-99
# Jaccard Index=0.1
```

### GO terms

```{r}
# SOD1
tyzack_barbar.dge_overlap_genes <- tyzack_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("ensID")
tyzack_barbar.dge_overlap_genes <- gost(tyzack_barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
tyzack_barbar.dge_overlap_gprofiles_filt <- tyzack_barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
tyzack_barbar.dge_overlap_gprofiles_filt
tyzack_barbar.dge_overlap_gprofiles_filt_dge.up <- tyzack_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping up")
tyzack_barbar.dge_overlap_gprofiles_filt_dge.down <- tyzack_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping down")
paste('"',unique(tyzack_barbar.dge_overlap_gprofiles_filt_dge.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
tyzack_barbar.dge.up_curated_list <- c("immune system process",
"regulation of cell death",
"response to stimulus",
"vesicle",
"ECM proteoglycans",
"anatomical structure morphogenesis",
"positive regulation of cell migration",
"extracellular exosome",
"cellular response to chemical stimulus",
"collagen-containing extracellular matrix",
"endoplasmic reticulum",
"Extracellular matrix organization")
tyzack_barbar.dge_overlap_gprofiles_filt_dge.up <- tyzack_barbar.dge_overlap_gprofiles_filt_dge.up %>% filter(term_name %in% tyzack_barbar.dge.up_curated_list) %>% mutate(query = "upregulated")
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("response to endoplasmic reticulum stress", "endoplasmic reticulum stress", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("cell morphogenesis involved in differentiation", "cell morphogenesis", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
paste('"',unique(tyzack_barbar.dge_overlap_gprofiles_filt_dge.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
tyzack_barbar.dge.up_curated_list <- c(
"neuron projection development",
"cell development",
"plasma membrane",
"generation of neurons",
"synaptic signaling",
"cell junction",
"cell projection",
"nervous system development",
"synapse")
tyzack_barbar.dge_overlap_gprofiles_filt_dge.down <- tyzack_barbar.dge_overlap_gprofiles_filt_dge.down %>% filter(term_name %in% tyzack_barbar.dge.up_curated_list) %>% mutate(query = "downregulated")
tyzack_barbar.dge_overlap_gprofiles_filt_dge.up_down <- bind_rows(tyzack_barbar.dge_overlap_gprofiles_filt_dge.up, tyzack_barbar.dge_overlap_gprofiles_filt_dge.down) %>% 
  mutate(query = factor(query, levels = c("upregulated", "downregulated")))
tyzack_barbar.go_curated <- curated_profiles(gprofiles = tyzack_barbar.dge_overlap_gprofiles_filt_dge.up_down, colours = 2) + labs(subtitle = "A1 & SOD1")
tyzack_barbar.go_curated
ggsave(tyzack_barbar.go_curated, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack_barbar.go_curated.png", height = 8, width = 5)

tyzack_barbar.dge_overlap_gprofiles_filt <- apply(tyzack_barbar.dge_overlap_gprofiles_filt,2,as.character)
write.csv(tyzack_barbar.dge_overlap_gprofiles_filt, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack_barbar_overlap_go_terms.csv")

 
# C9orf72
birger_barbar.dge_overlap_genes <- birger_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("ensID")
birger_barbar.dge_overlap_genes <- gost(birger_barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
birger_barbar.dge_overlap_gprofiles_filt <- birger_barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
birger_barbar.dge_overlap_gprofiles_filt
birger_barbar.dge_overlap_gprofiles_filt_dge.up <- birger_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping up")
birger_barbar.dge_overlap_gprofiles_filt_dge.down <- birger_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping down")
paste('"',unique(birger_barbar.dge_overlap_gprofiles_filt_dge.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
birger_barbar.dge.up_curated_list <- c(
"immune system process",
"response to hypoxia",
"response to chemical",
"cellular response to chemical stimulus",
"vesicle",
"cell motility",
"Elastic fibre formation",
"regulation of cell motility",
"mesenchymal cell differentiation",
"cell migration",
"regulation of developmental process",
"anatomical structure morphogenesis",
"tube development",
"collagen-containing extracellular matrix",
"Extracellular matrix organization")
birger_barbar.dge_overlap_gprofiles_filt_dge.up <- birger_barbar.dge_overlap_gprofiles_filt_dge.up %>% filter(term_name %in% birger_barbar.dge.up_curated_list) %>% mutate(query = "upregulated")
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("response to endoplasmic reticulum stress", "endoplasmic reticulum stress", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("cell morphogenesis involved in differentiation", "cell morphogenesis", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
# bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", bir_c9orf_dge_ir_gprofiles_curated_irdown_expup$term_name)
paste('"',unique(birger_barbar.dge_overlap_gprofiles_filt_dge.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
birger_barbar.dge.up_curated_list <- c(
"synapse organization",
"neuron differentiation",
"plasma membrane",
"glutamatergic synapse",
"generation of neurons",
"cell projection",
"synapse",
"cell junction",
"nervous system development")
birger_barbar.dge_overlap_gprofiles_filt_dge.down <- birger_barbar.dge_overlap_gprofiles_filt_dge.down %>% filter(term_name %in% birger_barbar.dge.up_curated_list) %>% mutate(query = "downregulated")
birger_barbar.dge_overlap_gprofiles_filt_dge.up_down <- bind_rows(birger_barbar.dge_overlap_gprofiles_filt_dge.up, birger_barbar.dge_overlap_gprofiles_filt_dge.down)
birger_barbar.go_curated <- curated_profiles(gprofiles = birger_barbar.dge_overlap_gprofiles_filt_dge.up_down, colours = 2) + labs(subtitle = "A1 & C9orf72")
birger_barbar.go_curated
ggsave(birger_barbar.go_curated, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger_barbar.go_curated.png", height = 8, width = 5)
```


### GSEA glut/phag

```{r}
# GSEA violin plots: Phagocytosis, Glutamate uptke
tyzack.res_phagocytosis <- filter(tyzack.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(gene_set = "phagocytosis", dataset = "SOD1")
tyzack.res_glutamate_uptake <- filter(tyzack.res, gene_name %in% glutamate.uptake.go) %>% mutate(gene_set = "glutamate uptake", dataset = "SOD1")
# tyzack.res_gene_sets <- bind_rows(tyzack.res_phagocytosis, tyzack.res_glutamate_uptake)
birger.res_phagocytosis <- filter(birger.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(gene_set = "phagocytosis", dataset = "C9orf72")
birger.res_glutamate_uptake <- filter(birger.res, gene_name %in% glutamate.uptake.go) %>% mutate(gene_set = "glutamate uptake", dataset = "C9orf72")
tyzack.birger.res_gene_sets <- bind_rows(tyzack.res_phagocytosis, tyzack.res_glutamate_uptake, birger.res_phagocytosis, birger.res_glutamate_uptake)

tyzack.birger.stat.test <- tyzack.birger.res_gene_sets %>% group_by(dataset, gene_set) %>%  t_test(log2FoldChange ~ 1, mu = 0) %>%  add_significance()  %>% add_xy_position(x = "log2FoldChange") %>% mutate(y.position = 4)
tyzack.birger.stat.test
#  gene_set         dataset .y.            group1 group2         n statistic    df        p p.signif y.position groups     xmin  xmax
#   <chr>            <chr>   <chr>          <chr>  <chr>      <int>     <dbl> <dbl>    <dbl> <chr>         <dbl> <list>    <int> <int>
# 1 glutamate uptake C9orf72 log2FoldChange 1      null model   404    -4.03    402 6.75e- 5 ****              4 <chr [2]>     1     2
# 2 phagocytosis     C9orf72 log2FoldChange 1      null model   364     1.53    252 1.26e- 1 ns                4 <chr [2]>     1     2
# 3 glutamate uptake SOD1    log2FoldChange 1      null model   404   -10.1     397 1.16e-21 ****              4 <chr [2]>     1     2
# 4 phagocytosis     SOD1    log2FoldChange 1      null model   364    -0.883   226 3.78e- 1 ns                4 <chr [2]>     1     2

birger.tyzack.gene_sets_sina_plot <- ggplot(tyzack.birger.res_gene_sets, aes(x = gene_set, y= log2FoldChange, color = gene_set)) + 
      geom_violin() + geom_sina(size = 0.5) + geom_boxplot(data = tyzack.birger.res_gene_sets, aes(fill = gene_set), colour = "black", width=0.2, outlier.shape = NA) +
      scale_colour_manual(values = c("firebrick2", "dodgerblue3")) + scale_fill_manual(values = c("firebrick2", "dodgerblue3")) +  
      theme_classic() + theme(legend.position = "none", axis.text=element_text(size=8)) + #, axis.text.x = element_text(angle = 45, hjust = 1)) +
      facet_wrap(. ~ dataset) + geom_hline(yintercept = 0, linetype = 2) +
  stat_pvalue_manual(tyzack.birger.stat.test,x = "gene_set", hide.ns = TRUE) +
  ylab(label = expression(GSEA~LFC~ALS:CTRL) ) + xlab("") + ylim(c(-4,4)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 1, yend= 4), arrow=arrow(length=unit(0.3,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -1, yend= -4),arrow=arrow(length=unit(0.3,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 2.5, label = "ALS Up", colour = "black", size = 2.8, angle = 90) +  annotate("text", x =2.35, y = -2.5, label = "ALS Down", colour = "black", size = 2.89, angle = 90) 
birger.tyzack.gene_sets_sina_plot
ggsave(birger.tyzack.gene_sets_sina_plot, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger.tyzack.GSEA_phagocytosis_glutamate.uptake.png", height = 6, width = 5)

# birger.tyzack.gene_sets_sina_plot <- sinaplot.one.sample(data =  tyzack.birger.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 11,  label_number = 0, stat.y.position = 4, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( GSEA~log[2]~Fold~Change~ALS:CTRL )) + xlab("")  + theme(axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
#   geom_segment(aes(x = 2.5, xend = 2.5, y= 1, yend= 4), arrow=arrow(length=unit(0.3,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -1, yend= -4),arrow=arrow(length=unit(0.3,"cm")), color = "black") +
#   annotate("text", x = 2.35, y = 2.5, label = "SOD1 Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -2.5, label = "SOD1 Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-4,4)) +
#   facet_wrap(. ~ dataset)
# birger.tyzack.gene_sets_sina_plot
# ggsave(tyzack.gene_sets_sina_plot, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack.GSEA_phagocytosis_glutamate.uptake.png", height = 6, width = 3)

# birger.res_phagocytosis <- filter(birger.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(gene_set = "phagocytosis")
# birger.res_glutamate_uptake <- filter(birger.res, gene_name %in% glutamate.uptake.go) %>% mutate(gene_set = "glutamate uptake")
# birger.res_gene_sets <- bind_rows(birger.res_phagocytosis, birger.res_glutamate_uptake)
# birger.gene_sets_sina_plot <- sinaplot.one.sample(data =  birger.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 11,  label_number = 0, stat.y.position = 4, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( GSEA~log[2]~Fold~Change~C9orf72:CTRL )) + xlab("")  + theme(axis.text=element_text(size=10), axis.text.x = element_text(angle = 45, hjust = 1)) +
#   geom_segment(aes(x = 2.5, xend = 2.5, y= 1, yend= 4), arrow=arrow(length=unit(0.3,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -1, yend= -4),arrow=arrow(length=unit(0.3,"cm")), color = "black") +
#   annotate("text", x = 2.35, y = 2.5, label = "C9orf72 Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -2.5, label = "C9orf72 Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-4,4)) 
# birger.gene_sets_sina_plot
# ggsave(birger.gene_sets_sina_plot, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger.GSEA_phagocytosis_glutamate.uptake.png", height = 6, width = 3)

# SOD1
# GSEA phagocytosis
tyzack.geneList <- tyzack.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(tyzack.geneList) <- tyzack.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
tyzack.geneList = sort(tyzack.geneList, decreasing = TRUE)
tyzack.phagocytosis_genes.list <- list("phagocytosis" = tyzack.res$gene_name[tyzack.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
tyzack.phagocytosis_fgseaRes <- fgsea(pathways = tyzack.phagocytosis_genes.list, stats = tyzack.geneList)
tyzack.phagocytosis_fgseaRes
#         pathway      pval      padj    log2err        ES       NES size                            leadingEdge
# 1: phagocytosis 0.3729642 0.3729642 0.08383611 -0.267089 -1.028932  227 XKR7,TAFA4,SYT7,VAV3,PECAM1,ADGRB1,...

# GSEA glutamate uptake
tyzack.glutamate.uptake_genes.list <- list("glutamate.uptake" = tyzack.res$gene_name[tyzack.res$gene_name %in% glutamate.uptake.go])
tyzack.glutamate.uptake_fgseaRes <- fgsea(pathways = tyzack.glutamate.uptake_genes.list, stats = tyzack.geneList)
tyzack.glutamate.uptake_fgseaRes
#             pathway  pval  padj log2err         ES       NES size                                    leadingEdge
# 1: glutamate.uptake 1e-10 1e-10      NA -0.6068929 -2.469888  397 NLGN4Y,SLITRK2,ADRA1A,SLITRK4,NETO1,LRRTM3,...

# C9orf72
# GSEA phagocytosis
birger.geneList <- birger.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(birger.geneList) <- birger.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
birger.geneList = sort(birger.geneList, decreasing = TRUE)
birger.phagocytosis_genes.list <- list("phagocytosis" = birger.res$gene_name[birger.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
birger.phagocytosis_fgseaRes <- fgsea(pathways = birger.phagocytosis_genes.list, stats = birger.geneList)
birger.phagocytosis_fgseaRes
#        pathway       pval       padj   log2err        ES      NES size                          leadingEdge
# 1: phagocytosis 0.08823529 0.08823529 0.2616635 0.2617714 1.168708  253 MYH2,NCF2,MARCO,HCK,PPARG,FCGR2A,...

# GSEA glutamate uptake
birger.glutamate.uptake_genes.list <- list("glutamate.uptake" = birger.res$gene_name[birger.res$gene_name %in% glutamate.uptake.go])
birger.glutamate.uptake_fgseaRes <- fgsea(pathways = birger.glutamate.uptake_genes.list, stats = birger.geneList)
birger.glutamate.uptake_fgseaRes
#             pathway        pval        padj   log2err         ES       NES size                                  leadingEdge
# 1: glutamate.uptake 2.42135e-07 2.42135e-07 0.6749629 -0.3842181 -1.695011  402 SYT4,CACNG3,ELAVL4,SLC1A6,GUCY1A1,LRRTM3,...
```

### PROGENy and DoRothEA

```{r}
library(progeny)
library(dorothea)
tyzack.res.matrix <- tyzack.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
tyzack.SOD1.PathwayActivity_zscore <- progeny(tyzack.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(tyzack.SOD1.PathwayActivity_zscore) <- "NES"
tyzack.SOD1.PathwayActivity_zscore_df <- as.data.frame(tyzack.SOD1.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
birger.res.matrix <- birger.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
birger.C9orf72.PathwayActivity_zscore <- progeny(birger.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(birger.C9orf72.PathwayActivity_zscore) <- "NES"
birger.C9orf72.PathwayActivity_zscore_df <- as.data.frame(birger.C9orf72.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
barbar.res.matrix <- barbar.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
barbar.A1.PathwayActivity_zscore <- progeny(barbar.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t() # enrichment analysis of each pathway activity
colnames(barbar.A1.PathwayActivity_zscore) <- "NES"
barbar.A1.PathwayActivity_zscore_df <- as.data.frame(barbar.A1.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))

barbar.tyzack.birger.PathwayActivity_zscore_df <- mutate(barbar.A1.PathwayActivity_zscore_df, dataset = "A1") %>% bind_rows(mutate(birger.C9orf72.PathwayActivity_zscore_df, dataset = "C9orf72")) %>% bind_rows(mutate(tyzack.SOD1.PathwayActivity_zscore_df, dataset = "SOD1")) 
barbar.tyzack.birger.PathwayActivity_zscore_df.meanNES <- barbar.tyzack.birger.PathwayActivity_zscore_df %>% group_by(Pathway) %>% summarise(mean_NES = mean(NES)) %>% arrange(mean_NES)
barbar.tyzack.birger.PathwayActivity_zscore_df$Pathway <- factor(barbar.tyzack.birger.PathwayActivity_zscore_df$Pathway, levels = barbar.tyzack.birger.PathwayActivity_zscore_df.meanNES$Pathway)

barbar.tyzack.birger.progeny.pathwayActivity <- ggplot(barbar.tyzack.birger.PathwayActivity_zscore_df, aes(x=Pathway, y = NES, fill = dataset)) + theme_classic() +
  geom_bar(stat="identity", position=position_dodge()) + theme(legend.position = "top", legend.title = element_blank(), axis.title = element_text(size = 12),
        axis.text.x = element_text(angle = 45, hjust = 1, size = 8), axis.text.y = element_text(size =8)) +
  geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") + scale_fill_manual( values = c("firebrick3", "dodgerblue3", "forestgreen")) + xlab(expression())  + ylab(expression(Normalised~enrichment~score)) + labs(subtitle = "Signalling Pathway activity")
barbar.tyzack.birger.progeny.pathwayActivity
ggsave(barbar.tyzack.birger.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar.tyzack.birger.progeny.pathwayActivity.png", height = 5, width = 8)

# # separate plots for each dataset
# tyzack.SOD1.progeny.pathwayActivity <- ggplot(tyzack.SOD1.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),
#         axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"), axis.text.y = element_text(size =8, face= "bold")) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Pathways") + ylab( expression(Normalised~Enrichment) )  + labs(subtitle = "SOD1 vs CTRL")
# tyzack.SOD1.progeny.pathwayActivity
# ggsave(tyzack.SOD1.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack.progeny.pathwayActivity.png", height = 6, width = 6)
# 
# birger.C9orf72.progeny.pathwayActivity <- ggplot(birger.C9orf72.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),
#         axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"), axis.text.y = element_text(size =8, face= "bold")) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +
#     xlab("Pathways") + ylab( expression(Normalised~Enrichment) )  + labs(subtitle = "C9orf72 vs CTRL")
# birger.C9orf72.progeny.pathwayActivity
# ggsave(birger.C9orf72.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/birger.progeny.pathwayActivity.png", height = 6, width = 6)
# 
# barbar.A1.progeny.pathwayActivity <- ggplot(barbar.A1.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
#     theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),
#         axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"), axis.text.y = element_text(size =8, face= "bold")) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +   xlab("Pathways") + ylab( expression(Normalised~Enrichment) ) + labs(subtitle = "A1 vs A0")
# barbar.A1.progeny.pathwayActivity
# ggsave(barbar.A1.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/barbar.A1.progeny.pathwayActivity.png", height = 6, width = 6)
# 
# tyzack.birger.progeny.pathwayActivity <- ggarrange(barbar.A1.progeny.pathwayActivity, tyzack.SOD1.progeny.pathwayActivity, birger.C9orf72.progeny.pathwayActivity, nrow = 1, ncol = 3)
# tyzack.birger.progeny.pathwayActivity
# ggsave(tyzack.birger.progeny.pathwayActivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack.birger.progeny.pathwayActivity.png", height = 5, width = 9)


# DoRothEA
library(dorothea)
data(dorothea_hs, package = "dorothea")
regulons <- dorothea_hs %>%  dplyr::filter(confidence %in% c("A", "B","C"))
barbar.A1.tf_activities_stat <- dorothea::run_viper(barbar.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
barbar.A1.tf_activities_stat_top25 <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
barbar.A1.tf_activities_stat.df <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(log2FC.A1 = "log2FoldChange")
tyzack.SOD1.tf_activities_stat <- dorothea::run_viper(tyzack.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
tyzack.SOD1.tf_activities_stat_top25 <- tyzack.SOD1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
tyzack.SOD1.tf_activities_stat.df <- tyzack.SOD1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.SOD1 = "log2FoldChange")
birger.C9orf72.tf_activities_stat <- dorothea::run_viper(birger.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
birger.C9orf72.tf_activities_stat_top25 <- birger.C9orf72.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
birger.C9orf72.tf_activities_stat.df <- birger.C9orf72.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.C9orf72 = "log2FoldChange")
barbar.A1.tf_activities_stat.df <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(log2FC.A1 = "log2FoldChange")
A1.SOD1.tf_activities <- tyzack.SOD1.tf_activities_stat.df %>% left_join(barbar.A1.tf_activities_stat.df, by = "gene_name")
A1.C9orf72.tf_activities <- birger.C9orf72.tf_activities_stat.df %>% left_join(barbar.A1.tf_activities_stat.df, by = "gene_name")


cor.test(x = A1.SOD1.tf_activities$log2FC.SOD1, y = A1.SOD1.tf_activities$log2FC.A1) # 0.21
t.test( x = A1.SOD1.tf_activities$log2FC.SOD1, y = A1.SOD1.tf_activities$log2FC.A1, paired = TRUE )
A1_SOD1_TFactivities.scatter <- ggscatter(A1.SOD1.tf_activities, x = "log2FC.SOD1", y = "log2FC.A1", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~SOD1:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = A1.SOD1.tf_activities, aes(x = log2FC.SOD1, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(A1.SOD1.tf_activities, log2FC.SOD1> 1, log2FC.A1 > 2), aes(x = log2FC.SOD1, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
      geom_text_repel(data = filter(A1.SOD1.tf_activities, log2FC.SOD1 < -1, log2FC.A1 < -2), aes(x = log2FC.SOD1, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=3, y=15, label="Up in SOD1 & A1", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="Down in SOD1 & A1", color="dodgerblue3", size = 3.2) + labs(subtitle = "SOD1 Transcription Factor activity")
A1_SOD1_TFactivities.scatter
ggsave(A1_SOD1_TFactivities.scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_SOD1_TFactivities.scatter.png", height = 8, width = 8)

barbar.A1.tf_activities_stat <- dorothea::run_viper(barbar.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
barbar.A1.tf_activities_stat_top25 <- barbar.A1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))

cor.test(x = A1.C9orf72.tf_activities$log2FC.C9orf72, y = A1.C9orf72.tf_activities$log2FC.A1) # 0.22
t.test( x = A1.C9orf72.tf_activities$log2FC.C9orf72, y = A1.C9orf72.tf_activities$log2FC.A1, paired = TRUE )
A1_C9orf72_TFactivities.scatter <- ggscatter(A1.C9orf72.tf_activities, x = "log2FC.C9orf72", y = "log2FC.A1", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson", xlab = expression( Log[2]~Fold~Change~C9orf72:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = A1.C9orf72.tf_activities, aes(x = log2FC.C9orf72, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(A1.C9orf72.tf_activities, log2FC.C9orf72> 1, log2FC.A1 > 2), aes(x = log2FC.C9orf72, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
      geom_text_repel(data = filter(A1.C9orf72.tf_activities, log2FC.C9orf72 < -1, log2FC.A1 < -2), aes(x = log2FC.C9orf72, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=3, y=15, label="Up in C9orf72 & A1", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="Down in C9orf72 & A1", color="dodgerblue3", size = 3.2) + labs(subtitle = "C9orf72 Transcription Factor activity")
A1_C9orf72_TFactivities.scatter
ggsave(A1_C9orf72_TFactivities.scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/A1_C9orf72_TFactivities.scatter.png", height = 8, width = 8)




# tyzack.SOD1.dorotheaActivity <- ggplot(tyzack.SOD1.tf_activities_stat_top25,aes(x = reorder(gene_name, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +  scale_fill_gradient2(low = "darkblue", high = "indianred",  mid = "whitesmoke", midpoint = 0) + theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"),axis.text.y = element_text(size =8, face= "bold")) +
#     geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +   xlab("Transcription Factors") + ylab( expression(Normalised~enrichment) )  + labs(subtitle = "SOD1 vs CTRL")
# tyzack.SOD1.dorotheaActivity
# ggsave(tyzack.SOD1.dorotheaActivity, filename = "/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/tyzack.dorotheaActivity.png", height = 6, width = 6)
# 
# birger.C9orf72.dorotheaActivity <- ggplot(birger.C9orf72.tf_activities_stat_top25,aes(x = reorder(gene_name, NES), y = NES)) + 
#     geom_bar(aes(fill = NES), stat = "identity") +  scale_fill_gradient2(low = "darkblue", high = "indianred",  mid = "whitesmoke", midpoint = 0) + theme_classic() +
#     theme(legend.position = "none", axis.title = element_text(face = "bold", size = 12),axis.text.x = element_text(angle = 45, hjust = 1, size =8, face= "bold"),axis.text.y = element_text(size =8, face= "bold")) +
#     geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") +   xlab("Transcription Factors") + ylab( expression(Normalised~enrichment) )  + labs(subtitle = "C9orf72 vs CTRL")
# birger.C9orf72.dorotheaActivity
# ggsave(birger.C9orf72.dorotheaActivity, filename = "/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/birger.dorotheaActivity.png", height = 6, width = 6)
# 
# tyzack.birger.dorothea.TFactivity <- ggarrange(tyzack.SOD1.dorotheaActivity, birger.C9orf72.dorotheaActivity, nrow = 1, ncol = 2)
# tyzack.birger.dorothea.TFactivity
# ggsave(tyzack.birger.dorothea.TFactivity, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack.birger.dorothea.TFactivity.png", height = 5, width = 9)

# tyzack.SOD1.tf_activities_stat.df <- tyzack.SOD1.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.SOD1 = "log2FoldChange")
# birger.C9orf72.tf_activities_stat.df <- birger.C9orf72.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(log2FC.C9orf72 = "log2FoldChange")
# C9orf72.SOD1.tf_activities <- tyzack.SOD1.tf_activities_stat.df %>% left_join(birger.C9orf72.tf_activities_stat.df, by = "gene_name")
# 
# cor.test(x = C9orf72.SOD1.tf_activities$log2FC.SOD1, y = C9orf72.SOD1.tf_activities$log2FC.C9orf72) # 0.265
# t.test( x = C9orf72.SOD1.tf_activities$log2FC.SOD1, y = C9orf72.SOD1.tf_activities$log2FC.C9orf72, paired = TRUE )
# 
# tyzack.birger.TF.dorothea.scatter <- ggscatter(C9orf72.SOD1.tf_activities, x = "log2FC.SOD1", y = "log2FC.C9orf72", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
#           cor.coef = TRUE, cor.method = "pearson",xlab = expression( Log[2]~Fold~Change~SOD1:CTRL), ylab = expression(Log[2]~Fold~Change~C9orf72:CTRL), size = 0.5) +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-6,6) + ylim(-6,6) +
#     geom_smooth(data = C9orf72.SOD1.tf_activities, aes(x = log2FC.SOD1, y = log2FC.C9orf72), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
#     geom_text_repel(data = filter(C9orf72.SOD1.tf_activities, abs(log2FC.SOD1) > 1, abs(log2FC.C9orf72) > 2), aes(x = log2FC.SOD1, y = log2FC.C9orf72, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
#   annotate(geom="text", x=3, y=6, label="Up in both SOD1 & C9orf72", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-6, label="Down in both SOD1 & C9orf72", color="dodgerblue3", size = 3.2)
# tyzack.birger.TF.dorothea.scatter
# ggsave(tyzack.birger.TF.dorothea.scatter, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/tyzack.birger.TFdorothea.scatter.png", height = 8, width = 8)

```



### Merge

```{r}
fig4.merged <- ggarrange(
  ggarrange(birger_tyzack_barbar_venn_up, a1_SOD1_scatter, tyzack_barbar.go_curated, ncol = 3, labels = c("A", "B", "D"), widths = c(0.8,1.1,1)), 
  ggarrange(birger_tyzack_barbar_venn_down, a1_C9orf72_scatter, birger_barbar.go_curated,ncol = 3, labels = c("", "C", "E"), widths = c(0.8,1.1,1)), 
  ggarrange(birger.tyzack.gene_sets_sina_plot, barbar.tyzack.birger.progeny.pathwayActivity, ncol = 2, labels = c("F", "G"), widths = c(0.9,1.1)),
  ggarrange(A1_SOD1_TFactivities.scatter, A1_C9orf72_TFactivities.scatter, ncol = 2, labels = c("H", "I")),
  nrow = 4, heights = c(1,1,1,1))
fig4.merged
ggsave(fig4.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/fig4.merged.png", height = 14, width = 10)
```

## Fig 5 TRAP TDP43 Membralin

### SOD1 TRAP

#### Venn

```{r}
# Overlap SOD1 SOD1_TRAP with A1 barbar
barbar.res.dge.up <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
barbar.res.dge.down <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
cleveland.res.dge.up <- cleveland.res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
cleveland.res.dge.down <- cleveland.res %>% filter(padj < 0.05, log2FoldChange < 0, gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

cleveland.A1.up_genes <- list("SOD1-TRAP" = cleveland.res.dge.up, "A1" = barbar.res.dge.up)
cleveland.A1.down_genes <- list("SOD1-TRAP" = cleveland.res.dge.down, "A1" = barbar.res.dge.down)
cleveland.A1_venn_up <- ggvenn(cleveland.A1.up_genes, fill_color = c("dodgerblue3", "firebrick2", "forestgreen"),
                               show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
cleveland.A1_venn_down <- ggvenn(cleveland.A1.down_genes, fill_color = c("dodgerblue3", "firebrick2"), show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "dodgerblue3"))
cleveland.A1_venn <- ggarrange(cleveland.A1_venn_up, cleveland.A1_venn_down, nrow = 2)
cleveland.A1_venn

# other venn packages
# as.ggplot(~display_venn(cleveland.A1.down_genes, fill = c("dodgerblue3", "firebrick2"))) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "dodgerblue3"))
# 
# library(eulerr)
# cleveland.A1.up_genes <- list("SOD1-TRAP" = nrow(filter(cleveland.res, padj < 0.05, log2FoldChange > 0, gene_name %in% gene2ens$gene_name)), "A1" = nrow(filter(barbar.res, padj < 0.05, log2FoldChange > 0)))
# cleveland.A1.down_genes <- list("SOD1-TRAP" = nrow(cleveland.res.dge.down), "A1" = nrow(barbar.res.dge.down))
# plot(euler(c("A"=5,"B"=4,"A&B"=2)),quantities = TRUE,fills=c("white","white","red"))
# plot(euler(cleveland.A1.up_genes),quantities = TRUE,fills=c("white","white","red"))
# 
# ggVennDiagram(cleveland.A1.down_genes, label_alpha = 0)

SOD1_TRAP_A1_up.go.obj <- newGeneOverlap(barbar.dge.up,
                         cleveland.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(SOD1_TRAP_A1_up.go.obj)
# GeneOverlap object:
# listA size=4686
# listB size=500
# Intersection size=153
# Overlapping p-value=3.1e-51
# Jaccard Index=0.0

SOD1_TRAP_A1_down.go.obj <- newGeneOverlap(barbar.dge.down,
                         cleveland.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(SOD1_TRAP_A1_down.go.obj)
# GeneOverlap object:
# listA size=4461
# listB size=356
# Intersection size=101
# Overlapping p-value=5.7e-33
# Jaccard Index=0.0

# Define ALS & A1 genes:
patani_birger_tyzack_barbar.dge.up <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP > 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 > 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0) & (FDR.A1 < 0.05 & log2FC.A1 > 0)) %>% pull(gene_name) %>% unique 
patani_birger_tyzack_barbar.dge.down <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP < 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 < 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0) & (FDR.A1 < 0.05 & log2FC.A1 < 0) ) %>% pull(gene_name) %>% unique 
cleveland.res.dge.up <- cleveland.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
cleveland.res.dge.down <- cleveland.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

cleveland.ALS.up_genes <- list("TRAP" = cleveland.res.dge.up, "ALS" = patani_birger_tyzack_barbar.dge.up)
cleveland.ALS.down_genes <- list("TRAP" = cleveland.res.dge.down, "ALS" = patani_birger_tyzack_barbar.dge.down)
cleveland.ALS_venn_up <- ggvenn(cleveland.ALS.up_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
cleveland.ALS_venn_up
cleveland.ALS_venn_down <- ggvenn(cleveland.ALS.down_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
cleveland.ALS_venn_down

TRAP_ALS_up.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.up,
                         cleveland.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TRAP_ALS_up.go.obj)
# GeneOverlap object:
# listA size=1541
# listB size=500
# Intersection size=69
# Overlapping p-value=4e-30
# Jaccard Index=0.0

TRAP_ALS_down.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.down,
                         cleveland.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TRAP_ALS_down.go.obj)
# GeneOverlap object:
# listA size=2159
# listB size=356
# Intersection size=70
# Overlapping p-value=1.6e-31
# Jaccard Index=0.0
```

#### Scatterplot

```{r}
cleveland.dge <- cleveland.res %>% dplyr::select(gene_name, log2FC.SOD1_TRAP = log2FoldChange, FDR.SOD1_TRAP = padj) %>% filter(gene_name %in% gene2ens$gene_name) 
barbar.dge <- barbar.res %>% dplyr::select(gene_name, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
cleveland_barbar.dge <- full_join(cleveland.dge, barbar.dge, by = c("gene_name"))

cleveland_barbar.dge <- cleveland_barbar.dge %>% mutate(overlap = case_when(FDR.SOD1_TRAP < 0.05 & FDR.A1 < 0.05 ~ "overlapping", # down in SOD1_TRAP
                                                        FDR.SOD1_TRAP < 0.05 ~ "SOD1_TRAP specific",
                                                        FDR.A1 < 0.05 ~ "A1 specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.SOD1_TRAP < 0.05 & log2FC.SOD1_TRAP > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in SOD1_TRAP
                                                                  FDR.SOD1_TRAP < 0.05 & log2FC.SOD1_TRAP < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.SOD1_TRAP < 0.05 & log2FC.SOD1_TRAP > 0 ~ "SOD1_TRAP specific up",
                                                                  FDR.SOD1_TRAP < 0.05 & log2FC.SOD1_TRAP < 0 ~ "SOD1_TRAP specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up",
                                                                  FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "SOD1_TRAP specific up", "SOD1_TRAP specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "SOD1_TRAP specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(cleveland_barbar.dge$overlap)
table(cleveland_barbar.dge$overlap_direction)
cleveland_barbar.dge

cor.test(x = cleveland_barbar.dge$log2FC.SOD1_TRAP, y = cleveland_barbar.dge$log2FC.A1) # 0.1088784 
t.test( x = cleveland_barbar.dge$log2FC.SOD1_TRAP, y = cleveland_barbar.dge$log2FC.A1, paired = TRUE )

A1_SOD1_TRAP_scatter <- ggscatter(cleveland_barbar.dge, x = "log2FC.SOD1_TRAP", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~SOD1~TRAP:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-5,5) + ylim(-10,11) +
    geom_smooth(data = cleveland_barbar.dge, aes(x = log2FC.SOD1_TRAP, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(cleveland_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go, log2FC.A1 < 9.5), aes(x = log2FC.SOD1_TRAP, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7, max.overlaps = 20) +  
  annotate(geom="text", x=3, y=11, label="SOD1 TRAP & A1 Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-10, label="SOD1 TRAP & A1 Down", color="dodgerblue3", size = 3.2) + labs(subtitle = "SOD1 TRAP vs A1")
A1_SOD1_TRAP_scatter
```


#### GO overlap

```{r}
# SOD1 TRAP
cleveland_barbar.dge_overlap_genes <- cleveland_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("gene_name")
cleveland_barbar.dge_overlap_genes <- gost(cleveland_barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
cleveland_barbar.dge_overlap_gprofiles_filt <- cleveland_barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
cleveland_barbar.dge_overlap_gprofiles_filt
cleveland_barbar.dge_overlap_gprofiles_filt_dge.up <- cleveland_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping up")
cleveland_barbar.dge_overlap_gprofiles_filt_dge.down <- cleveland_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping down")
paste('"',unique(cleveland_barbar.dge_overlap_gprofiles_filt_dge.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
cleveland_barbar.dge.up_curated_list <- c(
"cell communication",
"Signaling by Interleukins",
"cell surface receptor signaling pathway",
"inflammatory response",
"immune response",
"Interferon alpha/beta signaling",
"immune system process",
"defense response",
"response to stress",
"response to cytokine")
cleveland_barbar.dge_overlap_gprofiles_filt_dge.up <- cleveland_barbar.dge_overlap_gprofiles_filt_dge.up %>% filter(term_name %in% cleveland_barbar.dge.up_curated_list) %>% mutate(query = "upregulated")
cleveland_barbar.dge_overlap_gprofiles_filt_dge.up$term_name <- gsub("cell surface receptor", "receptor", cleveland_barbar.dge_overlap_gprofiles_filt_dge.up$term_name) %>% gsub("alpha/beta ", "", .)

paste('"',unique(cleveland_barbar.dge_overlap_gprofiles_filt_dge.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
cleveland_barbar.dge.up_curated_list <- c(
"spinal cord development",
"actin cytoskeleton",
"generation of neurons",
"adherens junction",
"neuron differentiation",
"Deactivation of the beta-catenin transactivating complex",
"nervous system development",
"cell cortex")
cleveland_barbar.dge_overlap_gprofiles_filt_dge.down <- cleveland_barbar.dge_overlap_gprofiles_filt_dge.down %>% filter(term_name %in% cleveland_barbar.dge.up_curated_list) %>% mutate(query = "downregulated")
cleveland_barbar.dge_overlap_gprofiles_filt_dge.down$term_name <- gsub("Deactivation of the beta-catenin transactivating complex", "beta-catenin complex", cleveland_barbar.dge_overlap_gprofiles_filt_dge.down$term_name)

cleveland_barbar.dge_overlap_gprofiles_filt_dge.up_down <- bind_rows(cleveland_barbar.dge_overlap_gprofiles_filt_dge.up, cleveland_barbar.dge_overlap_gprofiles_filt_dge.down) %>% 
  mutate(query = factor(query, levels = c("upregulated", "downregulated")))
cleveland_barbar.go_curated <- curated_profiles(gprofiles = cleveland_barbar.dge_overlap_gprofiles_filt_dge.up_down, colours = 2) + labs(subtitle = "A1 & SOD1 TRAP")
cleveland_barbar.go_curated

cleveland_barbar.dge_overlap_gprofiles_filt <- apply(cleveland_barbar.dge_overlap_gprofiles_filt,2,as.character)
write.csv(cleveland_barbar.dge_overlap_gprofiles_filt, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/cleveland_barbar_overlap_go_terms.csv")
```

#### GSEA glut/phag

```{r}
# GSEA violin plots: Phagocytosis, Glutamate uptke
cleveland.res_phagocytosis <-filter(cleveland.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  mutate(gene_set = "phagocytosis")
cleveland.res_glutamate_uptake <- filter(cleveland.res, gene_name %in% go.pathways.msigdb$GO_GLUTAMATERGIC_SYNAPSE) %>% mutate(gene_set = "glutamate uptake")
cleveland.res_gene_sets <- bind_rows(cleveland.res_phagocytosis, cleveland.res_glutamate_uptake) 

cleveland.gene_sets_sina_plot <- sinaplot.one.sample(data =  cleveland.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log[2]~Fold~Change~SOD1~TRAP:CTRL )) + xlab("")  +  theme(axis.text=element_text(size=9)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "SOD1 TRAP Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "SOD1 TRAP Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2))  + labs(subtitle = "GSEA SOD1-TRAP")
cleveland.gene_sets_sina_plot
# group            .y.   group1 group2         n statistic          p p.signif y.position
# * <chr>            <chr> <chr>  <chr>      <int>     <dbl>      <dbl> <chr>         <dbl>
# 1 glutamate uptake lfc   1      null model   399     43945 0.0793     ns                2
# 2 phagocytosis     lfc   1      null model   267     22136 0.00000162 ****              2

# GSEA phagocytosis
cleveland.geneList <- cleveland.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(cleveland.geneList) <- cleveland.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
cleveland.geneList = sort(cleveland.geneList, decreasing = TRUE)
cleveland.phagocytosis_genes.list <- list("phagocytosis" = cleveland.res$gene_name[cleveland.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
cleveland.phagocytosis_fgseaRes <- fgsea(pathways = cleveland.phagocytosis_genes.list, stats = cleveland.geneList)
cleveland.phagocytosis_fgseaRes
#         pathway         pval         padj   log2err        ES      NES size
# 1: phagocytosis 3.458555e-07 3.458555e-07 0.6749629 0.4797785 1.846946  256

# GSEA glutamate uptake
cleveland.glutamate.uptake_genes.list <- list("glutamate.uptake" = cleveland.res$gene_name[cleveland.res$gene_name %in% glutamate.uptake.go])
cleveland.glutamate.uptake_fgseaRes <- fgsea(pathways = cleveland.glutamate.uptake_genes.list, stats = cleveland.geneList)
cleveland.glutamate.uptake_fgseaRes
#            pathway pval padj    log2err        ES       NES size                              leadingEdge
# 1: glutamate.uptake    1    1 0.02779956 0.1386824 0.5631621  399 GRIN2B,GABRR1,SYNPO,FTCD,NPTX2,OPHN1,...
```

### TDP43 KO

#### Venn

```{r}
# Overlap TDP43_KO with A1 barbar
barbar.res.dge.up <- barbar.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
barbar.res.dge.down <- barbar.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
peng.res.dge.up <- peng.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
peng.res.dge.down <- peng.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

peng.A1.up_genes <- list("TDP43-KO" = peng.res.dge.up, "A1" = barbar.res.dge.up)
peng.A1.down_genes <- list("TDP43-KO" = peng.res.dge.down, "A1" = barbar.res.dge.down)
peng.A1_venn_up <- ggvenn(peng.A1.up_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
peng.A1_venn_down <- ggvenn(peng.A1.down_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "dodgerblue3"))
peng.A1_venn <- ggarrange(peng.A1_venn_up, peng.A1_venn_down, nrow = 2)
peng.A1_venn

TDP43_KO_A1_up.go.obj <- newGeneOverlap(barbar.res.dge.up,
                         peng.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TDP43_KO_A1_up.go.obj)
# GeneOverlap object:
# listA size=4686
# listB size=827
# Intersection size=214
# Overlapping p-value=2.8e-57
# Jaccard Index=0.0

TDP43_KO_A1_down.go.obj <- newGeneOverlap(barbar.res.dge.down,
                         peng.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TDP43_KO_A1_down.go.obj)
# GeneOverlap object:
# listA size=4461
# listB size=854
# Intersection size=255
# Overlapping p-value=6.5e-87
# Jaccard Index=0.1

# Define ALS & A1 genes:
patani_birger_tyzack_barbar.dge.up <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP > 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 > 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0) & (FDR.A1 < 0.05 & log2FC.A1 > 0)) %>% pull(gene_name) %>% unique 
patani_birger_tyzack_barbar.dge.down <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP < 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 < 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0) & (FDR.A1 < 0.05 & log2FC.A1 < 0) ) %>% pull(gene_name) %>% unique 
peng.res.dge.up <- peng.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
peng.res.dge.down <- peng.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

peng.ALS.up_genes <- list("TRAP" = peng.res.dge.up, "ALS" = patani_birger_tyzack_barbar.dge.up)
peng.ALS.down_genes <- list("TRAP" = peng.res.dge.down, "ALS" = patani_birger_tyzack_barbar.dge.down)
peng.ALS_venn_up <- ggvenn(peng.ALS.up_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
peng.ALS_venn_up
peng.ALS_venn_down <- ggvenn(peng.ALS.down_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
peng.ALS_venn_down

TDP43_KO_ALS_up.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.up,
                         peng.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TDP43_KO_ALS_up.go.obj)
# GeneOverlap object:
# listA size=1541
# listB size=827
# Intersection size=90
# Overlapping p-value=6.9e-31
# Jaccard Index=0.0

TDP43_KO_ALS_down.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.down,
                         peng.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(TDP43_KO_ALS_down.go.obj)
# GeneOverlap object:
# listA size=2159
# listB size=854
# Intersection size=185
# Overlapping p-value=2.1e-89
# Jaccard Index=0.1
```

#### Scatterplot

```{r}
peng.dge <- peng.res %>% dplyr::select(gene_name, log2FC.TDP43_KO = log2FoldChange, FDR.TDP43_KO = padj) %>% filter(gene_name %in% gene2ens$gene_name) 
barbar.dge <- barbar.res %>% dplyr::select(gene_name, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
peng_barbar.dge <- full_join(peng.dge, barbar.dge, by = c("gene_name"))

peng_barbar.dge <- peng_barbar.dge %>% mutate(overlap = case_when(FDR.TDP43_KO < 0.05 & FDR.A1 < 0.05 ~ "overlapping", # down in TDP43_KO
                                                        FDR.TDP43_KO < 0.05 ~ "TDP43_KO specific",
                                                        FDR.A1 < 0.05 ~ "A1 specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.TDP43_KO < 0.05 & log2FC.TDP43_KO > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in TDP43_KO
                                                                  FDR.TDP43_KO < 0.05 & log2FC.TDP43_KO < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.TDP43_KO < 0.05 & log2FC.TDP43_KO > 0 ~ "TDP43_KO specific up",
                                                                  FDR.TDP43_KO < 0.05 & log2FC.TDP43_KO < 0 ~ "TDP43_KO specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up",
                                                                  FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "TDP43_KO specific up", "TDP43_KO specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "TDP43_KO specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(peng_barbar.dge$overlap)
table(peng_barbar.dge$overlap_direction)
peng_barbar.dge

cor.test(x = peng_barbar.dge$log2FC.TDP43_KO, y = peng_barbar.dge$log2FC.A1) # 0.1088784 
t.test( x = peng_barbar.dge$log2FC.TDP43_KO, y = peng_barbar.dge$log2FC.A1, paired = TRUE )

A1_TDP43_KO_scatter <- ggscatter(peng_barbar.dge, x = "log2FC.TDP43_KO", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~TDP43~KO:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-5,5) + ylim(-10,11) +
    geom_smooth(data = peng_barbar.dge, aes(x = log2FC.TDP43_KO, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(peng_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go, log2FC.A1 < 9.5), aes(x = log2FC.TDP43_KO, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7, max.overlaps = 20) +  
  annotate(geom="text", x=3, y=11, label="TDP43-KO & A1 Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-10, label="TDP43-KO & A1 Down", color="dodgerblue3", size = 3.2)  + labs(subtitle = "TDP43-KO vs A1")
A1_TDP43_KO_scatter
```


#### GO overlap

```{r}
# TDP43 KO
peng_barbar.dge_overlap_genes <- peng_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("gene_name")
peng_barbar.dge_overlap_genes <- gost(peng_barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
peng_barbar.dge_overlap_gprofiles_filt <- peng_barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
peng_barbar.dge_overlap_gprofiles_filt
peng_barbar.dge_overlap_gprofiles_filt_dge.up <- peng_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping up")
peng_barbar.dge_overlap_gprofiles_filt_dge.down <- peng_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping down")
paste('"',unique(peng_barbar.dge_overlap_gprofiles_filt_dge.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
peng_barbar.dge.up_curated_list <- c(
"response to stress",
"MHC class I peptide loading complex",
"collagen-containing extracellular matrix",
"response to stimulus",
"Glycolysis",
"response to cytokine",
"HIF-1 signaling pathway",
"NADH metabolic process",
"protein binding",
"extracellular vesicle",
"response to chemical")
peng_barbar.dge_overlap_gprofiles_filt_dge.up <- peng_barbar.dge_overlap_gprofiles_filt_dge.up %>% filter(term_name %in% peng_barbar.dge.up_curated_list) %>% mutate(query = "upregulated")
peng_barbar.dge_overlap_gprofiles_filt_dge.up$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", peng_barbar.dge_overlap_gprofiles_filt_dge.up$term_name) %>% gsub(" loading complex", "", .)
paste('"',unique(peng_barbar.dge_overlap_gprofiles_filt_dge.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
peng_barbar.dge.up_curated_list <- c(
"cytoskeleton organization",
"synapse",
"cell junction organization",
"neuron development",
"gliogenesis",
"glial cell development",
"cell junction",
"plasma membrane",
"myelination",
"generation of neurons",
"nervous system development")
peng_barbar.dge_overlap_gprofiles_filt_dge.down <- peng_barbar.dge_overlap_gprofiles_filt_dge.down %>% filter(term_name %in% peng_barbar.dge.up_curated_list) %>% mutate(query = "downregulated")
# peng_barbar.dge_overlap_gprofiles_filt_dge.down$term_name <- gsub("Deactivation of the beta-catenin transactivating complex", "Deactivation beta-catenin complex", peng_barbar.dge_overlap_gprofiles_filt_dge.down$term_name)
peng_barbar.dge_overlap_gprofiles_filt_dge.up_down <- bind_rows(peng_barbar.dge_overlap_gprofiles_filt_dge.up, peng_barbar.dge_overlap_gprofiles_filt_dge.down) %>% 
  mutate(query = factor(query, levels = c("upregulated", "downregulated")))
peng_barbar.go_curated <- curated_profiles(gprofiles = peng_barbar.dge_overlap_gprofiles_filt_dge.up_down, colours = 2) + labs(subtitle = "A1 & TDP43 KO")
peng_barbar.go_curated

peng_barbar.dge_overlap_gprofiles_filt <- apply(peng_barbar.dge_overlap_gprofiles_filt,2,as.character)
write.csv(peng_barbar.dge_overlap_gprofiles_filt, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/peng_barbar_overlap_go_terms.csv")
```

#### GSEA glut/phag

```{r}
# GSEA violin plots: Phagocytosis, Glutamate uptke
peng.res_phagocytosis <-filter(peng.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  mutate(gene_set = "phagocytosis")
peng.res_glutamate_uptake <- filter(peng.res, gene_name %in% go.pathways.msigdb$GO_GLUTAMATERGIC_SYNAPSE) %>% mutate(gene_set = "glutamate uptake")
peng.res_gene_sets <- bind_rows(peng.res_phagocytosis, peng.res_glutamate_uptake) 

peng.gene_sets_sina_plot <- sinaplot.one.sample(data =  peng.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log[2]~Fold~Change~TDP43~KO:CTRL )) + xlab("")  + theme(axis.text=element_text(size=9)) + 
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "TDP43 KO Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "TDP43 KO Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2))  + labs(subtitle = "GSEA TDP43-KO")
peng.gene_sets_sina_plot
# group            .y.   group1 group2         n statistic        p p.signif y.position
# * <chr>            <chr> <chr>  <chr>      <int>     <dbl>    <dbl> <chr>         <dbl>
# 1 glutamate uptake lfc   1      null model   399     37153 0.305    ns                2
# 2 phagocytosis     lfc   1      null model   267     20699 0.000121 ***               2

# GSEA phagocytosis
peng.geneList <- peng.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(peng.geneList) <- peng.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
peng.geneList = sort(peng.geneList, decreasing = TRUE)
peng.phagocytosis_genes.list <- list("phagocytosis" = peng.res$gene_name[peng.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
peng.phagocytosis_fgseaRes <- fgsea(pathways = peng.phagocytosis_genes.list, stats = peng.geneList)
peng.phagocytosis_fgseaRes
#         pathway    pval    padj   log2err        ES     NES size                             leadingEdge
# 1: phagocytosis 0.26703 0.26703 0.1388051 0.2949358 1.05324  254 ALOX15,IGHV1-18,TNF,CD3G,TULP1,XKR9,...

# GSEA glutamate uptake
peng.glutamate.uptake_genes.list <- list("glutamate.uptake" = peng.res$gene_name[peng.res$gene_name %in% glutamate.uptake.go])
peng.glutamate.uptake_fgseaRes <- fgsea(pathways = peng.glutamate.uptake_genes.list, stats = peng.geneList)
peng.glutamate.uptake_fgseaRes
#             pathway pval padj    log2err        ES       NES size               leadingEdge
# 1: glutamate.uptake    1    1 0.06407038 0.1245761 0.4590739  397 FTCD,UROC1,ACTC1,NAGS,TAT
```

### Membralin with A1 & ALS

#### Venn

```{r}
# Overlap mem_ko with A1 barbar
barbar.res.dge.up <- barbar.res %>% filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_name) %>% unique 
barbar.res.dge.down <- barbar.res %>% filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_name) %>% unique 
jiang.res.dge.up <- jiang.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
jiang.res.dge.down <- jiang.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

jiang.A1.up_genes <- list("membralin-KO" = jiang.res.dge.up, "A1" = barbar.res.dge.up)
jiang.A1.down_genes <- list("membralin-KO" = jiang.res.dge.down, "A1" = barbar.res.dge.down)
jiang.A1_venn_up <- ggvenn(jiang.A1.up_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
jiang.A1_venn_down <- ggvenn(jiang.A1.down_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "dodgerblue3"))
jiang.A1_venn <- ggarrange(jiang.A1_venn_up, jiang.A1_venn_down, nrow = 2)
jiang.A1_venn

mem_ko_A1_up.go.obj <- newGeneOverlap(barbar.res.dge.up,
                         jiang.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(mem_ko_A1_up.go.obj)
# GeneOverlap object:
# listA size=4686
# listB size=844
# Intersection size=261
# Overlapping p-value=5.1e-88

mem_ko_A1_down.go.obj <- newGeneOverlap(barbar.res.dge.down,
                         jiang.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(mem_ko_A1_down.go.obj)
# GeneOverlap object:
# listA size=4461
# listB size=896
# Intersection size=276
# Overlapping p-value=1.2e-97
# Jaccard Index=0.1

# Define ALS & A1 genes:
patani_birger_tyzack_barbar.dge.up <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP > 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 > 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 > 0) & (FDR.A1 < 0.05 & log2FC.A1 > 0)) %>% pull(gene_name) %>% unique 
patani_birger_tyzack_barbar.dge.down <- patani_birger_tyzack_barbar.dge %>% filter((FDR.VCP < 0.05 & log2FC.VCP < 0) | (FDR.SOD1 < 0.05 & log2FC.SOD1 < 0)  | (FDR.C9orf72 < 0.05 & log2FC.C9orf72 < 0) & (FDR.A1 < 0.05 & log2FC.A1 < 0) ) %>% pull(gene_name) %>% unique 
jiang.res.dge.up <- jiang.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 
jiang.res.dge.down <- jiang.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% filter(gene_name %in% gene2ens$gene_name) %>% pull(gene_name) %>% unique 

jiang.ALS.up_genes <- list("TRAP" = jiang.res.dge.up, "ALS" = patani_birger_tyzack_barbar.dge.up)
jiang.ALS.down_genes <- list("TRAP" = jiang.res.dge.down, "ALS" = patani_birger_tyzack_barbar.dge.down)
jiang.ALS_venn_up <- ggvenn(jiang.ALS.up_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Up") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
jiang.ALS_venn_up
jiang.ALS_venn_down <- ggvenn(jiang.ALS.down_genes, fill_color = c("dodgerblue3", "firebrick2"),show_percentage = FALSE, stroke_size = 0.5, set_name_size = 3, text_size = 2.8) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5, size = 10, colour = "firebrick2"))
jiang.ALS_venn_down

mem_ko_ALS_up.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.up,
                         jiang.res.dge.up,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(mem_ko_ALS_up.go.obj)
# GeneOverlap object:
# listA size=1541
# listB size=844
# Intersection size=155
# Overlapping p-value=6.3e-85
# Jaccard Index=0.1

mem_ko_ALS_down.go.obj <- newGeneOverlap(patani_birger_tyzack_barbar.dge.down,
                         jiang.res.dge.down,
                         genome.size=nrow(distinct(gene2ens, gene_name)))
testGeneOverlap(mem_ko_ALS_down.go.obj)
# GeneOverlap object:
# listA size=2159
# listB size=896
# Intersection size=255
# Overlapping p-value=1e-154
# Jaccard Index=0.1
```

#### Scatterplot

```{r}
jiang.dge <- jiang.res %>% dplyr::select(gene_name, log2FC.mem_ko = log2FoldChange, FDR.mem_ko = padj) %>% filter(gene_name %in% gene2ens$gene_name) 
barbar.dge <- barbar.res %>% dplyr::select(gene_name, log2FC.A1 = log2FoldChange, FDR.A1 = padj)
jiang_barbar.dge <- full_join(jiang.dge, barbar.dge, by = c("gene_name"))

jiang_barbar.dge <- jiang_barbar.dge %>% mutate(overlap = case_when(FDR.mem_ko < 0.05 & FDR.A1 < 0.05 ~ "overlapping", # down in mem_ko
                                                        FDR.mem_ko < 0.05 ~ "mem_ko specific",
                                                        FDR.A1 < 0.05 ~ "A1 specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.mem_ko < 0.05 & log2FC.mem_ko > 0 & FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "overlapping up", # down in mem_ko
                                                                  FDR.mem_ko < 0.05 & log2FC.mem_ko < 0 & FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "overlapping down",
                                                                  FDR.mem_ko < 0.05 & log2FC.mem_ko > 0 ~ "mem_ko specific up",
                                                                  FDR.mem_ko < 0.05 & log2FC.mem_ko < 0 ~ "mem_ko specific down",
                                                                  FDR.A1 < 0.05 & log2FC.A1 > 0 ~ "A1 specific up",
                                                                  FDR.A1 < 0.05 & log2FC.A1 < 0 ~ "A1 specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "mem_ko specific up", "mem_ko specific down", "A1 specific up", "A1 specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "mem_ko specific", "A1 specific", "None"))) %>%
  arrange(overlap_direction)
table(jiang_barbar.dge$overlap)
table(jiang_barbar.dge$overlap_direction)
jiang_barbar.dge

cor.test(x = jiang_barbar.dge$log2FC.mem_ko, y = jiang_barbar.dge$log2FC.A1)
t.test( x = jiang_barbar.dge$log2FC.mem_ko, y = jiang_barbar.dge$log2FC.A1, paired = TRUE )

A1_mem_ko_scatter <- ggscatter(jiang_barbar.dge, x = "log2FC.mem_ko", y = "log2FC.A1", color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"), cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~membralin~KO:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   xlim(-5,5) + ylim(-10,11) +
    geom_smooth(data = jiang_barbar.dge, aes(x = log2FC.mem_ko, y = log2FC.A1), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(jiang_barbar.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go, log2FC.A1 < 9.5), aes(x = log2FC.mem_ko, y = log2FC.A1, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7, max.overlaps = 25) +  
  annotate(geom="text", x=3, y=11, label="mem-KO & A1 Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-10, label="mem-KO & A1 Down", color="dodgerblue3", size = 3.2)  + labs(subtitle = "membralin-KO vs A1")
A1_mem_ko_scatter
```


#### GO overlap

```{r}
# membralin KO
jiang_barbar.dge_overlap_genes <- jiang_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("gene_name")
jiang_barbar.dge_overlap_genes <- gost(jiang_barbar.dge_overlap_genes) # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
jiang_barbar.dge_overlap_gprofiles_filt <- jiang_barbar.dge_overlap_genes$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate( term_name = as.factor(term_name)) 
jiang_barbar.dge_overlap_gprofiles_filt
jiang_barbar.dge_overlap_gprofiles_filt_dge.up <- jiang_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping up")
jiang_barbar.dge_overlap_gprofiles_filt_dge.down <- jiang_barbar.dge_overlap_gprofiles_filt %>% filter(query == "overlapping down")
paste('"',unique(jiang_barbar.dge_overlap_gprofiles_filt_dge.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
jiang_barbar.dge.up_curated_list <- c("collagen-containing extracellular matrix",
"inflammatory response",
"cell communication",
"signaling",
"cell activation",
"response to virus",
"response to cytokine",
"response to stress",
"Interferon Signaling",
"immune response")
jiang_barbar.dge_overlap_gprofiles_filt_dge.up <- jiang_barbar.dge_overlap_gprofiles_filt_dge.up %>% filter(term_name %in% jiang_barbar.dge.up_curated_list) %>% mutate(query = "upregulated")
jiang_barbar.dge_overlap_gprofiles_filt_dge.up$term_name <- gsub("collagen-containing extracellular matrix", "collagen extracellular matrix", jiang_barbar.dge_overlap_gprofiles_filt_dge.up$term_name)
paste('"',unique(jiang_barbar.dge_overlap_gprofiles_filt_dge.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
jiang_barbar.dge.up_curated_list <- c(
"glutamatergic synapse",
"cytoskeleton organization",
"synaptic signaling",
"neurogenesis",
"neuron development",
"neuron differentiation",
"nervous system development",
"plasma membrane region",
"cell junction",
"neuron projection",
"synapse")
jiang_barbar.dge_overlap_gprofiles_filt_dge.down <- jiang_barbar.dge_overlap_gprofiles_filt_dge.down %>% filter(term_name %in% jiang_barbar.dge.up_curated_list) %>% mutate(query = "downregulated")
jiang_barbar.dge_overlap_gprofiles_filt_dge.up_down <- bind_rows(jiang_barbar.dge_overlap_gprofiles_filt_dge.up, jiang_barbar.dge_overlap_gprofiles_filt_dge.down) %>% 
  mutate(query = factor(query, levels = c("upregulated", "downregulated")))
jiang_barbar.go_curated <- curated_profiles(gprofiles = jiang_barbar.dge_overlap_gprofiles_filt_dge.up_down, colours = 2) + labs(subtitle = "A1 & membralin KO")
jiang_barbar.go_curated

jiang_barbar.dge_overlap_gprofiles_filt <- apply(jiang_barbar.dge_overlap_gprofiles_filt,2,as.character)
write.csv(jiang_barbar.dge_overlap_gprofiles_filt, "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/jiang_barbar_overlap_go_terms.csv")
```

#### GSEA glut/phag

```{r}
# GSEA violin plots: Phagocytosis, Glutamate uptke
jiang.res_phagocytosis <-filter(jiang.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>%  mutate(gene_set = "phagocytosis")
jiang.res_glutamate_uptake <- filter(jiang.res, gene_name %in% go.pathways.msigdb$GO_GLUTAMATERGIC_SYNAPSE) %>% mutate(gene_set = "glutamate uptake")
jiang.res_gene_sets <- bind_rows(jiang.res_phagocytosis, jiang.res_glutamate_uptake) 

jiang.gene_sets_sina_plot <- sinaplot.one.sample(data =  jiang.res_gene_sets, group = "gene_set", continuous = "log2FoldChange", labels = "gene_name", colours = 2,  label_number = 0, stat.y.position = 2, width = 0.1, flip = "no", stats.test = "t-test") + ylab(label = expression( log[2]~Fold~Change~mem~KO:CTRL )) + xlab("")  + theme(axis.text=element_text(size=9)) +
  geom_segment(aes(x = 2.5, xend = 2.5, y= 0.1, yend= 2), arrow=arrow(length=unit(0.2,"cm")), color = "black") +  geom_segment(aes(x = 2.5, xend = 2.5, y= -0.1, yend= -2),arrow=arrow(length=unit(0.2,"cm")), color = "black") +
  annotate("text", x = 2.35, y = 1, label = "membralin KO Up", colour = "black", size = 2.9, angle = 90) +  annotate("text", x =2.35, y = -1, label = "membralin KO Down", colour = "black", size = 2.9, angle = 90) + ylim(c(-2,2))  + labs(subtitle = "GSEA membralin-KO")
jiang.gene_sets_sina_plot
#   group      .y.   group1 group2      n statistic        p p.signif y.position
# * <chr>      <chr> <chr>  <chr>   <int>     <dbl>    <dbl> <chr>         <dbl>
# 1 glutamate? lfc   1      null m?   399     27395  1.21e-7 ****              2
# 2 phagocyto? lfc   1      null m?   267     20488  2.12e-6 ****              2

# GSEA phagocytosis
jiang.geneList <- jiang.res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(jiang.geneList) <- jiang.res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
jiang.geneList = sort(jiang.geneList, decreasing = TRUE)
jiang.phagocytosis_genes.list <- list("phagocytosis" = jiang.res$gene_name[jiang.res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
jiang.phagocytosis_fgseaRes <- fgsea(pathways = jiang.phagocytosis_genes.list, stats = jiang.geneList)
jiang.phagocytosis_fgseaRes
#         pathway         pval         padj   log2err        ES      NES size
# 1: phagocytosis 3.370399e-05 3.370399e-05 0.5573322 0.4375708 1.662556  246

# GSEA glutamate uptake
jiang.glutamate.uptake_genes.list <- list("glutamate.uptake" = jiang.res$gene_name[jiang.res$gene_name %in% go.pathways.msigdb$GO_GLUTAMATERGIC_SYNAPSE])
jiang.glutamate.uptake_fgseaRes <- fgsea(pathways = jiang.glutamate.uptake_genes.list, stats = jiang.geneList)
jiang.glutamate.uptake_fgseaRes
#             pathway pval padj    log2err        ES       NES size
# 1: glutamate.uptake    1    1 0.06130261 -0.185918 -0.762217  397
```



### Merge

```{r}
fig5.merged <- ggarrange(
  ggarrange(A1_SOD1_TRAP_scatter, cleveland.A1_venn, cleveland_barbar.go_curated, ncol = 3, labels = c("A", "B", "C"), widths = c(1.2,0.6,1)), 
  ggarrange(A1_TDP43_KO_scatter, peng.A1_venn, peng_barbar.go_curated, ncol = 3, labels = c("D", "E", "F"), widths = c(1.1,0.6,1)), 
  ggarrange(A1_mem_ko_scatter, jiang.A1_venn, jiang_barbar.go_curated, ncol = 3, labels = c("G", "H", "I"), widths = c(1.1,0.6,1)), 
  nrow = 3, heights = c(1,1,1,1))
fig5.merged
ggsave(fig5.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/fig5.merged.png", height = 14, width = 10)


ggarrange(cleveland.gene_sets_sina_plot, peng.gene_sets_sina_plot,jiang.gene_sets_sina_plot, ncol = 3)

```

## Sup Fig 1

#### PROGENy and DoRothEA

```{r}
# PROGENy
cleveland.res.matrix <- cleveland.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
cleveland.PathwayActivity_zscore <- progeny(cleveland.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(cleveland.PathwayActivity_zscore) <- "NES"
cleveland.PathwayActivity_zscore_df <- as.data.frame(cleveland.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
peng.res.matrix <- peng.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
peng.PathwayActivity_zscore <- progeny(peng.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(peng.PathwayActivity_zscore) <- "NES"
peng.PathwayActivity_zscore_df <- as.data.frame(peng.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
jiang.res.matrix <- jiang.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
jiang.PathwayActivity_zscore <- progeny(jiang.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(jiang.PathwayActivity_zscore) <- "NES"
jiang.PathwayActivity_zscore_df <- as.data.frame(jiang.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))
barbar.res.matrix <- barbar.res %>% dplyr::select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
barbar.A1.PathwayActivity_zscore <- progeny(barbar.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t() # enrichment analysis of each pathway activity
colnames(barbar.A1.PathwayActivity_zscore) <- "NES"
barbar.A1.PathwayActivity_zscore_df <- as.data.frame(barbar.A1.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% dplyr::arrange(NES) %>%  dplyr::mutate(Pathway = factor(Pathway))

barbar.cleveland.peng.jiang.PathwayActivity_zscore_df <- mutate(barbar.A1.PathwayActivity_zscore_df, dataset = "A1") %>% bind_rows(mutate(peng.PathwayActivity_zscore_df, dataset = "TDP43 KO")) %>% bind_rows(mutate(cleveland.PathwayActivity_zscore_df, dataset = "SOD1 TRAP"))  %>% bind_rows(mutate(jiang.PathwayActivity_zscore_df, dataset = "membralin KO"))
barbar.cleveland.peng.jiang.PathwayActivity_zscore_df.meanNES <- barbar.cleveland.peng.jiang.PathwayActivity_zscore_df %>% group_by(Pathway) %>% summarise(mean_NES = mean(NES)) %>% arrange(mean_NES)
barbar.cleveland.peng.jiang.PathwayActivity_zscore_df$Pathway <- factor(barbar.cleveland.peng.jiang.PathwayActivity_zscore_df$Pathway, levels = barbar.cleveland.peng.jiang.PathwayActivity_zscore_df.meanNES$Pathway)

barbar.cleveland.peng.jiang.progeny.pathwayActivity <- ggplot(barbar.cleveland.peng.jiang.PathwayActivity_zscore_df, aes(x=Pathway, y = NES, fill = dataset)) + theme_classic() +
  geom_bar(stat="identity", position=position_dodge()) + theme(legend.position = "top", legend.title = element_blank(), legend.spacing.x = unit(0.5, 'cm'), axis.title = element_text(size = 12),
        axis.text.x = element_text(angle = 45, hjust = 1, size = 8), axis.text.y = element_text(size =8)) +
  geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") + scale_fill_manual( values = c("firebrick1", "dodgerblue1", "forestgreen", "gold2")) + xlab(expression())  + ylab(expression(Normalised~enrichment~score)) + labs(subtitle = "Signalling Pathway activity")
barbar.cleveland.peng.jiang.progeny.pathwayActivity

# DoRothEA
data(dorothea_hs, package = "dorothea")
regulons <- dorothea_hs %>%  dplyr::filter(confidence %in% c("A", "B","C"))
barbar.tf_activities_stat <- dorothea::run_viper(barbar.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
barbar.tf_activities_stat_top25 <- barbar.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
barbar.tf_activities_stat.df <- barbar.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(log2FC.barbar = "log2FoldChange")
cleveland.tf_activities_stat <- dorothea::run_viper(cleveland.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
cleveland.tf_activities_stat_top25 <- cleveland.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
cleveland.tf_activities_stat.df <- cleveland.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.cleveland = "log2FoldChange")
peng.tf_activities_stat <- dorothea::run_viper(peng.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
peng.tf_activities_stat_top25 <- peng.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
peng.tf_activities_stat.df <- peng.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.peng = "log2FoldChange")
jiang.tf_activities_stat <- dorothea::run_viper(jiang.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE))
jiang.tf_activities_stat_top25 <- jiang.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name") %>% dplyr::rename(NES = "log2FoldChange") %>%  dplyr::top_n(25, wt = abs(NES)) %>%  dplyr::arrange(NES) %>% dplyr::mutate(gene_name = factor(gene_name))
jiang.tf_activities_stat.df <- jiang.tf_activities_stat %>% as.data.frame() %>% rownames_to_column(var = "gene_name")  %>% dplyr::rename(log2FC.jiang= "log2FoldChange")

barbar.cleveland.tf_activities <- cleveland.tf_activities_stat.df %>% left_join(barbar.tf_activities_stat.df, by = "gene_name")
barbar.peng.tf_activities <- peng.tf_activities_stat.df %>% left_join(barbar.tf_activities_stat.df, by = "gene_name")
barbar.jiang.tf_activities <- jiang.tf_activities_stat.df %>% left_join(barbar.tf_activities_stat.df, by = "gene_name")

# barbar vs cleveland
cor.test(x = barbar.cleveland.tf_activities$log2FC.cleveland, y = barbar.cleveland.tf_activities$log2FC.barbar)
t.test( x = barbar.cleveland.tf_activities$log2FC.cleveland, y = barbar.cleveland.tf_activities$log2FC.barbar, paired = TRUE )
barbar_cleveland_TFactivities.scatter <- ggscatter(barbar.cleveland.tf_activities, x = "log2FC.cleveland", y = "log2FC.barbar", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson",
          xlab = expression( Log[2]~Fold~Change~SOD1~TRAP:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = barbar.cleveland.tf_activities, aes(x = log2FC.cleveland, y = log2FC.barbar), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(barbar.cleveland.tf_activities, log2FC.cleveland > 1, log2FC.barbar > 2), aes(x = log2FC.cleveland, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
      geom_text_repel(data = filter(barbar.cleveland.tf_activities, log2FC.cleveland < -1, log2FC.barbar < -2), aes(x = log2FC.cleveland, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=3, y=15, label="SOD1-TRAP & A1 Up", color="firebrick2", size = 3)  + annotate(geom="text", x=-1, y=-5, label="SOD1-TRAP & A1 Down", color="dodgerblue3", size = 2.8) + labs(subtitle = "SOD1-TRAP TF activity")
barbar_cleveland_TFactivities.scatter

# barbar vs peng
cor.test(x = barbar.peng.tf_activities$log2FC.peng, y = barbar.peng.tf_activities$log2FC.barbar) # 0.22
t.test( x = barbar.peng.tf_activities$log2FC.peng, y = barbar.peng.tf_activities$log2FC.barbar, paired = TRUE )
barbar_peng_TFactivities.scatter <- ggscatter(barbar.peng.tf_activities, x = "log2FC.peng", y = "log2FC.barbar", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson", xlab = expression( Log[2]~Fold~Change~TDP43~KO:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = barbar.peng.tf_activities, aes(x = log2FC.peng, y = log2FC.barbar), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(barbar.peng.tf_activities, log2FC.peng > 1, log2FC.barbar > 2), aes(x = log2FC.peng, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
      geom_text_repel(data = filter(barbar.peng.tf_activities, log2FC.peng < -1, log2FC.barbar < -2), aes(x = log2FC.peng, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=3, y=15, label="TDP43-KO & A1 Up", color="firebrick2", size = 3)  + annotate(geom="text", x=-1, y=-5, label="TDP43-KO & A1 Down", color="dodgerblue3", size = 2.8) + labs(subtitle = "TDP43-KO TF activity")
barbar_peng_TFactivities.scatter

# barbar vs jiang
cor.test(x = barbar.jiang.tf_activities$log2FC.jiang, y = barbar.jiang.tf_activities$log2FC.barbar) # 0.22
t.test( x = barbar.jiang.tf_activities$log2FC.jiang, y = barbar.jiang.tf_activities$log2FC.barbar, paired = TRUE )
barbar_jiang_TFactivities.scatter <- ggscatter(barbar.jiang.tf_activities, x = "log2FC.jiang", y = "log2FC.barbar", #, color = "overlap_direction", palette = c("firebrick2", "dodgerblue3", "grey", "grey", "grey", "grey", "grey"),
          cor.coef = TRUE, cor.method = "pearson", xlab = expression( Log[2]~Fold~Change~membralin~KO:CTRL), ylab = expression(Log[2]~Fold~Change~A1:A0), size = 0.5, cor.coef.size = 3) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +   #xlim(-10,10) + ylim(-10,10) +
    geom_smooth(data = barbar.jiang.tf_activities, aes(x = log2FC.jiang, y = log2FC.barbar), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 1) + theme(legend.position = "none") +
    geom_text_repel(data = filter(barbar.jiang.tf_activities, log2FC.jiang > 1, log2FC.barbar > 2), aes(x = log2FC.jiang, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
      geom_text_repel(data = filter(barbar.jiang.tf_activities, log2FC.jiang < -1, log2FC.barbar < -2), aes(x = log2FC.jiang, y = log2FC.barbar, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.7) +
  annotate(geom="text", x=4, y=15, label="mem-KO & A1 Up", color="firebrick2", size = 3)  + annotate(geom="text", x=-1, y=-5, label="mem-KO & A1 Down", color="dodgerblue3", size = 2.8) + labs(subtitle = "membralin-KO TF activity")
barbar_jiang_TFactivities.scatter

```

### Merge

```{r}
figS2.merged <- ggarrange(
  ggarrange(barbar.gene_sets_sina_plot, cleveland.gene_sets_sina_plot,peng.gene_sets_sina_plot,jiang.gene_sets_sina_plot, ncol = 4, labels = c("A", "B", "C", "D"), widths = c(1,1,1,1)), 
  barbar.cleveland.peng.jiang.progeny.pathwayActivity, 
  ggarrange(barbar_cleveland_TFactivities.scatter, barbar_peng_TFactivities.scatter, barbar_jiang_TFactivities.scatter, ncol = 3, labels = c("F", "G", "H"), widths = c(1,1,1)), 
  nrow = 3, heights = c(1,1.1), labels = c("", "E", ""))
figS2.merged
ggsave(figS2.merged, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/figS2.merged.png", height = 12, width = 12)
```

## GO terms Table

```{r}
patani.barbar.dge_overlap_genes <- patani_barbar.dge %>% filter(overlap_direction == "overlapping up" | overlap_direction == "overlapping down" ) %>% split(.$overlap_direction) %>%  map("ensID") %>% gost()
patani.barbar.dge_all_go_terms <- filter(patani.barbar.dge_overlap_genes$result, str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) )  %>% mutate(query = case_when(query == "overlapping up" ~ "VCP & A1 up", query == "overlapping down" ~ "VCP & A1 down"))
tyzack.barbar.dge_all_go_terms <- tyzack_barbar.dge_overlap_gprofiles_filt %>% mutate(query = case_when(query == "overlapping up" ~ "SOD1 & A1 up", query == "overlapping down" ~ "SOD1 & A1 down"))
birger.barbar.dge_all_go_terms <- birger_barbar.dge_overlap_gprofiles_filt %>% mutate(query = case_when(query == "overlapping up" ~ "C9orf72 & A1 up", query == "overlapping down" ~ "C9orf72 & A1 down"))
cleveland.barbar.dge_all_go_terms <- cleveland_barbar.dge_overlap_gprofiles_filt %>% mutate(query = case_when(query == "overlapping up" ~ "SOD1 TRAP & A1 up", query == "overlapping down" ~ "SOD1 TRAP & A1 down"))
peng.barbar.dge_all_go_terms <- peng_barbar.dge_overlap_gprofiles_filt %>% mutate(query = case_when(query == "overlapping up" ~ "TDP43 KO & A1 up", query == "overlapping down" ~ "TDP43 KO & A1 down"))
jiang.barbar.dge_all_go_terms <- jiang_barbar.dge_overlap_gprofiles_filt %>% mutate(query = case_when(query == "overlapping up" ~ "membralin KO & A1 up", query == "overlapping down" ~ "membralin KO & A1 down"))

all_go_terms.all_datasets <- bind_rows(patani.barbar.dge_all_go_terms, tyzack.barbar.dge_all_go_terms, birger.barbar.dge_all_go_terms, cleveland.barbar.dge_all_go_terms, peng.barbar.dge_all_go_terms, jiang.barbar.dge_all_go_terms) %>% arrange(query) %>% dplyr::select(query, term_name, p_value, everything(), -significant, -precision, -recall)
all_go_terms.all_datasets <- apply(all_go_terms.all_datasets,2,as.character)
write.csv(all_go_terms.all_datasets, file = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/TableS5.all_go_terms.all_datasets.csv", row.names = FALSE)
```

## Additional analyses

#### CARNIVAL

Run as a distinct script as run_carnival leads to Bus Error.
/camp/lab/luscomben/home/users/ziffo/projects/ac-mn-comparison/scripts/CARNIVAL.R

https://github.com/saezlab/transcriptutorial/blob/master/scripts/05_network_reconstruction_with_CARNIVAL.md
https://github.com/saezlab/transcriptutorial/blob/master/scripts/06_analysis_CARNIVAL_results.md
https://saezlab.github.io/CARNIVAL/ 

CARNIVAL (CAusal Reasoning for Network identification using Integer VALue programming) identifies upstream regulatory signalling pathways from downstream gene expression data. CARNIVAL identifies interactions from a prior knowledge network, that represents potential regulated pathways linking known or potential targets of perturbation, to active transcription factors derived from RNAseq.

Use TF activity from DoRothEA and prior knowledge network from Omnipath. PROGENy scores also help CARNIVAL infer representative genes for each pathway.

### Network reconstruction 

```{r}
library(progeny)
library(dorothea)
library(CARNIVAL)
library(OmnipathR)
library(visNetwork)

## Support functions are in irfinder_functions.R

# get network scaffold from Omnipath (signed & directed interactions)
omniR <- import_Omnipath_Interactions() 
omnipath_sd <- omniR %>% dplyr::filter(consensus_direction == 1 & (consensus_stimulation == 1 | consensus_inhibition == 1))
# changing 0/1 criteria in consensus_stimulation/inhibition to -1/1
omnipath_sd$consensus_stimulation[which( omnipath_sd$consensus_stimulation == 0)] = -1
omnipath_sd$consensus_inhibition[which( omnipath_sd$consensus_inhibition == 1)] = -1
omnipath_sd$consensus_inhibition[which( omnipath_sd$consensus_inhibition == 0)] = 1
# check consistency on consensus sign and select only those in a SIF format
sif <- omnipath_sd[,c('source_genesymbol', 'consensus_stimulation', 'consensus_inhibition', 'target_genesymbol')] %>%
      dplyr::filter(consensus_stimulation==consensus_inhibition) %>%
      unique.data.frame()
sif$consensus_stimulation <- NULL
colnames(sif) <- c('source', 'interaction', 'target')
# remove complexes
sif$source <- gsub(":", "_", sif$source)
sif$target <- gsub(":", "_", sif$target)
write_tsv(sif, "/camp/home/ziffo/home/genomes/transcription-factors/omnipath_carnival.tsv")


############# Astrocytes ############# 
print("Analysing Astrocytes")

ac_vcp_vs_ctrl.tf_activities_CARNIVALinput <- read_csv("/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/ac_vcp_vs_ctrl.TFActivity_CARNIVALinput.csv")
ac_vcp_vs_ctrl.PathwayActivity_CARNIVALinput <- read_csv("/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/ac_vcp_vs_ctrl.PROGENy_PathwayActivity_CARNIVALinput.csv")

# Format DoRothEA and PROGENy output for CARNIVAL
ac_vcp_vs_ctrl.tf_activities_carnival <- data.frame(ac_vcp_vs_ctrl.tf_activities_CARNIVALinput, stringsAsFactors = F)
rownames(ac_vcp_vs_ctrl.tf_activities_carnival) <- ac_vcp_vs_ctrl.tf_activities_CARNIVALinput$TF
ac_vcp_vs_ctrl.tf_activities_carnival$TF <- NULL
ac_vcp_vs_ctrl.tfList = generateTFList(ac_vcp_vs_ctrl.tf_activities_carnival, top=50, access_idx = 1)

# progeny for CARNIVAL
load(file = system.file("progenyMembers.RData",package="CARNIVAL"))
ac_vcp_vs_ctrl.PathwayActivity_carnival <- data.frame(ac_vcp_vs_ctrl.PathwayActivity_CARNIVALinput, stringsAsFactors = F)
rownames(ac_vcp_vs_ctrl.PathwayActivity_carnival) <- ac_vcp_vs_ctrl.PathwayActivity_carnival$Pathway
ac_vcp_vs_ctrl.PathwayActivity_carnival$Pathway <- NULL
ac_vcp_vs_ctrl.progenylist = assignPROGENyScores(progeny = t(ac_vcp_vs_ctrl.PathwayActivity_carnival), 
                                                 progenyMembers = progenyMembers, 
                                                 id = "gene", 
                                                 access_idx = 1)

# Find causal link between TF activity and pertubed nodes
# get initial nodes
iniMTX = base::setdiff(sif$source, sif$target)
iniciators = base::data.frame(base::matrix(data = NaN, nrow = 1, ncol = length(iniMTX)), stringsAsFactors = F)
colnames(iniciators) = iniMTX

# run carnival
print("running CARNIVAL")

ac_vcp_vs_ctrl.carnival_result = runCARNIVAL( inputObj= iniciators, # pertubed nodes
                                              measObj = ac_vcp_vs_ctrl.tfList$log2FoldChange, # TF activities from DoRothEA
                                              netObj = sif, # network map TF & pertubed nodes
                                              weightObj = ac_vcp_vs_ctrl.progenylist$score, # optional but helps find optimal solution faster
                                              timelimit=7200,
                                              mipGAP=0,
                                              poolrelGAP=0 )
                                              
# Visualise the network
print("Visualising CARNIVAL")

ac_vcp_vs_ctrl.carnival_result$weightedSIF <- data.frame(ac_vcp_vs_ctrl.carnival_result$weightedSIF, stringsAsFactors = F)
ac_vcp_vs_ctrl.carnival_result$weightedSIF$Sign <- as.numeric(ac_vcp_vs_ctrl.carnival_result$weightedSIF$Sign)
ac_vcp_vs_ctrl.carnival_result$weightedSIF$Weight <- as.numeric(ac_vcp_vs_ctrl.carnival_result$weightedSIF$Weight)
ac_vcp_vs_ctrl.carnival_result$nodesAttributes <- data.frame(ac_vcp_vs_ctrl.carnival_result$nodesAttributes, stringsAsFactors = F)
ac_vcp_vs_ctrl.carnival_result$nodesAttributes$ZeroAct <- as.numeric(ac_vcp_vs_ctrl.carnival_result$nodesAttributes$ZeroAct)
ac_vcp_vs_ctrl.carnival_result$nodesAttributes$UpAct <- as.numeric(ac_vcp_vs_ctrl.carnival_result$nodesAttributes$UpAct)
ac_vcp_vs_ctrl.carnival_result$nodesAttributes$DownAct <- as.numeric(ac_vcp_vs_ctrl.carnival_result$nodesAttributes$DownAct)
ac_vcp_vs_ctrl.carnival_result$nodesAttributes$AvgAct <- as.numeric(ac_vcp_vs_ctrl.carnival_result$nodesAttributes$AvgAct)

saveRDS(ac_vcp_vs_ctrl.carnival_result,"/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/ac_vcp_vs_ctrl.carnival_result.rds")

# visualization
ac_vcp_vs_ctrl.visNet = carnival_visNet(evis = ac_vcp_vs_ctrl.carnival_result$weightedSIF, nvis = ac_vcp_vs_ctrl.carnival_result$nodesAttributes)
visSave(ac_vcp_vs_ctrl.visNet, file = paste0('/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/ac_vcp_vs_ctrl.carnival_visNetwork.html'), selfcontained = TRUE) #visNet

############# Motor neurons ############# 
print("Analysing motor neurons")
mn_vcp_vs_ctrl.tf_activities_CARNIVALinput <- read_csv("/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/mn_vcp_vs_ctrl.TFActivity_CARNIVALinput.csv")
mn_vcp_vs_ctrl.PathwayActivity_CARNIVALinput <- read_csv("/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/mn_vcp_vs_ctrl.PROGENy_PathwayActivity_CARNIVALinput.csv")

# Format DoRothEA and PROGENy output for CARNIVAL
mn_vcp_vs_ctrl.tf_activities_carnival <- data.frame(mn_vcp_vs_ctrl.tf_activities_CARNIVALinput, stringsAsFactors = F)
rownames(mn_vcp_vs_ctrl.tf_activities_carnival) <- mn_vcp_vs_ctrl.tf_activities_CARNIVALinput$TF
mn_vcp_vs_ctrl.tf_activities_carnival$TF <- NULL
mn_vcp_vs_ctrl.tfList = generateTFList(mn_vcp_vs_ctrl.tf_activities_carnival, top=50, access_idx = 1)

# progeny for CARNIVAL
load(file = system.file("progenyMembers.RData",package="CARNIVAL"))
mn_vcp_vs_ctrl.PathwayActivity_carnival <- data.frame(mn_vcp_vs_ctrl.PathwayActivity_CARNIVALinput, stringsAsFactors = F)
rownames(mn_vcp_vs_ctrl.PathwayActivity_carnival) <- mn_vcp_vs_ctrl.PathwayActivity_carnival$Pathway
mn_vcp_vs_ctrl.PathwayActivity_carnival$Pathway <- NULL
mn_vcp_vs_ctrl.progenylist = assignPROGENyScores(progeny = t(mn_vcp_vs_ctrl.PathwayActivity_carnival), 
                                                 progenyMembers = progenyMembers, 
                                                 id = "gene", 
                                                 access_idx = 1)

# Find causal link between TF activity and pertubed nodes
# get initial nodes
iniMTX = base::setdiff(sif$source, sif$target)
iniciators = base::data.frame(base::matrix(data = NaN, nrow = 1, ncol = length(iniMTX)), stringsAsFactors = F)
colnames(iniciators) = iniMTX

# run carnival
print("Running CARNIVAL")

mn_vcp_vs_ctrl.carnival_result = runCARNIVAL( inputObj= iniciators, # pertubed nodes
                                              measObj = mn_vcp_vs_ctrl.tfList$log2FoldChange, # TF activities from DoRothEA
                                              netObj = sif, # network map TF & pertubed nodes
                                              weightObj = mn_vcp_vs_ctrl.progenylist$score, # optional but helps find optimal solution faster
                                              timelimit=7200,
                                              mipGAP=0,
                                              poolrelGAP=0 )
# Visualise the network
print("Visualising CARNIVAL")

mn_vcp_vs_ctrl.carnival_result$weightedSIF <- data.frame(mn_vcp_vs_ctrl.carnival_result$weightedSIF, stringsAsFactors = F)
mn_vcp_vs_ctrl.carnival_result$weightedSIF$Sign <- as.numeric(mn_vcp_vs_ctrl.carnival_result$weightedSIF$Sign)
mn_vcp_vs_ctrl.carnival_result$weightedSIF$Weight <- as.numeric(mn_vcp_vs_ctrl.carnival_result$weightedSIF$Weight)
mn_vcp_vs_ctrl.carnival_result$nodesAttributes <- data.frame(mn_vcp_vs_ctrl.carnival_result$nodesAttributes, stringsAsFactors = F)
mn_vcp_vs_ctrl.carnival_result$nodesAttributes$ZeroAct <- as.numeric(mn_vcp_vs_ctrl.carnival_result$nodesAttributes$ZeroAct)
mn_vcp_vs_ctrl.carnival_result$nodesAttributes$UpAct <- as.numeric(mn_vcp_vs_ctrl.carnival_result$nodesAttributes$UpAct)
mn_vcp_vs_ctrl.carnival_result$nodesAttributes$DownAct <- as.numeric(mn_vcp_vs_ctrl.carnival_result$nodesAttributes$DownAct)
mn_vcp_vs_ctrl.carnival_result$nodesAttributes$AvgAct <- as.numeric(mn_vcp_vs_ctrl.carnival_result$nodesAttributes$AvgAct)

saveRDS(mn_vcp_vs_ctrl.carnival_result,"/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/mn_vcp_vs_ctrl.carnival_result.rds")

# visualization
mn_vcp_vs_ctrl.visNet = carnival_visNet(evis = mn_vcp_vs_ctrl.carnival_result$weightedSIF, nvis = mn_vcp_vs_ctrl.carnival_result$nodesAttributes)
visSave(mn_vcp_vs_ctrl.visNet, file = paste0('/camp/home/ziffo/home/projects/ac-mn-comparison/expression/deseq2/mn_vcp_vs_ctrl.carnival_visNetwork.html'), selfcontained = TRUE) #visNet
```

```{r}


```

Use the approach that was used in Doaa's manuscript to see whether there are any master regulators common to both VCP and A1. Also if there are any differences in splicing of these. I think above youve already seen 15 genes with decreased IR and increased expression in both A1 and VCP.

Used two approaches:
1. VIPER analysis of DoRothEA database https://bioconductor.org/packages/release/bioc/vignettes/viper/inst/doc/viper.pdf
DoRothEA estimates human transcription factors (TF) activities.
PROGENy infers human pathway activity.

2. TRRUST network analysis https://www.grnpedia.org/trrust/
TRRUST is a database of human transcriptional regulatory networks - 800 human TFs & 8,444 TF-target regulatory relationships.
Visualised with RedeR https://www.bioconductor.org/packages/release/bioc/vignettes/RedeR/inst/doc/RedeR.html

The ConservedFootprints package integrates both these together https://github.com/saezlab/ConservedFootprints/blob/master/analyses/dorothea_benchmark/dorothea_benchmark_pipeline.Rmd

https://bioconductor.org/packages/release/data/experiment/vignettes/dorothea/inst/doc/bulk_vignette.html
https://bioconductor.org/packages/release/bioc/vignettes/viper/inst/doc/viper.pdf
https://saezlab.github.io/dorothea/reference/run_viper.html
http://bioconductor.org/packages/release/data/experiment/vignettes/dorothea/inst/doc/bulk_vignette.html
https://github.com/saezlab/dorothea
https://github.com/saezlab/ConservedFootprints/tree/3cee21853a90f78bd13b6eebcb5538fc4c129cab
https://www.grnpedia.org/trrust/

```{r}
# http://127.0.0.1:21881/library/dorothea/doc/bulk_vignette.html
library(tidyverse)
library(dplyr)
library(DESeq2)
library(viper)
library(dorothea) # needs R version 4.0.0 & Bioconducter version 3.12 (not available on CAMP)
library(pheatmap)
library(rstatix)
gene2ens <- readRDS("/Volumes/lab-luscomben/home/users/ziffo/genomes/ensembl/Homo_sapiens.GRCh38.99.gene2ens.RDS") %>% unique
data(dorothea_hs, package = "dorothea") # import human dorothea regulon gene sets (curated TF & transcriptional targets)
regulons = dorothea_hs %>%  filter(confidence %in% c("A", "B")) # restrict dorothea to levels A & B (highest confidence only)

### run_viper on VCP & A1 merged VSD object
# create gene expression matrix - merge VSD (normalised counts) of VCP & A1 into a single matrix - columns samples, rows genes.
vsd <- readRDS("/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/vsd_VCP_A1_together.RDS")
rownames(vsd) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(vsd)] # convert ensID to gene_name as viper needs HGNC/MGI symbols

# viper wrapper in dorothea package - takes 20 mins. https://saezlab.github.io/dorothea/reference/run_viper.html
tf_activities <- run_viper(assay(vsd), regulons, options =  list(method = "scale", minsize = 25, eset.filter = FALSE, cores = 4)) # options passed to viper https://rdrr.io/bioc/viper/man/viper.html
tf_activities # matrix of normalised enirchment score for each TF across all samples
saveRDS(tf_activities, "/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/dorothea_hs.tf_activities.RDS")

# calculate mean NES per group
tf_activities.mean <- tf_activities %>% as_tibble(rownames = "TF") %>% mutate(CTRL = (ctrl1 + ctrl2 + ac_bar_a0m + ac_bar_a0f + ac_bar_a0u)/5,
                                                                                   VCP = (R191Q + R155C)/2,
                                                                                   A1 = (ac_bar_a1m + ac_bar_a1f + ac_bar_a1u)/3) %>% 
  arrange((VCP+A1)) %>% dplyr::select(TF, CTRL, VCP, A1)

# calculate p value from t-test within each condition
tf_activities.long <- tf_activities %>% as_tibble(rownames = "TF") %>% pivot_longer(!TF, names_to = "sample", values_to = "nes") %>% 
  mutate(group = case_when(grepl("ctrl", sample) ~ "CTRL", grepl("ac_bar_a0", sample) ~ "CTRL", grepl("R1", sample) ~ "VCP", grepl("ac_bar_a1", sample) ~ "A1"))

tf_activities.long <- tf_activities.long %>% mutate(TF = factor(tf_activities.long$TF, levels = tf_activities.mean$TF))

tf_activities_stat.test <- tf_activities.long %>%
  group_by(TF) %>%
  t_test(nes ~ group)  %>% # adjust_pvalue(method = "bonferroni") %>%
  add_significance()
tf_activities_stat.test

# heatmap per condition
tf_activities.mean.mat <- make_matrix(select(tf_activities.mean,-TF),pull(tf_activities.mean,TF))
tf_activities.mean.mat
tf_activities_stat.test.vcp <- tf_activities_stat.test %>% filter(group1 == "CTRL" & group2 == "VCP")
tf_activities_stat.test.a1 <- tf_activities_stat.test %>% filter(group1 == "A1" & group2 == "CTRL")
tf_activities_stat.test.vcp.a1 <- select(tf_activities_stat.test.vcp, TF, VCP = p) %>% full_join(select(tf_activities_stat.test.a1, TF, A1 = p))
tf_activities.stat <- tf_activities_stat.test.vcp.a1 %>%
  mutate(VCP.sig = case_when(VCP <= 0.05 ~ "\u2731", TRUE ~ ""),
         A1.sig =  case_when(A1 <= 0.05 ~ "\u2731", TRUE ~ ""),
         CTRL.sig = "") %>% 
  dplyr::select(TF, CTRL.sig, VCP.sig, A1.sig) 
tf_activities.stat.mat <- make_matrix(select(tf_activities.stat,-TF),pull(tf_activities.stat,TF))  # order the matrix by the annotation dataframe
tf_activities.stat.mat

tf_activities.mean_heatmap <- pheatmap(tf_activities.mean.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 6, fontsize_row = 10, fontsize_col = 10, display_numbers = tf_activities.stat.mat)
tf_activities.mean_heatmap
ggsave(tf_activities.mean_heatmap[[4]], filename = "/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/master_regulators.viper_dorothea.vcp_a1_together.heatmap.png", height = 7, width = 3)


### run_viper on VCP & A1 separately then merge later
# gene expression objects
vsd_ac_who_vcp_vs_ctrl <- readRDS("/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-fractionation-bulk-rnaseq/deseq2/vsd_ac_who_vcp_vs_ctrl.RDS")
colnames(vsd_ac_who_vcp_vs_ctrl) <- colnames(vsd_ac_who_vcp_vs_ctrl) %>% gsub("whole_", "", .) %>% gsub("5", "2", .) %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
rownames(vsd_ac_who_vcp_vs_ctrl) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(vsd_ac_who_vcp_vs_ctrl)] # convert ensID to gene_name as viper needs HGNC/MGI symbols
vsd_ac_bar_a1_vs_a0 <- readRDS("/Volumes/lab-luscomben/home/users/ziffo/downloaded-public-data/reactive-astrocyte-barbar-2020/expression/deseq/vsd_ac_bar_a1_vs_a0.RDS")
rownames(vsd_ac_bar_a1_vs_a0) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(vsd_ac_bar_a1_vs_a0)]

# viper wrapper in dorothea package - takes 20 mins. https://saezlab.github.io/dorothea/reference/run_viper.html
tf_activities.vcp <- run_viper(assay(vsd_ac_who_vcp_vs_ctrl), regulons,options =  list(method = "scale", minsize = 25, eset.filter = FALSE, cores = 4)) # options passed to viper https://rdrr.io/bioc/viper/man/viper.html
tf_activities.vcp # matrix of normalised enirchment score for each TF across all samples
saveRDS(tf_activities.vcp, "/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/dorothea_hs.tf_activities.vcp.RDS")

tf_activities.a1 <- run_viper(assay(vsd_ac_bar_a1_vs_a0), regulons, options =  list(method = "scale", minsize = 25, eset.filter = FALSE, cores = 4)) # options passed to viper https://rdrr.io/bioc/viper/man/viper.html
tf_activities.a1 # matrix of normalised enirchment score for each TF across all samples
saveRDS(tf_activities.a1, "/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/dorothea_hs.tf_activities.a1.RDS")

# calculate mean NES per group
tf_activities.mean <- as_tibble(tf_activities.vcp, rownames = "TF") %>% full_join(as_tibble(tf_activities.a1, rownames = "TF"), by = "TF") %>% 
  mutate(CTRL = (ctrl1 + ctrl2 + ac_bar_a0m + ac_bar_a0f + ac_bar_a0u)/5,
         VCP = (R191Q + R155C)/2,
         A1 = (ac_bar_a1m + ac_bar_a1f + ac_bar_a1u)/3) %>% 
  arrange((VCP+A1)) %>% dplyr::select(TF, CTRL, VCP, A1)

# calculate p value from t-test within each condition
tf_activities.long <- tf_activities %>% as_tibble(rownames = "TF") %>% pivot_longer(!TF, names_to = "sample", values_to = "nes") %>% 
  mutate(group = case_when(grepl("ctrl", sample) ~ "CTRL", grepl("ac_bar_a0", sample) ~ "CTRL", grepl("R1", sample) ~ "VCP", grepl("ac_bar_a1", sample) ~ "A1")) 
tf_activities.long <- tf_activities.long %>% mutate(TF = factor(tf_activities.long$TF, levels = tf_activities.mean$TF))
  
tf_activities_stat.test <- tf_activities.long %>%
  group_by(TF) %>%
  t_test(nes ~ group)  %>% # adjust_pvalue(method = "bonferroni") %>%
  add_significance()
tf_activities_stat.test

# heatmap per condition
tf_activities.mean.mat <- make_matrix(select(tf_activities.mean,-TF),pull(tf_activities.mean,TF))
tf_activities.mean.mat
tf_activities.stat.mat <- tf_activities_stat.test %>% 
  mutate(VCP.sig = case_when(group1 == "VCP" & group2 == "CTRL" & p <= 0.05 ~ "\u2217",
                             group1 == "CTRL" & group2 == "VCP" & p <= 0.05 ~ "\u2217",
                             TRUE ~ ""),
         A1.sig = case_when(group1 == "A1" & group2 == "CTRL" & p <= 0.05 ~ "\u2217",
                             group1 == "CTRL" & group2 == "A1" & p <= 0.05 ~ "\u2217",
                             TRUE ~ "")) %>% 
  dplyr::select(TF, VCP.sig, A1.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
tf_activities.stat.mat

tf_activities.mean_heatmap <- pheatmap(tf_activities.mean.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 6, fontsize_row = 10, fontsize_col = 10)#, display_numbers = tf_activities.stat.mat)
tf_activities.mean_heatmap
ggsave(tf_activities.mean_heatmap[[4]], filename = "/Volumes/lab-luscomben/home/users/ziffo/projects/astrocyte-meta-analysis/expression/deseq2/ac-vcp-barbar-overlap/master_regulators.viper_dorothea.vcp_a1_separate.heatmap.png", height = 7, width = 3)
```

#### TF Gene expression heatmap

```{r}
# Heatmap of gene expression in VCP & A1
master.regulators <- tf_activities.mean$TF
paste('"',unique(tf_activities.mean$TF),'",', sep = "", collapse = '\n') %>% cat()

dge_res_ac_vcp_a1 <- dplyr::select(patani.res, gene_name, VCP = log2FoldChange) %>% full_join(dplyr::select(barbar.res, gene_name, A1 = log2FoldChange), by = c("gene_name")) %>% filter(gene_name %in% master.regulators) %>% arrange(factor(gene_name, levels = master.regulators)) 
dge_res_ac_vcp_a1.mat <- make_matrix(select(dge_res_ac_vcp_a1,-gene_name),pull(dge_res_ac_vcp_a1,gene_name))
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat == NA] <- 0
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat > 2.0] <- 2.0
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat < -2.0] <- -2.0
dge_res_ac_vcp_a1_padj <- dplyr::select(patani.res, gene_name, VCP = padj) %>% full_join(dplyr::select(barbar.res, gene_name, A1 = padj), by = c("gene_name")) %>% filter(gene_name %in% master.regulators) %>% arrange(factor(gene_name, levels = master.regulators)) %>% 
  mutate(VCP.sig = case_when(VCP <= 0.05 ~ "\u2217", TRUE ~ ""), A1.sig = case_when(A1 <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
  dplyr::select(gene_name, VCP.sig, A1.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
dge_res_ac_vcp_a1_heatmap <- pheatmap(dge_res_ac_vcp_a1.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, display_numbers =dge_res_ac_vcp_a1_padj) 
                  # color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList, cellheight = 9, cellwidth = 12, border_color=NA, legend = TRUE)
dge_res_ac_vcp_a1_heatmap
ggsave(dge_res_ac_vcp_a1_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/master_regulators.viper_dorothea.vcp_a1_gene_exp.heatmap.png", height = 7, width = 3)
```



### Clustering {.tabset .tabset-fade .tabset-pills}
***


```{r}
# PCA
pcaData <- plotPCA(vsd, intgroup=c("condition"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
pca <- ggplot(pcaData, aes(PC1, PC2, color=condition)) + geom_text_repel(aes(label=colnames(vsd)), size = 3) + 
  scale_colour_manual( values = get_palette("npg", 10)) +  theme_bw() +  
  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.2, 'cm')) + geom_point(size=1) + 
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +  ylab(paste0("PC2: ",percentVar[2],"% variance")) + coord_fixed()
pca
ggsave(pca, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/pca.png", height = 4, width = 6)

# Dendrogram
sampleDists <- dist(t(assay(vsd)))
hc <- hclust(sampleDists, method = "ward.D2")
a <- c("black","black","firebrick2", "firebrick2", "firebrick2",,"firebrick2", "firebrick2","black","black", "black")
dendrogram <- ggdendrogram(hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = a), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ggsave(dendrogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/dendrogram.png", height = 5, width = 2)

# Gene clustering according to top 40 variable genes
topVarGenes <- head(order(-rowVars(assay(vsd))),25)
gen <- assay(vsd)[topVarGenes, ]
rownames(gen) <- ens_to_gene(rownames(gen)) # annotate with gene gene_name
gen.clust <- pheatmap(gen, scale = "row", cluster_rows=FALSE, show_rownames=TRUE, cluster_cols=TRUE, main = "", fontsize = 10, silent = TRUE)
ggsave(gen.clust, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/gen.clust.png", height = 6, width = 6)


# # Sample2sample heatmap
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
sample2sample <- pheatmap(sampleDistMatrix,clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, fontsize = 6, col=colorRampPalette( rev(brewer.pal(9, "Blues")) )(255), silent = TRUE)
sample2sample
ggsave(sample2sample, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/sample2sample.png", height = 6, width = 6)
```

#### Results {.tabset .tabset-fade .tabset-pills}
***

Since most common genes were involved in immune function he thinks that this may contribute to phagocytosis and glutamate uptake reduction in function, regardless of the GO terms (as they are not exhaustive lists).

This was the idea for the manuscript he had: 
1) VCP ACs exhibits a reactive phenotype and impaired phagocytosis and glutamate uptake.
2) the differentially expressed genes highlight immune assets, we therefore asked if inflammatory activation can directly perturb phagocytosis and glutamate uptake in control ACs - the answer is yes (largely known / predictable, but still important)
3) Overlap between mutant and A1 - list of 60 genes all regulated positively or negatively by X (Y and Z), which are potentially master regulators e.g. NfKB, STAT3 etc. We had 4 genes that are commonly implicated as showing reduced IR / increased expression across VCP, SOD1 and C9ORF72, so these may be good to start with / target
4) KD / pharmacologically antagonise (or overexpress / stimulate) X in control it has an effect on reactive transformation (protective or deleterious)
5) Predictably manipulate the reactive state in mutant to make them more protective and show that this is functionally important with ACM on mutant MNs looking at RBP mislocalisation and cell death.


#### Heatmap markers {.tabset .tabset-fade .tabset-pills}
***

```{r}
# lists of IN & MN gene markers
roof.plate <- c("GDF7")          
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","ZIC1","ZIC4")
motor.neuron.progenitor <- c("NKX6-1","OLIG2","GLI2","NKX2-2")
neuronal <- c("TUBB3","NEFH")
all.gene.markers <- c(roof.plate, dorsal.progenitor, motor.neuron.progenitor, neuronal)
all.gene.markers.df = data.frame("gene_name" = all.gene.markers) %>% 
  mutate(group = case_when(gene_name %in% roof.plate ~ "roof.plate",
                           gene_name %in% dorsal.progenitor ~ "dorsal.progenitor",
                           gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor",
                           gene_name %in% neuronal ~ "neuronal"))
annot <- data.frame(row.names = all.gene.markers.df$gene_name, group = factor(all.gene.markers.df$group))# Annotation for IN & MN markers
# ordered matrix of IN & MN markers
ctrl.d18.vsd_counts.in.mn.marker <- assay(vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id") %>% filter(gene_name %in% all.gene.markers) %>%
  mutate(`RA + Shh` = (`RA + Shh ctrl1` + `RA + Shh ctrl2` + `RA + Shh ctrl3`) / 3, `RA Only` = (`RA Only ctrl1` + `RA Only ctrl2` + `RA Only ctrl3`) / 3) %>%  dplyr::select(gene_name, `RA + Shh`, `RA Only`)
ctrl.d18.vsd_counts.in.mn.marker <- ctrl.d18.vsd_counts.in.mn.marker %>% arrange(factor(gene_name, levels = rownames(annot))) # order the matrix by the annotation dataframe
ctrl.d18.vsd_counts.in.mn.marker.mat <- make_matrix(dplyr::select(ctrl.d18.vsd_counts.in.mn.marker,-gene_name), pull(ctrl.d18.vsd_counts.in.mn.marker,gene_name))
# ctrl.d18.vsd_counts.in.mn.marker.mat[ctrl.d18.vsd_counts.in.mn.marker.mat > 5] <- 5 # to improve colour distribution cap outliers
ctrl.d18_in.mn.marker_heatmap <- pheatmap(ctrl.d18.vsd_counts.in.mn.marker.mat, show_rownames=T, show_colnames = T, cluster_rows=F, cluster_cols=F, fontsize = 8, gaps_row = c(1,10,14), cellheight = 8, cellwidth = 24, legend = TRUE)
ctrl.d18_in.mn.marker_heatmap.gg <- as.ggplot(ctrl.d18_in.mn.marker_heatmap) + 
  annotate("text", x = 0.115, y = 0.91, label = "roof plate", size = 2) + 
  annotate("text", x = 0.115, y = 0.71, label = "dorsal progen", size = 2) + 
  annotate("text", x = 0.115, y = 0.44, label = "ventral progen", size = 2) +
  annotate("text", x = 0.115, y = 0.31, label = "neuronal", size = 2)
ctrl.d18_in.mn.marker_heatmap.gg
ggsave(ctrl.d18_in.mn.marker_heatmap.gg, filename = "/camp/lab/luscomben/home/users/ziffo/projects/inter-neuron-bulk-rnaseq/expression/deseq/figures/ctrl.d18_in.mn.marker_heatmap.gg.png", height = 3.0, width = 3.5)


# HOX gene markers: HOXA1 as a marker of upper rhombomeres, HOXB4 and HOXA5 for lower cervical/brachial, HOXC6 and HOXC8 for brachial motor neuron column, HOXC9 for thoracic, and HOXD10 for lumbar.
HOX.markers <- c("HOXA1", "HOXB4", "HOXA5", "HOXC6", "HOXC8", "HOXC9", "HOXD10")
HOX.annot <- data.frame(row.names = HOX.markers, group = factor(HOX.markers))# Annotation for IN & MN markers
ctrl.d18.vsd_counts.in.mn.HOX.marker <- assay(vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id") %>% filter(gene_name %in% HOX.markers) %>%
  mutate(`RA + Shh` = (`RA + Shh ctrl1` + `RA + Shh ctrl2` + `RA + Shh ctrl3`) / 3, `RA Only` = (`RA Only ctrl1` + `RA Only ctrl2` + `RA Only ctrl3`) / 3) %>%  dplyr::select(gene_name, `RA + Shh`, `RA Only`)
ctrl.d18.vsd_counts.in.mn.HOX.marker <- ctrl.d18.vsd_counts.in.mn.HOX.marker %>% arrange(factor(gene_name, levels = rownames(HOX.annot))) # order the matrix by the annotation dataframe
ctrl.d18.vsd_counts.in.mn.HOX.marker.mat <- make_matrix(dplyr::select(ctrl.d18.vsd_counts.in.mn.HOX.marker,-gene_name), pull(ctrl.d18.vsd_counts.in.mn.HOX.marker,gene_name))
ctrl.d18.in.mn_HOX.marker.heatmap <- pheatmap(ctrl.d18.vsd_counts.in.mn.HOX.marker.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8, cellheight = 8, cellwidth = 24, legend = TRUE)
ggsave(ctrl.d18.in.mn_HOX.marker.heatmap, filename = "/camp/lab/luscomben/home/users/ziffo/projects/inter-neuron-bulk-rnaseq/expression/deseq/figures/ctrl.d18.in.mn_HOX.marker.heatmap.png", height = 2, width = 2)
```



#### Barplot markers {.tabset .tabset-fade .tabset-pills}
***

```{r}
# Top markers barplot
topmarkers <- c("OLIG3", "MSX1", "PAX3", "PAX7", "LBX1", "ZIC1", "ZIC4",
                "NKX6-1", "OLIG2", "LHX3", "MNX1", "NEUROG3")
ctrl.d18.normalized_counts.in.mn.marker <- counts(dds.ctrl.d18, normalized=TRUE) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens, by = "gene_id") %>% filter(gene_name %in% topmarkers) %>%
  dplyr::select(gene_name, `RA + Shh ctrl1`, `RA + Shh ctrl2`, `RA + Shh ctrl3`, `RA Only ctrl1`, `RA Only ctrl2`, `RA Only ctrl3`) %>%
  dplyr::mutate_if(is.numeric, funs(log2(.))) %>% dplyr::mutate_if(is.numeric, funs(ifelse(. <0, 0, .)))# %>% dplyr::mutate_if(is.numeric, funs(. + 1)) # log2 transform & add pseduocount of 1 to get rid of negative values
ctrl.d18.normalized_counts.in.mn.marker.data <- ctrl.d18.normalized_counts.in.mn.marker %>% 
  pivot_longer(!gene_name, names_to = "sample", values_to = "count") %>% 
  mutate(induction = case_when(grepl("Shh", sample) ~ "RA + Shh", grepl("Only", sample) ~ "RA Only")) 
ctrl.d18.normalized_counts.in.mn.marker.data <- ctrl.d18.normalized_counts.in.mn.marker.data %>% mutate(gene_name = factor(ctrl.d18.normalized_counts.in.mn.marker.data$gene_name, levels = topmarkers))

ctrl.d18_stat.test <- ctrl.d18.normalized_counts.in.mn.marker.data %>%
  group_by(gene_name) %>%
  t_test(count ~ induction)  %>% # adjust_pvalue(method = "bonferroni") %>%
  add_significance() %>%
  add_xy_position(x = "day", fun = "mean_sd", dodge = 0.9)
ctrl.d18_stat.test

ctrl.d18_topmarker.barplot <- ggbarplot(
  ctrl.d18.normalized_counts.in.mn.marker.data, x = "induction", y = "count", fill = "induction", palette = c('black','firebrick2'), add = "mean", position = position_dodge(0.9),facet.by = "gene_name") +
  stat_summary(aes(x = induction, y = count, fill = induction), fun.min = mean, fun.max = function(x) mean(x) + sd(x), geom="errorbar", color="black", width=0.2, position=position_dodge(0.9)) + 
  stat_pvalue_manual(ctrl.d18_stat.test, label = "p.signif", tip.length = 0.05, hide.ns = TRUE) +
  scale_y_continuous(expand = expansion(mult = c(0.01, 0.1))) + theme(legend.title = element_blank(), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.2, 'cm')) + 
  ylab(label = expression(Log[2]~normalised~gene~counts) ) + xlab(label = "") 
ctrl.d18_topmarker.barplot
ggsave(ctrl.d18_topmarker.barplot, filename = "/camp/lab/luscomben/home/users/ziffo/projects/inter-neuron-bulk-rnaseq/expression/deseq/figures/ctrl.d18_topmarker.barplot.png", height = 8, width = 8)
```



```{r}
# histogram of differentially expressed genes
patani.dge_numbers <- c(sum(patani.res$padj <= 0.05 & patani.res$log2FoldChange >= 0, na.rm=TRUE), sum(patani.res$padj <= 0.05 & patani.res$log2FoldChange <= 0, na.rm=TRUE))
patani.dge_summary  <- as.data.frame(matrix(data = patani.dge_numbers , nrow = 2, ncol = 1))
colnames(patani.dge_summary) <- "de_genes"
patani.dge_summary$direction <- c("Up","Down")
patani.dge_summary$sample <- "patani"
patani.dge_histogram <- ggbarplot(patani.dge_summary, x="sample" , y="de_genes", fill="direction", label = TRUE, lab.pos = "in", position = position_dodge(0.75), xlab = "") + 
  scale_fill_manual( values = c("dodgerblue3", "firebrick2") ) +  theme_bw() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top") + ylab(expression(Numbers~of~DEGs~FDR<0.05)) 
patani.dge_histogram
ggsave(patani.dge_histogram, filename = "/camp/lab/luscomben/home/users/ziffo/projects/inter-neuron-bulk-rnaseq/expression/deseq/figures/patani.dge_histogram.png", height = 6, width = 6)
```


## Overlap Gene Expression 

### Histogram

```{r}
# histogram of differentially expressed genes
ac_who.dge_numbers <- c(sum(dge_res_ac_who_vcp_vs_ctrl$padj <= 0.05 & dge_res_ac_who_vcp_vs_ctrl$log2FoldChange >= 0, na.rm=TRUE), sum(dge_res_ac_who_vcp_vs_ctrl$padj <= 0.05 & dge_res_ac_who_vcp_vs_ctrl$log2FoldChange <= 0, na.rm=TRUE))
ac_who.dge_summary  <- as.data.frame(matrix(data = ac_who.dge_numbers , nrow = 2, ncol = 1))
colnames(ac_who.dge_summary) <- "de_genes"
ac_who.dge_summary$direction <- c("Up","Down")
ac_who.dge_summary$sample <- "VCP vs CTRL"

a1.dge_numbers <- c(sum(dge_res_ac_bar_a1_vs_a0$padj <= 0.05 & dge_res_ac_bar_a1_vs_a0$log2FoldChange >= 0, na.rm=TRUE), sum(dge_res_ac_bar_a1_vs_a0$padj <= 0.05 & dge_res_ac_bar_a1_vs_a0$log2FoldChange <= 0, na.rm=TRUE))
a1.dge_summary  <- as.data.frame(matrix(data = a1.dge_numbers , nrow = 2, ncol = 1))
colnames(a1.dge_summary) <- "de_genes"
a1.dge_summary$direction <- c("Up","Down")
a1.dge_summary$sample <- "Stimulated vs Untreated"

ac_who.a1.dge_summary <- bind_rows(ac_who.dge_summary, a1.dge_summary)
ac_who.a1.dge_histogram <- ggbarplot(ac_who.a1.dge_summary, x="sample" , y="de_genes", fill="direction", label = TRUE, lab.pos = "out", position = position_dodge(0.75), xlab = "") + 
  scale_fill_manual( values = c("dodgerblue3", "firebrick2") ) +  theme_bw() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top") + 
  ylab(expression(Numbers~of~DEGs~FDR<0.05))
ac_who.a1.dge_histogram
ggsave(ac_who.a1.dge_histogram, filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/ac_who.a1.dge_histogram.png", height = 6, width = 6)
```


## Gene sets of interest

```{r}
# Phagocytosis

# Glutamate update
# als_a1.dge %>% filter(padj_vcp < 0.05, padj_a1 < 0.05) %>% pull(gene_name)
# als_a1.dge %>% filter(padj_vcp < 0.05, padj_a1 < 0.05) %>% filter(gene_name %in% all.reactivity.go) # "FAM107A" "SLC15A4" "GJB2"    "CDH1"    "TEK"     "DDIT4"   "ABCA13"  "IFITM1"  "ADRA1A"  "ANGPT1"  "ACTC1"   "PTPRT"   "COL4A4"  "AMIGO2"  "COL26A1" "ABI3BP"  "MYH11" 
# als_a1.dge %>% filter(padj_vcp < 0.05, padj_a1 < 0.05) %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS)
# als_a1.dge %>% filter(padj_vcp < 0.05, padj_a1 < 0.05) %>% filter(gene_name %in% glutamate.uptake.go) # CDH1, ADRA1A, ACTC1, PRPRT

```

#### Additional annalyses

```{r}
dorothea_hs.regulon <- df2regulon(dorothea_hs)
saveRDS(dorothea_hs.regulon, "/Volumes/lab-luscomben/home/users/ziffo/genomes/transcription-factors/dorothea_hs.regulon.RDS")
write.csv(dorothea_hs, "/Volumes/lab-luscomben/home/users/ziffo/genomes/transcription-factors/dorothea_hs.csv")
dorothea_hs.A.B = dorothea_hs %>% dplyr::filter(confidence %in% c("A", "B"))
dorothea_hs.A.B.regulon <- df2regulon(dorothea_hs.A.B) # https://github.com/saezlab/ConservedFootprints/blob/3cee21853a90f78bd13b6eebcb5538fc4c129cab/src/dorothea_analysis.R
saveRDS(dorothea_hs.A.B.regulon, "/Volumes/lab-luscomben/home/users/ziffo/genomes/transcription-factors/dorothea_hs.A.B.regulon.RDS")
dorothea_hs.A.B.regulon <- readRDS("/camp/home/ziffo/home/genomes/transcription-factors/dorothea_hs.A.B.regulon.RDS")



gene2ens <- readRDS("/camp/home/ziffo/home/genomes/ensembl/Homo_sapiens.GRCh38.99.gene2ens.RDS") %>% unique
vsd_ac_who_vcp_vs_ctrl <- readRDS("/camp/lab/luscomben/home/users/ziffo/projects/astrocyte-fractionation-bulk-rnaseq/expression/deseq2/vsd_ac_who_vcp_vs_ctrl.RDS")
colnames(vsd_ac_who_vcp_vs_ctrl) <- colnames(vsd_ac_who_vcp_vs_ctrl) %>% gsub("whole_", "", .) %>% gsub("5", "2", .) %>% gsub("cb1e", "R191Q", .) %>% gsub("glia", "R155C", .)
rownames(vsd_ac_who_vcp_vs_ctrl) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(vsd_ac_who_vcp_vs_ctrl)]
vsd_ac_bar_a1_vs_a0 <- readRDS("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data/reactive-astrocyte-barbar-2020/expression/deseq/vsd_ac_bar_a1_vs_a0.RDS")
rownames(vsd_ac_bar_a1_vs_a0) <- gene2ens$gene_name[gene2ens$gene_id %in% rownames(vsd_ac_bar_a1_vs_a0)]
make_matrix <- function(df,rownames = NULL){
  my_matrix <-  as.matrix(df)
  if(!is.null(rownames))
    rownames(my_matrix) = rownames
  my_matrix
}
vsd_bar_vcp <- as_tibble(assay(vsd_ac_who_vcp_vs_ctrl), rownames = "gene_name") %>% full_join(as_tibble(assay(vsd_ac_bar_a1_vs_a0), rownames = "gene_name"), by = "gene_name")
vsd_bar_vcp.mat <- make_matrix(select(vsd_bar_vcp,-gene_name),pull(vsd_bar_vcp,gene_name))






# 1. Generate gene expression signature - subset columns for samples to compare
ctrl.cols <- which(colnames(vsd_bar_vcp.mat) %in% c("ctrl1", "ctrl2"))
vcp.cols <- which(colnames(vsd_bar_vcp.mat) %in% c("R191Q", "R155C"))
a0.cols <- which(colnames(vsd_bar_vcp.mat) %in% c("ac_bar_a0m", "ac_bar_a0f", "ac_bar_a0u"))
a1.cols <- which(colnames(vsd_bar_vcp.mat) %in% c("ac_bar_a1m", "ac_bar_a1f", "ac_bar_a1u"))
signature.vcp <- rowTtest(vsd_bar_vcp.mat[, ctrl.cols], vsd_bar_vcp.mat[, vcp.cols]) # rowTest performs t-test comparing two matrices row-by-row
signature.vcp <- (qnorm(signature.vcp$p.value/2, lower.tail=F) * sign(signature.vcp$statistic))[, 1] # estimate z-score values for signatures
signature.a1 <- rowTtest(vsd_bar_vcp.mat[, a0.cols], vsd_bar_vcp.mat[, a1.cols]) # rowTest performs t-test comparing two matrices row-by-row
signature.a1 <- (qnorm(signature.a1$p.value/2, lower.tail=F) * sign(signature.a1$statistic))[, 1] # estimate z-score values for signatures

# a0.cols <- which(colnames(assay(vsd_ac_bar_a1_vs_a0)) %in% c("ac_bar_a0m", "ac_bar_a0f", "ac_bar_a0u"))
# a1.cols <- which(colnames(assay(vsd_ac_bar_a1_vs_a0)) %in% c("ac_bar_a1m", "ac_bar_a1f", "ac_bar_a1u"))
# signature.a1 <- rowTtest(assay(vsd_ac_bar_a1_vs_a0)[, a0.cols], assay(vsd_ac_bar_a1_vs_a0)[, a1.cols]) # rowTest performs t-test comparing two matrices row-by-row
# signature.a1 <- (qnorm(signature.a1$p.value/2, lower.tail=F) * sign(signature.a1$statistic))[, 1] # estimate z-score values for signatures

# 2. NULL model by sample permutations - permute samples at random to account for the correlation structure between genes
nullmodel.vcp <- ttestNull(x = vsd_bar_vcp.mat[, vcp.cols], y= vsd_bar_vcp.mat[, ctrl.cols], per=1000, repos=T)
nullmodel.a1 <-  ttestNull(x = vsd_bar_vcp.mat[, a1.cols], y = vsd_bar_vcp.mat[, a0.cols], per=1000, repos=T)

# 3. msVIPER analysis estimates normalised enrichment score & p-value
mrs.vcp <- msviper(signature.vcp, dorothea_hs.A.B.regulon, nullmodel.vcp)
mrs.a1 <-  msviper(signature.a1, dorothea_hs.A.B.regulon, nullmodel.a1)

summary(mrs.vcp) # top 10 master regulators
summary(mrs.a1) # top 10 master regulators
plot(mrs.vcp, cex=.7)
plot(mrs.a1, cex=.7)

# 4. Leading-edge analysis - identify genes enriched in the enrichment signature
mrs.vcp <- ledge(mrs.vcp)
summary(mrs.vcp)
mrs.a1 <- ledge(mrs.a1)
summary(mrs.a1)

enframe(mrs.a1$es$size, name = "TF", value = "size") %>% full_join(enframe(mrs.vcp$es$nes, name = "TF", value = "VCP.nes"), by = "TF") %>% full_join(enframe(mrs.vcp$es$p.value, name = "TF", value = "VCP.pvalue"), by = "TF") %>% full_join(enframe(mrs.a1$es$nes, name = "TF", value = "A1.nes"), by = "TF") %>% full_join(enframe(mrs.a1$es$p.value, name = "TF", value = "A1.pvalue"), by = "TF") %>% write.csv("/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/master_regulators.viper_dorothea.vcp_a1.csv")

# 5. Heatmap of NES in VCP & A1
mrs.vcp.a1.nes <- enframe(mrs.vcp$es$nes, name = "gene_name", value = "VCP") %>% full_join(enframe(mrs.a1$es$nes, name = "gene_name", value = "A1"), by = "gene_name") %>% arrange((VCP+A1))
mrs.vcp.a1.nes.mat <- make_matrix(select(mrs.vcp.a1.nes,-gene_name),pull(mrs.vcp.a1.nes,gene_name))
mrs.vcp.a1.nes.mat
# mrs.vcp.a1.pvalue.mat <- enframe(mrs.vcp$es$p.value, name = "gene_name", value = "VCP") %>% full_join(enframe(mrs.a1$es$p.value, name = "gene_name", value = "A1"), by = "gene_name") %>% mutate(VCP.sig = case_when(VCP <= 0.05 ~ "\u2217", TRUE ~ "")) %>% dplyr::select(sig) %>%  as.matrix  # order the matrix by the annotation dataframe
# mrs.vcp.a1.pvalue.mat
mrs.vcp.a1_heatmap <- pheatmap(mrs.vcp.a1.nes.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 6, fontsize_row = 10, fontsize_col = 10)#, display_numbers = mrs.vcp.a1.pvalue.mat)
mrs.vcp.a1_heatmap
ggsave(mrs.vcp.a1_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/master_regulators.viper_dorothea.vcp_a1_nes.heatmap.png", height = 7, width = 3)

# Heatmap of gene expression in VCP & A1
master.regulators <- rownames(mrs.vcp.a1.nes.mat)
dge_res_ac_vcp_a1 <- dplyr::select(patani.res, gene_name, VCP = log2FoldChange) %>% full_join(dplyr::select(barbar.res, gene_name, A1 = log2FoldChange), by = c("gene_name")) %>% filter(gene_name %in% master.regulators) %>% arrange(factor(gene_name, levels = master.regulators)) 
dge_res_ac_vcp_a1.mat <- make_matrix(select(dge_res_ac_vcp_a1,-gene_name),pull(dge_res_ac_vcp_a1,gene_name))
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat == NA] <- 0
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat > 2.0] <- 2.0
dge_res_ac_vcp_a1.mat[dge_res_ac_vcp_a1.mat < -2.0] <- -2.0
# vcp_dge_res_padj <- ac_who_vcp_vs_ctrl.all %>% filter(gene_name %in% astrocyte.reactive.genes) %>% arrange(IR.lfc) %>% distinct(gene_name, .keep_all = TRUE) %>%
#   mutate(IR.sig = case_when(p0.05_reliable == "significant" ~ "\u2217", TRUE ~ ""), DGE.sig = case_when(padj <= 0.05 ~  "\u2217", TRUE ~ "")) %>%
#   dplyr::select(gene_name, IR.sig, DGE.sig) %>%
#   arrange(factor(gene_name, levels = rownames(annot))) %>% dplyr::select(IR.sig,DGE.sig) %>%  as.matrix  # order the matrix by the annotation dataframe
dge_res_ac_vcp_a1_heatmap <- pheatmap(dge_res_ac_vcp_a1.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 8)#, display_numbers = t(vcp_dge_res_padj), 
                  # color = colorRampPalette(rev(greenred(75)))(length(breaksList)),breaks = breaksList, cellheight = 9, cellwidth = 12, border_color=NA, legend = TRUE)
dge_res_ac_vcp_a1_heatmap
ggsave(dge_res_ac_vcp_a1_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/master_regulators.viper_dorothea.vcp_a1_gene_exp.heatmap.png", height = 7, width = 3)


# VIPER with null model - use control samples as reference for differences in gene expression
ctrl.cols <- which(colnames(assay(vsd_ac_who_vcp_vs_ctrl)) %in% c("ctrl1", "ctrl2"))
vcp.cols <- which(colnames(assay(vsd_ac_who_vcp_vs_ctrl)) %in% c("R191Q", "R155C"))
vpsig.vcp <- viperSignature(eset = assay(vsd_ac_who_vcp_vs_ctrl)[, vcp.cols], ref = assay(vsd_ac_who_vcp_vs_ctrl)[, ctrl.cols])
vpres.vcp <- viper(vpsig.vcp, dorothea_hs.A.B.regulon)

a0.cols <- which(colnames(assay(vsd_ac_bar_a1_vs_a0)) %in% c("ac_bar_a0m", "ac_bar_a0f", "ac_bar_a0u"))
a1.cols <- which(colnames(assay(vsd_ac_bar_a1_vs_a0)) %in% c("ac_bar_a1m", "ac_bar_a1f", "ac_bar_a1u"))
vpsig.a1 <- viperSignature(eset = assay(vsd_ac_bar_a1_vs_a0)[, a1.cols], ref = assay(vsd_ac_bar_a1_vs_a0)[, a0.cols])
vpres.a1 <- viper(vpsig.a1, dorothea_hs.A.B.regulon)

vpsig.ctrl <- viperSignature(eset = assay(vsd_ac_bar_a1_vs_a0)[, a0.cols], ref = assay(vsd_ac_who_vcp_vs_ctrl)[, ctrl.cols])
vpres.ctrl <- viper(vpsig.ctrl, dorothea_hs.A.B.regulon)

vpres.vcp.a1 <- vpres.ctrl %>% as_tibble(rownames = "TF") %>% full_join(as_tibble(vpres.vcp, rownames = "TF"), by = "TF") %>% full_join(as_tibble(vpres.a1, rownames = "TF"), by = "TF")  %>% mutate(CTRL =((ac_bar_a0m + ac_bar_a0f + ac_bar_a0u) /3), VCP = ((R191Q + R155C)/2), A1 = ((ac_bar_a1m + ac_bar_a1f + ac_bar_a1u) /3)) %>% select(TF, CTRL, VCP, A1) %>% arrange((VCP+A1))
vpres.vcp.a1.mat <- make_matrix(select(vpres.vcp.a1,-TF),pull(vpres.vcp.a1,TF))
vpres.vcp.a1.mat[vpres.vcp.a1.mat > 2.0] <- 2.0
vpres.vcp.a1.mat[vpres.vcp.a1.mat < -2.0] <- -2.0
vpres.vcp.a1_heatmap <- pheatmap(vpres.vcp.a1.mat, cluster_rows=F, show_rownames=T, show_colnames = T, cluster_cols=F, fontsize = 6, fontsize_row = 10, fontsize_col = 10)#, display_numbers = mrs.vcp.a1.pvalue.mat)
vpres.vcp.a1_heatmap
ggsave(vpres.vcp.a1_heatmap[[4]], filename = "/camp/home/ziffo/home/projects/astrocyte-a1-als-meta/expression/deseq2/master_regulators.viper_dorothea.vcp_a1_vpres.heatmap.png", height = 7, width = 4)


# unsupervised heirarchical clustering
library(ballgown)
pos <- ballgown::pData(vpres.a1)[["description"]] %in% c("M", "CB", "CC")
d1 <- exprs(vpres)[, pos]
colnames(d1) <- pData(vpres)[["description"]][pos]
dd <- dist(t(d1), method = "euclidean")
heatmap(as.matrix(dd), Rowv = as.dendrogram(hclust(dd, method = "average")), symm = T)
dd <- viperSimilarity(d1)
heatmap(as.matrix(as.dist(dd)), Rowv = as.dendrogram(hclust(as.dist(dd),method = "average")), symm = T)


# 7. Shadow analysis - check for confounding where a regulator appears to be activated based on its regulon's analysis.
 mrshadow <- shadow(mrs, pval=25)
 summary(mrshadow)
 
 # 8. Synergy analysis - predict interactions between regulators
 mrs <- marinaCombinatorial(mrs, regulators=25) # calculate enrichment of co-regulons for top 25 regulators - intersections between regulons
 mrs <- marinaSynergy(mrs) # compare enrichment of co-regulon vs union of corresponding regulons
summary(mrs)

# 9. VIPER - transform gene expression matrix to regulator activity matrix.
vpres <- viper(dset, regulon) # generate matrix of regulator activity

# 10. VIPER with NULL model
vpsig <- viperSignature(dset[, -naiveBcell], dset[, naiveBcell])
vpres <- viper(vpsig, regulon)
d1 <- vpres[, match(unlist(normalSamples[names(normalSamples) != "N"]), colnames(vpres))]
colnames(d1) <- rep(c("M", "CB", "CC"), c(5, 5, 5))
dd <- dist(t(d1), method="euclidean")
heatmap(as.matrix(dd), Rowv=as.dendrogram(hclust(dd, method="average")), symm=T)
dd <- signatureDistance(d1, scale.=F)
heatmap(as.matrix(as.dist(dd)), Rowv=as.dendrogram(hclust(as.dist(dd), method="average")), symm=T)

# https://www.bioconductor.org/packages/release/bioc/html/RedeR.html
```

